---
title: "Tail_Integration_col_0_1_2wpa"
author: "Tobias Gerber"
date: "7/24/2020"
output: html_document
---

mapping
```{bash}
STAR --genomeDir /mnt/SingleCellGenomics/genome_data/STAR_AxolotlGenome/AmexG_v6_chr_unscaffolded/ --readFilesIn ./123455/123455_S4_L001_R2_001.fastq.gz,./123455/123455_S4_L002_R2_001.fastq.gz ./123455/123455_S4_L001_R1_001.fastq.gz,./123455/123455_S4_L002_R1_001.fastq.gz --soloType CB_UMI_Simple --soloCBwhitelist ~/cellranger-3.1.0/cellranger-cs/3.1.0/lib/python/cellranger/barcodes/3M-february-2018.txt --soloCBmatchWLtype 1MM_multi_Nbase_pseudocounts --soloUMIfiltering MultiGeneUMI_CR --soloUMIdedup 1MM_CR --readFilesCommand zcat --outFileNamePrefix count_123455_3mpa_col1a2_Full_ --runThreadN 45 --soloMultiMappers Uniform --soloCellFilter EmptyDrops_CR --soloCBstart 1  --soloCBlen 16  --soloUMIstart 17  --soloUMIlen 12 --soloFeatures Gene GeneFull Velocyto --outSAMtype BAM Unsorted --soloBarcodeReadLength 0

```



```{r}
setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

#load annotation file

#load annotation file

anno = read.delim("/mnt/SingleCellGenomics/genome_data/10xAxolotlGenome/AnnotationFile_AmexG_v6_chr_unscaffolded_CherryGFP_v1.2.csv",sep = ";",header = F)
rownames(anno) = anno[,1]


############# remove weird transposon / unannotated genes
unannotated.transcripts = rownames(anno)[anno$V2 %in% anno$V2[grep("PEG10",anno$V2)] ]
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("L1TD1",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("RTL",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("GIN1",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("L1\\-RT",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("^N\\/A",anno$V2)] ], unannotated.transcripts))
#unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("^AMEX",anno$V2)] ], unannotated.transcripts))

anno = anno[!rownames(anno) %in% unannotated.transcripts,]


############# remove rp genes

rp.genes_AmexG = rownames(anno)[anno$V2 %in% anno$V2[grep("RPL",anno$V2)]]
rp.genes_AmexG = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("RPS",anno$V2)] ], rp.genes_AmexG))

anno = anno[!rownames(anno) %in% rp.genes_AmexG,]

```


##Merge 1wpa 2wpa

###Run Seurat
```{r}
setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

library(Seurat)

#load data sets


Tail_1wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_1wpa/Tail_col_1wpa_CT_SeuratObj.RDS")
Tail_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_2wpa/Tail_col_2wpa_CT_SeuratObj.RDS")


#merge ojects
#merge(x = NULL, y = NULL, add.cell.ids = NULL, merge.data = TRUE, project = "SeuratProject", ...)

Int_col_BL_CT = merge(Tail_1wpa, y = Tail_2wpa, add.cell.ids = c("wpa1", "wpa2"), project = "Tail_col_Integration_onlyBL")


#parallize scaling
library(future)
plan("multiprocess", workers = 8)
plan()

#set maximum to 10GB for each worker
options(future.globals.maxSize = 50000 * 1024^2)


#get cell cycle genes

g2m.genes <- cc.genes$g2m.genes
g2m.contig <- intersect(rownames(anno)[anno$V2 %in% g2m.genes ] , rownames(Int_col_BL_CT) )

s.genes <- cc.genes$s.genes
s.contig <- intersect(rownames(anno)[anno$V2 %in% s.genes ] , rownames(Int_col_BL_CT) )

#cell cycle scoring
Int_col_BL_CT <- CellCycleScoring(Int_col_BL_CT, s.features = s.contig, g2m.features = g2m.contig, set.ident = TRUE)



#scale
#ScaleData(  object,  features = NULL,  assay = NULL,  vars.to.regress = NULL,  split.by = NULL,  model.use = "linear",  use.umi = FALSE,  do.scale = TRUE,  do.center = TRUE,  scale.max = 10,  block.size = 1000,  min.cells.to.block = 3000,  verbose = TRUE)

all.genes <- rownames(Int_col_BL_CT)

Int_col_BL_CT = ScaleData(  Int_col_BL_CT, features = all.genes ,vars.to.regress = c("nCount_RNA","percent.mt"))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

Int_col_BL_CT = RunPCA(Int_col_BL_CT,features = all.genes,npcs = 100)

#plot PCA heatmaps
#DimHeatmap(pbmc, dims = 1, cells = 500, balanced = TRUE)

library(RColorBrewer)
library(ggplot2)

pdf("Int_col_onlyBL_CT_raw_PCAheatmaps.pdf",width=30,height=20)
DimHeatmap(Int_col_BL_CT, dims = 1:16, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT, dims = 17:32, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT, dims = 33:48, cells = 1000, balanced = TRUE,  ncol = 4, fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
dev.off()

#elbow plot
pdf("Int_col_onlyBL_CT_raw_PCelbow.pdf",width=20,height=10)
ElbowPlot(Int_col_BL_CT, ndims = 100)
dev.off()

#Cluster cells
Int_col_BL_CT<- FindNeighbors(Int_col_BL_CT, dims = 1:30)
Int_col_BL_CT <- FindClusters(Int_col_BL_CT, resolution = 0.6)

#run UMAP
Int_col_BL_CT = RunUMAP(Int_col_BL_CT,dims = 1:30)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_onlyBL_CT_raw_UMAP_PC30_res0.6.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT, reduction = 'umap', pt.size = 3)
DimPlot(object = Int_col_BL_CT, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
FeaturePlot(Int_col_BL_CT, pt.size = 1, features = c("nFeature_RNA","nCount_RNA","percent.mt","AMEX60DD002658"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()

#They cluster only based on batch --> integrate

saveRDS(Int_col_BL_CT, file = "Int_col_onlyBL_CT_raw_SeuratObj.RDS")

```

###Run Harmony

```{r}

setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

Int_col_BL_CT = readRDS( file = "Int_col_onlyBL_CT_raw_SeuratObj.RDS")


#library(devtools)
#install_github("immunogenomics/harmony")
library(Seurat)
library(dplyr)
library(harmony)

#Harmony

Int_col_BL_CT_int = RunHarmony(object = Int_col_BL_CT,group.by.vars= "orig.ident", assay.use = "RNA", dims.use = 1:100, plot_convergence = TRUE, max.iter.cluster = 30 ,max.iter.harmony=50,theta=2,lambda=1)



#Cluster cells
Int_col_BL_CT_int<- FindNeighbors(Int_col_BL_CT_int, dims = 1:50, reduction = "harmony")
Int_col_BL_CT_int <- FindClusters(Int_col_BL_CT_int, resolution = 0.6)


#run UMAP
Int_col_BL_CT_int = RunUMAP(Int_col_BL_CT_int,dims = 1:50, reduction = "harmony",min.dist = 0.001)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_PC100_UMAP_PC50_res06.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap', pt.size = 3,shuffle = T)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap',group.by = "orig.ident", pt.size = 3,shuffle = T)
FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD031414","AMEX60DD009987","AMEX60DD002658","AMEX60DD052322"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()

#one small oulier detected --> remove

tmp = FetchData(Int_col_BL_CT_int,c("UMAP_1","UMAP_2"))
select.cells = rownames(tmp)[tmp$UMAP_2 < 5]

Int_col_BL_CT_int = subset(Int_col_BL_CT_int,cells = select.cells)



#Cluster cells
Int_col_BL_CT_int<- FindNeighbors(Int_col_BL_CT_int, dims = 1:50, reduction = "harmony")
Int_col_BL_CT_int <- FindClusters(Int_col_BL_CT_int, resolution = 0.8)


#run UMAP
Int_col_BL_CT_int = RunUMAP(Int_col_BL_CT_int,dims = 1:50, reduction = "harmony",min.dist = 0.001)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_PC100_UMAP_PC50_res08.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap', pt.size = 3,shuffle = T)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap',group.by = "orig.ident", pt.size = 3,shuffle = T)
FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD031414","AMEX60DD009987","AMEX60DD002658","AMEX60DD052322"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


#one super low nUMI cluster only 2wpa cells --> remove

Int_col_BL_CT_int = subset(Int_col_BL_CT_int,idents = c(0,1,2,3,4,5,6,7,8,9,11,12,13,14,15))



#Cluster cells
Int_col_BL_CT_int<- FindNeighbors(Int_col_BL_CT_int, dims = 1:50, reduction = "harmony")
Int_col_BL_CT_int <- FindClusters(Int_col_BL_CT_int, resolution = 0.6)


#run UMAP
Int_col_BL_CT_int = RunUMAP(Int_col_BL_CT_int,dims = 1:50, reduction = "harmony",min.dist = 0.1)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_PC100_UMAP_PC50_res06.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap', pt.size = 3,shuffle = T)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap',group.by = "orig.ident", pt.size = 3,shuffle = T)
FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD031414","AMEX60DD009987","AMEX60DD002658","AMEX60DD052322"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


#one super low nUMI cluster only 2wpa cells --> remove

Int_col_BL_CT_int = subset(Int_col_BL_CT_int,idents = c(0,1,2,3,4,6,7,8,9,10,11))



#Cluster cells
Int_col_BL_CT_int<- FindNeighbors(Int_col_BL_CT_int, dims = 1:50, reduction = "harmony")
Int_col_BL_CT_int <- FindClusters(Int_col_BL_CT_int, resolution = 0.6)


#run UMAP
Int_col_BL_CT_int = RunUMAP(Int_col_BL_CT_int,dims = 1:50, reduction = "harmony",min.dist = 0.02,seed.use = 30L)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_PC100_UMAP_PC50_res06.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap', pt.size = 3,shuffle = T)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap',group.by = "orig.ident", pt.size = 3,shuffle = T)
FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD031414","AMEX60DD009987","AMEX60DD002658","AMEX60DD052322"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


markers <- FindAllMarkers(Int_col_BL_CT_int, only.pos = TRUE,  logfc.threshold = 0.3)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Int_col_onlyBL_CT_HarmonyTimeInt_allMarker.csv")

saveRDS(Int_col_BL_CT_int, file = "Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")

gene_ids = c("AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD023726","AMEX60DD054625","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD022355","AMEX60DD009984","AMEX60DD009989","AMEX60DD009968")
	
gene_names =c("PRRX1","RSPO3","FGF8","WNT5A","TBX2","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","DLX5","SOST","ETV4","PHOSPHO1")

gg_Fig <- FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_feature_SpecificMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()


gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	
18-21

gg_Fig <- FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_feature_GeneralMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()

gene_names = c("UNC5B",
"BMP2",
"DLX3",
"DLX5",
"MSX2",
"IL11",
"MMP11",
"HOXA13",
"SLITRK6",
"SPRY2",
"MYH11",
"BOX5-DLX6",
"MKX",
"CILP2",
"CILP",
"ASPN",
"THBS4",
"CDH15",
"ZFHX3",
"DR999_PMT18929-SCX")
gene_ids = c("AMEX60DD035718",
"AMEX60DD009952",
"AMEX60DD022355",
"AMEX60DD028758",
"AMEX60DD016524",
"AMEX60DD000336",
"AMEX60DD021953",
"AMEX60DD048713",
"AMEX60DD048778",
"AMEX60DD020580",
"AMEX60DD009950",
"AMEX60DD051903",
"AMEX60DD021849",
"AMEX60DD013919",
"AMEX60DD003432",
"AMEX60DD023964",
"AMEX60DD042297",
"AMEX60DD017092",
"AMEX60DD016992",
"AMEX60DD040818")


gg_Fig <- FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_feature_InterestingMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()


#plot markers for paper

gene_ids = c("AMEX60DD021181","AMEX60DD052891","AMEX60DD009987","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD031236","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD029436","AMEX60DD007033","AMEX60DD009984","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD018450")
	
gene_names =c("LFNG","FGF8","MEOX1","SCX","TNMD","PAX7","MYF5","SOX9","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","TWIST3","FAM180A","SOST","F13B","LTBP4","F13A1","PRRX1")

my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[rev(1:5)])


gg_Fig <- VlnPlot(Tail_1_2wpa,features = gene_ids,pt.size = 0, group.by = "celltype", cols = my_spectral_palette(length(unique(Tail_1_2wpa@meta.data$celltype))))
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_Vln_PaperMarker.pdf",width=22,height=15)
VlnPlot(Tail_1_2wpa,features = gene_ids,pt.size = 0, group.by = "celltype", cols = my_spectral_palette(length(unique(Tail_1_2wpa@meta.data$celltype))))
CombinePlots( gg_Fig )
dev.off()

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_Dot_PaperMarker.pdf",width=9,height=5)
DotPlot(Tail_1_2wpa,features = gene_ids, group.by = "celltype") +scale_colour_gradientn(colors =  c(viridis_pal(option = "plasma")(12))) 
dev.off()

#+ scale_color_manual(values = colorRampPalette(viridis_pal(option = "magma")(12))(100))
#cols = my_spectral_palette(length(unique(Tail_1_2wpa@meta.data$celltype))))





#plot some ERK genes


GO.ERK = read.delim("GObp_ERK.txt",sep = "\t",header = F)[,1]

anno.ERK = vector()

for (i in 1:length(GO.ERK)) {
  anno.ERK = rbind(anno.ERK,anno[grep(pattern = paste(GO.ERK[i],"$",sep = ""),anno$V2),])
}

tmp = intersect(anno.ERK$V1,VariableFeatures(Tail_1_2wpa_pax_filter))
anno.ERK = anno.ERK[anno.ERK$V1 %in% tmp,]


gg_Fig <- FeaturePlot(Tail_1_2wpa, pt.size = 1, features = anno.ERK$V1,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(anno.ERK$V1), function(x) { gg_Fig[[x]] + labs(title=anno.ERK$V2[x]) & NoLegend() & NoAxes()})

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_feature_ERKMarker.pdf",width=30,height=30)
CombinePlots( gg_Fig )
dev.off()



gene_names = c(
#Asomitic
"MEOX1",
"LFNG",
"MKX",
"SCX",
"TNMD",
#Progenitors 1/2wpa
"KIF26A",
"PCLAF",
"TOP2A",
"IL11",
"LEP",
#Progenitors 0wpa
#"CRABP2",
#"NGFR",
#"OTOR",
#"CDO1",
#"VWC2L",
#Committed chondrocytes
#"CHRDL1",
#"CHRDL2",
#"FAT4",
#"C17ORF67",
#"FGFR2",
#Cartilage
"CHRDL1",
"CNMD",
"ACAN",
"SOX9",
"MATN1",
"FGFBP2",
#Mesenchyme
"PRRX1",
"SOST",
"LTBP4",
"DLX5",
"DKK2",
"F13A1",
#Dermis
"DPT",
"TWIST3",
"PAH",
"FAM180A",
"OLFML2A",
#Muscle
"PAX7",
"MYF5",
"MYL1",
"MYLPF",
"ACTC1")

gene_ids = c(
#Asomitic
"AMEX60DD009987",
"AMEX60DD021181",
"AMEX60DD021849",
"AMEX60DD040818",
"AMEX60DD037674",
#Progenitors 1/2wpa
"AMEX60DD011082",
"AMEX60DD003844",
"AMEX60DD010148",
"AMEX60DD016524",
"AMEX60DD007977",
#Progenitors 0wpa
#"AMEX60DD015718",
#"AMEX60DD009964",
#"AMEX60DD035833",
#"AMEX60DD042801",
#"AMEX60DD019831",
#Committed chondrocytes
#"AMEX60DD037123",
#"AMEX60DD049819",
#"AMEX60DD044847",
#"AMEX60DD031033",
#"AMEX60DD053280",
#Cartilage
"AMEX60DD037123",
"AMEX60DD048972",
"AMEX60DD004178",
"AMEX60DD031236",
"AMEX60DD005459",
"AMEX60DD045825",
#Mesenchyme
"AMEX60DD018450",
"AMEX60DD009984",
"AMEX60DD024616",
"AMEX60DD022355",
"AMEX60DD044644",
"AMEX60DD038300",
#Dermis
"AMEX60DD047546",
"AMEX60DD029436",
"AMEX60DD008421",
"AMEX60DD007033",
"AMEX60DD050542",
#Muscle
"AMEX60DD052322",
"AMEX60DD007644",
"AMEX60DD055382",
"AMEX60DD027986",
"AMEX60DD025304")



Tail_1_2wpa@meta.data$celltype <- factor(Tail_1_2wpa@meta.data$celltype, levels = c("Asomitic cells","Intermediate Progenitors","Chondrocytes","Fin Mesenchyme","Dermis","Myocytes"))



gg_Fig <- VlnPlot(Tail_1_2wpa, pt.size = 0, features = gene_ids, group.by = "celltype", cols = c("plum4","darkseagreen2","turquoise1","coral2","darkgoldenrod3","turquoise4","violetred2","darkgoldenrod3")) 
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_1_2wpa_sc_Harmony_Vln_PaperMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()


pdf("Tail_1_2wpa_sc_Harmony_Dot_PaperMarker.pdf",width=5,height=10)
DotPlot(Tail_1_2wpa, features = gene_ids, group.by = "celltype") + scale_size(range = c(1,8)) + RotatedAxis() + coord_flip() + theme(axis.text.x=element_text(size=7),axis.text.y=element_text(size=7)) + scale_x_discrete(labels=c(gene_names)) + scale_colour_gradientn(colors = c(rev(viridis_pal(option = "viridis")(12))))
dev.off()



```

###Run Velocity (velocyto)

```{r}

setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

Int_col_BL_CT_int = readRDS(file = "Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")


Tail_col_1wpa_spliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/Col1a2_tail_BL_10x/count_123456_114342_1wpa_col1a2_Full_Solo.out/Velocyto/raw/spliced")
Tail_col_1wpa_spliced = Tail_col_1wpa_spliced[rownames(anno), ,drop = FALSE]
Tail_col_1wpa_spliced <- CreateSeuratObject(counts = Tail_col_1wpa_spliced, project = "Tail_col_1wpa_spliced", min.cells = 1, min.features = 1)

Tail_col_1wpa_unspliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/Col1a2_tail_BL_10x/count_123456_114342_1wpa_col1a2_Full_Solo.out/Velocyto/raw/unspliced")
Tail_col_1wpa_unspliced = Tail_col_1wpa_unspliced[rownames(anno), ,drop = FALSE]
Tail_col_1wpa_unspliced <- CreateSeuratObject(counts = Tail_col_1wpa_unspliced, project = "Tail_col_1wpa_unspliced", min.cells = 1, min.features = 1)


Tail_col_2wpa_spliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/HLFYTDRXX_R9843/count_123456_2wpa_col1a2_Full_Solo.out/Velocyto/raw/spliced")
Tail_col_2wpa_spliced = Tail_col_2wpa_spliced[rownames(anno), ,drop = FALSE]
Tail_col_2wpa_spliced <- CreateSeuratObject(counts = Tail_col_2wpa_spliced, project = "Tail_col_2wpa_spliced", min.cells = 1, min.features = 1)

Tail_col_2wpa_unspliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/HLFYTDRXX_R9843/count_123456_2wpa_col1a2_Full_Solo.out/Velocyto/raw/unspliced")
Tail_col_2wpa_unspliced = Tail_col_2wpa_unspliced[rownames(anno), ,drop = FALSE]
Tail_col_2wpa_unspliced <- CreateSeuratObject(counts = Tail_col_2wpa_unspliced, project = "Tail_col_2wpa_unspliced", min.cells = 1, min.features = 1)



Int_col_BL_CT_spliced = merge(Tail_col_1wpa_spliced, y = Tail_col_2wpa_spliced, add.cell.ids = c("wpa1", "wpa2"), project = "Tail_col_Integration_onlyBL_spliced")

Int_col_BL_CT_unspliced = merge(Tail_col_1wpa_unspliced, y = Tail_col_2wpa_unspliced, add.cell.ids = c("wpa1", "wpa2"), project = "Tail_col_Integration_onlyBL_unspliced")




#Velocity

library(velocyto.R)

Int_col_BL_CT_spliced = subset(Int_col_BL_CT_spliced,cells = colnames(Int_col_BL_CT_int))
Int_col_BL_CT_unspliced = subset(Int_col_BL_CT_unspliced,cells = colnames(Int_col_BL_CT_int))


emat <- GetAssayData(Int_col_BL_CT_spliced, slot = "counts")
nmat <- GetAssayData(Int_col_BL_CT_unspliced, slot = "counts")

# take cluster labels
cluster.label <- Int_col_BL_CT_int@meta.data$seurat_cluster
names(cluster.label) = colnames(Int_col_BL_CT_int)

gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

cell.colors = gg_color_hue(length(unique(Int_col_BL_CT_int@meta.data$seurat_cluster)))
#cell.colors <-  colorRampPalette(brewer.pal(12,"Set1"))(length(unique(BLct_CSDct_filter@meta.data$orig.ident)))

names(cell.colors) <- unique(cluster.label)

cell.cols <- cell.colors[cluster.label]
names(cell.cols) = colnames(Int_col_BL_CT_int)

# take embedding
emb <- Embeddings(Int_col_BL_CT_int, "umap")

# filter genes
emat <- filter.genes.by.cluster.expression(emat,cluster.label,min.max.cluster.average = 0.5)
nmat <- filter.genes.by.cluster.expression(nmat,cluster.label,min.max.cluster.average = 0.05)

length(intersect(rownames(emat),rownames(emat)))

#Estimate RNA velocity (using gene-relative model with k=20 cell kNN pooling and using top/bottom 2% quantiles for gamma fit):

fit.quantile <- 0.3
rvel.cd <- gene.relative.velocity.estimates(emat,nmat,deltaT=1,kCells=10,fit.quantile=fit.quantile)
#cell.dist=cell.dist,

#Visualize velocity on the t-SNE embedding, using velocity vector fields:
pdf("Int_col_onlyBL_CT_int_UMAP_Velocity.pdf",width=10,height=10)
show.velocity.on.embedding.cor(emb,rvel.cd,n=200,scale='sqrt',cell.colors=cell.cols,cex=0.8,arrow.scale=2,show.grid.flow=TRUE,min.grid.cell.mass=0.1,grid.n=80,do.par=F,n.cores=50,cell.border.alpha = 0)
dev.off()

#Zhisong example
#show.velocity.on.embedding.cor(spring.coor,rvel.cd,n=200,scale='sqrt',cell.colors=cell.cols,cex=2,arrow.scale=1000,show.grid.flow=TRUE,min.grid.cell.mass=0.1,grid.n=80,do.par=T,cell.border.alpha = 0.1,n.cores=25, arrow.lwd = 2, bty="n", xaxt="n", yaxt="n")




```


###Run Velocity (scvelo)

run in bash
check possible environments
conda info --envs

conda activate /home/tobias_gerber/.conda/envs/axoloom

python

Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scvelo as scv




```{r}
setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")


Int_col_BL_CT_spliced = subset(Int_col_BL_CT_spliced,cells = colnames(Tail_1_2wpa))
Int_col_BL_CT_unspliced = subset(Int_col_BL_CT_unspliced,cells = colnames(Tail_1_2wpa))

#emat <- GetAssayData(Int_col_BL_CT_spliced, slot = "counts")
#nmat <- GetAssayData(Int_col_BL_CT_unspliced, slot = "counts")



#make anndata object for scVelo


library("reticulate")
use_condaenv("/home/tobias_gerber/.conda/envs/axoloom", required = TRUE)
scv <- import("scvelo")

ad <- import("anndata", convert = FALSE)
dfobs <- data.frame(seurat_cluster = Tail_1_2wpa@meta.data$seurat_cluster,orig.ident =  Tail_1_2wpa@meta.data$orig.ident, celltype =  Tail_1_2wpa@meta.data$celltype)
rownames(dfobs) <- colnames(Tail_1_2wpa)
#dfvar <- data.frame(genes_attributes)
#rownames(dfvar) <- genes

#bring all matrices to the same gene number

subsample = unique(intersect(intersect(intersect(rownames(Int_col_BL_CT_spliced),rownames(Tail_1_2wpa)),intersect(rownames(Int_col_BL_CT_unspliced),rownames(Tail_1_2wpa))),intersect(rownames(Int_col_BL_CT_spliced),rownames(Int_col_BL_CT_unspliced))))

subsample_mat <- subset(Tail_1_2wpa, features = subsample)
subsample_mat = as.matrix(GetAssayData(subsample_mat, slot = "counts"))

subsample_emat <- subset(Int_col_BL_CT_spliced, features = subsample)
subsample_emat = as.matrix(GetAssayData(subsample_emat, slot = "counts"))

subsample_nmat <- subset(Int_col_BL_CT_unspliced, features = subsample)
subsample_nmat = as.matrix(GetAssayData(subsample_nmat, slot = "counts"))


#subsample_mat <- subset(Tail_1_2wpa, features = subsample, subset = orig.ident == "Tail_col_1wpa")
#subsample_mat = as.matrix(GetAssayData(subsample_mat, slot = "counts"))

#subsample_emat <- subset(Int_col_BL_CT_spliced, features = subsample, subset = orig.ident == "Tail_col_1wpa_spliced")
#subsample_emat = as.matrix(GetAssayData(subsample_emat, slot = "counts"))

#subsample_nmat <- subset(Int_col_BL_CT_unspliced, features = subsample, subset = orig.ident == "Tail_col_1wpa_unspliced")
#subsample_nmat = as.matrix(GetAssayData(subsample_nmat, slot = "counts"))

#select only 2wpa cells
#ad <- import("anndata", convert = FALSE)
#dfobs <- data.frame(seurat_cluster = subset(Tail_1_2wpa, subset = orig.ident == "Tail_col_1wpa")@meta.data$seurat_cluster,orig.ident =  subset(Tail_1_2wpa, subset = orig.ident == "Tail_col_1wpa")@meta.data$orig.ident)
#rownames(dfobs) <- colnames(subset(Tail_1_2wpa, subset = orig.ident == "Tail_col_1wpa"))



#adata <- ad$AnnData(
#  X=t(subsample_mat),
#  obs=dfobs,
# # var=dfvar,
#  layers=list('spliced'=t(subsample_emat), 'unspliced'=t(subsample_nmat)),
#  obsm=list('X_umap'=Embeddings(subset(Tail_1_2wpa, subset = orig.ident == "Tail_col_1wpa"), "umap"), 'X_pca'=Embeddings(subset(Tail_1_2wpa, subset = orig.ident == "Tail_col_1wpa"), "pca")) 
 # )

adata <- ad$AnnData(
  X=t(subsample_mat),
  obs=dfobs,
 # var=dfvar,
  layers=list('spliced'=t(subsample_emat), 'unspliced'=t(subsample_nmat)),
  obsm=list('X_umap'=Embeddings(Tail_1_2wpa, "umap"), 'X_pca'=Embeddings(Tail_1_2wpa, "pca")) )

## run scvelo dynamic model
#scv$pp$filter_genes(adata) ## filter
scv$pp$filter_and_normalize(adata, min_shared_counts=as.integer(20), n_top_genes=as.integer(5000)) ##filter

scv$pp$moments(adata,n_pcs=as.integer(50),n_neighbors=as.integer(30)) ## normalize and compute moments
#scv$tl$recover_dynamics(adata,n_jobs = as.integer(30)) ## model

## takes awhile, so uncomment to save
#adata$write('data/pancreas.h5ad', compression='gzip')
#adata = scv$read('data/pancreas.h5ad')

## plot (creates pop up window)

#scv$tl$velocity(adata, mode='dynamical')
scv$tl$velocity(adata)
scv$tl$velocity_graph(adata,n_jobs = as.integer(30))
scv$pl$velocity_embedding_stream(adata,layer=c('velocity'),color=c('celltype'), basis='umap',dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_Celltype_pc50_n30_min20_top5k.svg')
scv$pl$velocity_embedding_stream(adata,layer=c('velocity'),color=c('orig.ident'), basis='umap',dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_Origin_pc50_n30_min20_top5k.svg')

scv$pl$proportions(adata,groupby='celltype',dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_Celltype_Proportions.svg')

#diffusion time

scv$pl$velocity_graph(adata,color=c('celltype'), threshold=.3, dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_pc50_n30_veloGraph.svg')

scv$tl$velocity_pseudotime(adata)
scv$pl$scatter(adata, color='velocity_pseudotime', cmap='gnuplot',dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_pc50_n30_veloPTime.svg')

#PAGA

#bug fix
adata$uns['neighbors']['distances'] = adata$obsp['distances']
adata$uns['neighbors']['connectivities'] = adata$obsp['connectivities']


scv$tl$paga(adata, groups='celltype')
df = scv$get_df(adata, 'paga/transitions_confidence', precision=as.integer(2))$T
#df$style$background_gradient(cmap='Blues')$format('{:.2g}')


scv$pl$paga(adata, basis='umap',color=c('celltype'), size=50, alpha=.1,min_edge_width=2, node_size_scale=1.5, dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_pc50_n30_min20_top5k_PAGA.svg')


#based on clusters
scv$tl$paga(adata, groups='seurat_cluster',use_time_prior = T)

df = scv$get_df(adata, 'paga/transitions_confidence', precision=as.integer(2))$T
#df$style$background_gradient(cmap='Blues')$format('{:.2g}')


scv$pl$paga(adata, basis='umap',color=c('seurat_cluster'), size=50, alpha=.1,min_edge_width=2, node_size_scale=1.5, dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_pc50_n30_min20_top5k_PAGA_clusters_prior.svg')

scv$tl$paga(adata, groups='seurat_cluster',use_time_prior = F)

df = scv$get_df(adata, 'paga/transitions_confidence', precision=as.integer(2))$T
#df$style$background_gradient(cmap='Blues')$format('{:.2g}')


scv$pl$paga(adata, basis='umap',color=c('seurat_cluster'), size=50, alpha=.1,min_edge_width=2, node_size_scale=1.5, dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_pc50_n30_min20_top5k_PAGA_clusters_Noprior.svg')


scv$pl$paga(adata, basis='umap',color=c('seurat_cluster'), size=50,layout="eq_tree",root = 10, alpha=.1,min_edge_width=2, node_size_scale=1.5, dpi = 300, save = 'Tail_1_2wpa_UMAP_scVelo_pc50_n30_min20_top5k_PAGA_clusters_Noprior_Tree.svg')










####### remove asomitic cells to see if it changes the behavior of PAGA

Tail_1_2wpa_sub = subset(Tail_1_2wpa,subset = UMAP_2 > 4, invert = T)

Tail_1_2wpa_sub = subset(Tail_1_2wpa_sub,subset = celltype %in% "Asomitic cells", invert = T)

ad <- import("anndata", convert = FALSE)
dfobs <- data.frame(seurat_cluster = Tail_1_2wpa_sub@meta.data$seurat_cluster,orig.ident =  Tail_1_2wpa_sub@meta.data$orig.ident, celltype =  Tail_1_2wpa_sub@meta.data$celltype)
rownames(dfobs) <- colnames(Tail_1_2wpa_sub)
#dfvar <- data.frame(genes_attributes)
#rownames(dfvar) <- genes

#bring all matrices to the same gene number

subsample = unique(intersect(intersect(intersect(rownames(Int_col_BL_CT_spliced),rownames(Tail_1_2wpa_sub)),intersect(rownames(Int_col_BL_CT_unspliced),rownames(Tail_1_2wpa_sub))),intersect(rownames(Int_col_BL_CT_spliced),rownames(Int_col_BL_CT_unspliced))))

subsample_mat <- subset(Tail_1_2wpa_sub, features = subsample)
subsample_mat = as.matrix(GetAssayData(subsample_mat, slot = "counts"))

subsample_emat <- subset(Int_col_BL_CT_spliced, features = subsample, cells = colnames(Tail_1_2wpa_sub))
subsample_emat = as.matrix(GetAssayData(subsample_emat, slot = "counts"))

subsample_nmat <- subset(Int_col_BL_CT_unspliced, features = subsample, cells = colnames(Tail_1_2wpa_sub))
subsample_nmat = as.matrix(GetAssayData(subsample_nmat, slot = "counts"))

pdf("test.umap.pdf",width=10,height=10)
DimPlot(object = Tail_1_2wpa_sub, raster = F, reduction = 'umap', pt.size = 1, group.by = "celltype")
DimPlot(object = Tail_1_2wpa, raster = F, reduction = 'umap', pt.size = 1, group.by = "celltype")
dev.off()


adata <- ad$AnnData(
  X=t(subsample_mat),
  obs=dfobs,
 # var=dfvar,
  layers=list('spliced'=t(subsample_emat), 'unspliced'=t(subsample_nmat)),
  obsm=list('X_umap'=Embeddings(Tail_1_2wpa_sub, "umap"), 'X_pca'=Embeddings(Tail_1_2wpa_sub, "pca")) )

## run scvelo dynamic model
#scv$pp$filter_genes(adata) ## filter
scv$pp$filter_and_normalize(adata, min_shared_counts=as.integer(20), n_top_genes=as.integer(5000)) ##filter

scv$pp$moments(adata,n_pcs=as.integer(50),n_neighbors=as.integer(30)) ## normalize and compute moments
#scv$tl$recover_dynamics(adata,n_jobs = as.integer(30)) ## model

## takes awhile, so uncomment to save
#adata$write('data/pancreas.h5ad', compression='gzip')
#adata = scv$read('data/pancreas.h5ad')

## plot (creates pop up window)

#scv$tl$velocity(adata, mode='dynamical')
scv$tl$velocity(adata)
scv$tl$velocity_graph(adata,n_jobs = as.integer(30))
scv$pl$velocity_embedding_stream(adata,layer=c('velocity'),color=c('celltype'), basis='umap',dpi = 300, save = 'Tail_1_2wpa_noAso_UMAP_scVelo_Celltype_pc50_n30_min20_top5k.svg')
scv$pl$velocity_embedding_stream(adata,layer=c('velocity'),color=c('orig.ident'), basis='umap',dpi = 300, save = 'Tail_1_2wpa_noAso_UMAP_scVelo_Origin_pc50_n30_min20_top5k.svg')


#diffusion time

#scv$pl$velocity_graph(adata,color=c('celltype'), threshold=.3, dpi = 300, save = 'Tail_1_2wpa_noAso_UMAP_scVelo_pc50_n30_veloGraph.svg')

#scv$tl$velocity_pseudotime(adata)
#scv$pl$scatter(adata, color='velocity_pseudotime', cmap='gnuplot',dpi = 300, save = 'Tail_1_2wpa_noAsoUMAP_scVelo_pc50_n30_veloPTime.svg')

#PAGA

#bug fix
#adata$uns['neighbors']['distances'] = adata$obsp['distances']
#adata$uns['neighbors']['connectivities'] = adata$obsp['connectivities']

#based on cell types
scv$tl$paga(adata, groups='celltype',use_time_prior = F)

df = scv$get_df(adata, 'paga/transitions_confidence', precision=as.integer(2))$T
#df$style$background_gradient(cmap='Blues')$format('{:.2g}')


scv$pl$paga(adata, basis='umap',color=c('celltype'), size=50, alpha=.1,min_edge_width=2, node_size_scale=1.5, dpi = 300, save = 'Tail_1_2wpa_noAso_UMAP_scVelo_pc50_n30_min20_top5k_PAGA_celltypes.svg')

#based on clusters
scv$tl$paga(adata, groups='seurat_cluster',use_time_prior = T)

df = scv$get_df(adata, 'paga/transitions_confidence', precision=as.integer(2))$T
#df$style$background_gradient(cmap='Blues')$format('{:.2g}')


scv$pl$paga(adata, basis='umap',color=c('seurat_cluster'), size=50, alpha=.1,min_edge_width=2, node_size_scale=1.5, dpi = 300, save = 'Tail_1_2wpa_noAso_UMAP_scVelo_pc50_n30_min20_top5k_PAGA_clusters.svg')


```

###Pseudotime
```{r}

#velocity pseudotime gives no reasonable results --> calculate manually with diffusion time


######Calc pseudo time 

harmony.embedding = as.data.frame(Embeddings(Tail_1_2wpa, "harmony"),stringsAsFactors = F)

meta.data = FetchData(object = Tail_1_2wpa, vars = c("celltype",'ident', 'orig.ident','UMAP_1','UMAP_2'))

cells_AP = rownames(meta.data)[meta.data$celltype %in% c("Asomitic cells")]
cells_P = rownames(meta.data)[meta.data$celltype %in% c("Intermediate Progenitors")]
cells_M = rownames(meta.data)[meta.data$celltype %in% c("Myocytes")]
cells_C = rownames(meta.data)[meta.data$celltype %in% c("Chondrocytes")]
cells_D = rownames(meta.data)[meta.data$celltype %in% c("Dermis")]
cells_MES = rownames(meta.data)[meta.data$celltype %in% c("Fin Mesenchyme")]


#calculate pseudo time separate on P and differentiated cells

harmony.embedding.AP = as.data.frame(harmony.embedding[c(cells_AP),])
harmony.embedding.M = as.data.frame(harmony.embedding[c(cells_M),])
harmony.embedding.P = as.data.frame(harmony.embedding[cells_P,])
harmony.embedding.D = as.data.frame(harmony.embedding[cells_D,])
harmony.embedding.MES = as.data.frame(harmony.embedding[cells_MES,])
harmony.embedding.C = as.data.frame(harmony.embedding[cells_C,])


library("destiny")

#calculate pseudotime for progenitors
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.P),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.P = dpt[random_root(dpt), ]

harmony.embedding.P$pt = as.numeric(pseudotime.P)
harmony.embedding.P = harmony.embedding.P[order(harmony.embedding.P$pt),]
harmony.embedding.P$rank = (1:nrow(harmony.embedding.P))
harmony.embedding.P$rank.norm = harmony.embedding.P$rank / max(harmony.embedding.P$rank)

#calculate pseudotime for muscles
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.M),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.M = dpt[random_root(dpt), ]

harmony.embedding.M$pt = as.numeric(pseudotime.M)
harmony.embedding.M = harmony.embedding.M[order(harmony.embedding.M$pt),]
harmony.embedding.M$rank = rev(1:nrow(harmony.embedding.M)) #+ max(harmony.embedding.P$rank)
harmony.embedding.M$rank.norm = harmony.embedding.M$rank / max(harmony.embedding.M$rank) 
harmony.embedding.M$rank.norm= harmony.embedding.M$rank.norm + 1

#calculate pseudotime for dermis
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.D),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.D = dpt[random_root(dpt), ]

harmony.embedding.D$pt = as.numeric(pseudotime.D)
harmony.embedding.D = harmony.embedding.D[order(harmony.embedding.D$pt),]
harmony.embedding.D$rank = rev(1:nrow(harmony.embedding.D))  
#+ max(harmony.embedding.P$rank)
harmony.embedding.D$rank.norm = harmony.embedding.D$rank / max(harmony.embedding.D$rank) 
harmony.embedding.D$rank.norm = harmony.embedding.D$rank.norm + 1

#calculate pseudotime for cartilage
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.C),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.C = dpt[random_root(dpt), ]

harmony.embedding.C$pt = as.numeric(pseudotime.C)
harmony.embedding.C = harmony.embedding.C[order(harmony.embedding.C$pt),]
harmony.embedding.C$rank = (1:nrow(harmony.embedding.C)) 
#+ max(harmony.embedding.P$rank)
harmony.embedding.C$rank.norm = harmony.embedding.C$rank / max(harmony.embedding.C$rank) 
harmony.embedding.C$rank.norm = harmony.embedding.C$rank.norm + 1

#calculate pseudotime for fin mesenchyme
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.MES),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.MES = dpt[random_root(dpt), ]

harmony.embedding.MES$pt = as.numeric(pseudotime.MES)
harmony.embedding.MES = harmony.embedding.MES[order(harmony.embedding.MES$pt),]
harmony.embedding.MES$rank = rev(1:nrow(harmony.embedding.MES)) #+ max(harmony.embedding.P$rank)
harmony.embedding.MES$rank.norm = harmony.embedding.MES$rank / max(harmony.embedding.MES$rank) 
harmony.embedding.MES$rank.norm = harmony.embedding.MES$rank.norm + 1


#Asomitic cell are set to 0

harmony.embedding.AP$pt = 0
harmony.embedding.AP$rank = 0
harmony.embedding.AP$rank.norm = 0

#merge
pseudotime = rbind(harmony.embedding.AP,harmony.embedding.P,harmony.embedding.D,harmony.embedding.C,harmony.embedding.MES,harmony.embedding.M,stringsAsFactors = F)

plot.pt = cbind(meta.data,rank.norm = pseudotime[match(rownames(meta.data),rownames(pseudotime)),"rank.norm"], stringsAsFactors = F)

Tail_1_2wpa@meta.data$PT.norm = plot.pt$rank.norm

pdf("Tail_1_2wpa_sc_Harmony_PT.pdf",width=8,height=7)
FeaturePlot(Tail_1_2wpa, pt.size = 1, features = "PT.norm",order = F,cols = c(viridis_pal(option = "plasma")(100)))
dev.off()

saveRDS(Tail_1_2wpa, file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")


#PLot genes as line plot separated by lineage

tmp=c("AMEX60DD027495",
"AMEX60DD010148",
"AMEX60DD052476",
"AMEX60DD050893",
"AMEX60DD020388",
"AMEX60DD052322",
"AMEX60DD007644",
"AMEX60DD024484",
"AMEX60DD055040",
"AMEX60DD034577",
"AMEX60DD031139",
"AMEX60DD017092",
"AMEX60DD008545",
"AMEX60DD024851",
"AMEX60DD056342",
"AMEX60DD055382",
"AMEX60DD027986",
"AMEX60DD055570",
"AMEX60DD025304")

plot.data = FetchData(object = Tail_1_2wpa, vars = c('ident', 'orig.ident',"celltype","PT.norm",tmp))
plot.data = melt(plot.data,id.vars = c('ident', 'orig.ident',"celltype","PT.norm"))


pdf("Tail_1_2wpa_sc_Harmony_PT_LinePlot.pdf",width=8,height=7)
ggplot(plot.data, aes(x = PT.norm, y = value ,color = celltype)) + geom_smooth(se = F)  + facet_wrap(. ~ variable, scales = "free")
dev.off()

tmp.name=c("UBE2C",
"TOP2A",
"CDK1",
"ENTPD2",
"NTN3-NTN1",
"PAX7",
"MYF5",
"AXL",
"NRP2",
"SMOC2",
"CACNG1",
"CDH15",
"MYOG",
"CKM",
"DES",
"MYL1",
"MYLPF",
"TTN",
"ACTC1")
names(tmp.name) = tmp

tmp = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")

plot.data = FetchData(object = Tail_1_2wpa, vars = c('ident', 'orig.ident',"celltype","PT.norm",tmp))
plot.data = melt(plot.data,id.vars = c('ident', 'orig.ident',"celltype","PT.norm"))


pdf("Tail_1_2wpa_sc_Harmony_PT_LinePlot_GeneralMarker.pdf",width=20,height=17)
ggplot(plot.data, aes(x = PT.norm, y = value ,color = celltype)) + geom_smooth(se = F)  + facet_wrap(. ~ variable, scales = "free")
dev.off()

```

##Extract Pax7 lineage from blastema cells

```{r}
Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")

#subset asomitic cells and muscle lineage cells

Tail_1_2wpa_pax = subset(Tail_1_2wpa,idents = c(0,6,10,9))


#Cluster cells
Tail_1_2wpa_pax<- FindNeighbors(Tail_1_2wpa_pax, dims = 1:50, reduction = "harmony")
Tail_1_2wpa_pax <- FindClusters(Tail_1_2wpa_pax, resolution = 1)


#run UMAP
Tail_1_2wpa_pax = RunUMAP(Tail_1_2wpa_pax,dims = 1:50, reduction = "harmony",n.components = 3L, min.dist = 0.1)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Tail_1_2wpa_pax_Harmony_UMAP_PC50_res1.pdf",width=20,height=20)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax, dims = c(1, 3), shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax, dims = c(2, 3), shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3, cols = c("deepskyblue1","dodgerblue","firebrick3","firebrick4"))
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', group.by = "celltype", pt.size = 3, cols = c("turquoise1","turquoise3","plum4","turquoise4","indianred","coral2","violetred2","darkgoldenrod3"))
FeaturePlot(Tail_1_2wpa_pax,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()

#look for markers to clean out progenitors that are on the way to other lineages


gene_ids = c("AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD023726","AMEX60DD054625","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD022355","AMEX60DD009984","AMEX60DD009989","AMEX60DD009968")
	
gene_names =c("PRRX1","RSPO3","FGF8","WNT5A","TBX2","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","DLX5","SOST","ETV4","PHOSPHO1")

gg_Fig <- FeaturePlot(Tail_1_2wpa_pax, pt.size = 1, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_1_2wpa_pax_Harmony_feature_SpecificMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()




gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	

gg_Fig <- FeaturePlot(Tail_1_2wpa_pax, pt.size = 1, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_1_2wpa_pax_Harmony_feature_GeneralMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()

#remove mesenchyme and cartilage lineage cells and intermediate progenitors

Tail_1_2wpa_pax_filter = subset(Tail_1_2wpa_pax,idents = c(3,4,5,10))


#Cluster cells
Tail_1_2wpa_pax_filter<- FindNeighbors(Tail_1_2wpa_pax_filter, dims = 1:50, reduction = "harmony")
Tail_1_2wpa_pax_filter <- FindClusters(Tail_1_2wpa_pax_filter, resolution = 1)


#run UMAP
Tail_1_2wpa_pax_filter = RunUMAP(Tail_1_2wpa_pax_filter,dims = 1:50, reduction = "harmony", min.dist = 0.1)



pdf("Tail_1_2wpa_pax_Harmony_UMAP_PC50_res1.pdf",width=20,height=20)
DimPlot(object = Tail_1_2wpa_pax_filter,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax_filter,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3, cols = c("deepskyblue1","dodgerblue","firebrick3","firebrick4"))
DimPlot(object = Tail_1_2wpa_pax_filter,  shuffle= T,reduction = 'umap', group.by = "celltype", pt.size = 3, cols = c("turquoise1","turquoise3","plum4","turquoise4","indianred","coral2","violetred2","darkgoldenrod3"))
FeaturePlot(Tail_1_2wpa_pax_filter,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


```

###Hallmark genes

```{r}
#intersect with var genes


Tail_1_2wpa_pax_filter <- FindVariableFeatures(Tail_1_2wpa_pax_filter, nfeatures = 10000)

EMT.ha = read.delim("HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION.v2023.1.Hs.txt",sep = "\t",header = F)[,1]

anno.emt = vector()

for (i in 1:length(EMT.ha)) {
  anno.emt = rbind(anno.emt,anno[grep(pattern = paste(EMT.ha[i],"$",sep = ""),anno$V2),])
}

anno.emt = intersect(anno.emt$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.emt),
  name = 'EMT.score',
  ctrl = 10
)


Motility.GObp = read.delim("GObp_CellMotility.txt",sep = "\t",header = F)[,1]

anno.mot = vector()

for (i in 1:length(Motility.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.mot),
  name = 'Mot.score',
  ctrl = 10
)


AmeboidMig.GObp = read.delim("GObp_AmeboidalMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(AmeboidMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(AmeboidMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.mig),
  name = 'AmeboidMig.score',
  ctrl = 10
)



SmoothMuscleMig.GObp = read.delim("GObp_SmoothMuscleMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(SmoothMuscleMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(SmoothMuscleMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.mig),
  name = 'SmoothMuscleMig.score',
  ctrl = 10
)


PosFibMig.GObp = read.delim("GObp_PosFibMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(PosFibMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(PosFibMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.mig),
  name = 'PosFibMig.score',
  ctrl = 10
)


Motility.Actin.GObp = c("ACTR3","ARPC","ENAH","EPS8","JMY","NCKAP1L")

anno.mot = vector()

for (i in 1:length(Motility.Actin.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.Actin.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.mot),
  name = 'Mot.Act.score',
  ctrl = 10
)


Motility.Meso.GObp = c("APELA","EPB41L5","FGF8","LRP5","MESP1")

anno.mot = vector()

for (i in 1:length(Motility.Meso.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.Meso.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.mot),
  name = 'Mot.Meso.score',
  ctrl = 10
)


Wu.mig = read.delim("Wu_CellMigration.txt",sep = "\t",header = F)[,1]

anno.wu = vector()

for (i in 1:length(Wu.mig)) {
  anno.wu = rbind(anno.wu,anno[grep(pattern = paste(Wu.mig[i],"$",sep = ""),anno$V2),])
}

anno.wu = intersect(anno.wu$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.wu),
  name = 'Wu.mig.score',
  ctrl = 10
)




GO.ERK = read.delim("GObp_ERK.txt",sep = "\t",header = F)[,1]

anno.ERK = vector()

for (i in 1:length(GO.ERK)) {
  anno.ERK = rbind(anno.ERK,anno[grep(pattern = paste(GO.ERK[i],"$",sep = ""),anno$V2),])
}

anno.ERK = intersect(anno.ERK$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.ERK),
  name = 'GObp.ERK.score',
  ctrl = 10
)

BioCart.ERK = read.delim("BioCart_ERK.txt",sep = "\t",header = F)[,1]

anno.ERK = vector()

for (i in 1:length(BioCart.ERK)) {
  anno.ERK = rbind(anno.ERK,anno[grep(pattern = paste(BioCart.ERK[i],"$",sep = ""),anno$V2),])
}

anno.ERK = intersect(anno.ERK$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.ERK),
  name = 'BioCart.ERK.score',
  ctrl = 10
)

#cell proliferation

g2m.genes = cc.genes$g2m.genes
s.genes = cc.genes$s.genes

anno.g2m = vector()

for (i in 1:length(g2m.genes)) {
  anno.g2m = rbind(anno.g2m,anno[grep(pattern = paste(g2m.genes[i],"$",sep = ""),anno$V2),])
}

anno.g2m = intersect(anno.g2m$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

anno.s = vector()
for (i in 1:length(s.genes)) {
  anno.s = rbind(anno.s,anno[grep(pattern = paste(s.genes[i],"$",sep = ""),anno$V2),])
}

anno.s = intersect(anno.s$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- CellCycleScoring(Tail_1_2wpa_pax_filter, s.features = anno.s, g2m.features = anno.g2m, set.ident = F)



GO.PosFibProl = read.delim("GObp_PosCellProl.txt",sep = "\t",header = F)[,1]

anno.prol = vector()

for (i in 1:length(GO.PosFibProl)) {
  anno.prol = rbind(anno.prol,anno[grep(pattern = paste(GO.PosFibProl[i],"$",sep = ""),anno$V2),])
}

anno.prol = intersect(anno.prol$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.prol),
  name = 'GObp.PosFibProl.score',
  ctrl = 10
)



GO.PosCellPopProl = read.delim("GObp_PosCellPopProl.txt",sep = "\t",header = F)[,1]

anno.prol = vector()

for (i in 1:length(GO.PosCellPopProl)) {
  anno.prol = rbind(anno.prol,anno[grep(pattern = paste(GO.PosCellPopProl[i],"$",sep = ""),anno$V2),])
}

anno.prol = intersect(anno.prol$V1,VariableFeatures(Tail_1_2wpa_pax_filter))

Tail_1_2wpa_pax_filter <- AddModuleScore(
  object = Tail_1_2wpa_pax_filter,
  features = list(anno.prol),
  name = 'GObp.PosCellPopProl.score',
  ctrl = 10
)

pdf("Tail_1_2wpa_pax_Harmony_UMAP_PC50_res1.pdf",width=20,height=20)
DimPlot(object = Tail_1_2wpa_pax_filter,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax_filter,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3, cols = c("deepskyblue1","dodgerblue","firebrick3","firebrick4"))
DimPlot(object = Tail_1_2wpa_pax_filter,  shuffle= T,reduction = 'umap', group.by = "celltype", pt.size = 3, cols = c("turquoise1","turquoise3","plum4","turquoise4","indianred","coral2","violetred2","darkgoldenrod3"))
FeaturePlot(Tail_1_2wpa_pax_filter,raster = F,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
FeaturePlot(Tail_1_2wpa_pax_filter,raster = F,  pt.size = 2, features = c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1"), order = T, cols = c("grey",rev(viridis_pal(option = "viridis")(12)))) & NoLegend()
dev.off()



```


###SPRING

```{r}

###Prepare data

harmony.embedding = Embeddings(Tail_1_2wpa_pax_filter, "harmony")


#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}



#extract norm data from Seurat object

norm.data = GetAssayData(object = Tail_1_2wpa_pax_filter, slot = "data")
norm.data = as.matrix(norm.data)


meta.data = FetchData(object = Tail_1_2wpa_pax_filter, vars = c('ident', 'orig.ident','celltype',"nFeature_RNA","nCount_RNA"))

##############################
#SPRING
##############################

### export for SPRING (export norma and not scaled data)

#cutoff 0.8
y=as.data.frame(t(meta.data[,c("orig.ident","ident","celltype")]),stringsAsFactors=F)

write.table(y,sep = ",",col.names = F,file = "SPRING_Export_Tail_1_2wpa_pax_meta.csv")

x = as.data.frame(t(harmony.embedding[,c(1:50)]))

write.table(x,sep = ",",col.names = F,file ="SPRING_Export_Tail_1_2wpa_pax_HarmonyMatrix.csv")

z=rownames(x)

write.table(z,col.names = F,row.names = F,sep = ",",file = "SPRING_Export_Tail_1_2wpa_pax_GeneList.csv")

###Run with : 10 PCs gene / 5 neighbors 

#import SPRING tables

coords = read.csv("SPRING_Import_Tail_1_2wpa_pax_coordinates.txt",header = F)
colnames(coords) = c("Cell","x","y")
edges = read.csv("SPRING_Import_Tail_1_2wpa_pax_edge_list.txt",header = F)
colnames(edges) = c("start","end")

rownames(coords) = rownames(harmony.embedding)



### set some parameters
# cells
colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 2
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

coords$celltype=Tail_1_2wpa_pax_filter@meta.data$celltype

coords$color[coords$celltype=="PSMlike_bl"]= "darkgoldenrod3"
coords$color[coords$celltype=="Progenitors_bl"]="turquoise4"
coords$color[coords$celltype=="Myocytes_bl"]="violetred2"

plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]

pdf("./Tail_1_2wpa_pax_SPRING_celltype.pdf",width=15,height=8)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()


########## plot many genes at once

#set genes to plot - cluster markers



tmp = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
names(tmp) =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	

data.plot=as.data.frame(t(norm.data))
data.plot = cbind(data.plot,FetchData(Tail_1_2wpa_pax_filter,c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")))

# create 200 colors (colstart --> colend)
colramp = colorRampPalette(c("grey",rev(viridis_pal(option = "viridis")(12))))
colgrad = colramp(101)

data.plot = data.plot[match(rownames(data.plot),rownames(coords)),]
data.plot = cbind(coords,data.plot)
##########

pdf("./Tail_1_2wpa_pax_SPRING_Features.pdf",width=15,height=8 )

for (i in colnames(data.plot[,tmp])) {

  # normalize expression [0-1]
  exp = data.plot[,i]/max(data.plot[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  data.plot[,i] = round(exp, digits=2) *100 +1
  data.plot = data.plot[order(data.plot[,i]),]
  
  #plot gradient

    plot(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="", main = names(tmp)[tmp %in% i])

    edge_start_coords = data.plot[match(edges$start, data.plot$Cell), c("x","y")]
    edge_end_coords = data.plot[match(edges$end, data.plot$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="")

}

dev.off()


tmp = c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")
names(tmp) =c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")
	

data.plot=as.data.frame(t(norm.data))
data.plot = cbind(data.plot,FetchData(Tail_1_2wpa_pax_filter,c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")))

# create 200 colors (colstart --> colend)
colramp = colorRampPalette(c((viridis_pal(option = "magma")(12))))
colgrad = colramp(101)

data.plot = data.plot[match(rownames(data.plot),rownames(coords)),]
data.plot = cbind(coords,data.plot)
##########

pdf("./Tail_1_2wpa_pax_SPRING_Scores.pdf",width=15,height=8 )

for (i in colnames(data.plot[,tmp])) {

  # normalize expression [0-1]
  exp = data.plot[,i]/max(data.plot[,i])
  exp = exp+abs(min(exp))
  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  data.plot[,i] = round(exp, digits=2) *100 +1
  data.plot = data.plot[order(data.plot[,i]),]
  
  #plot gradient

    plot(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="", main = names(tmp)[tmp %in% i])

    edge_start_coords = data.plot[match(edges$start, data.plot$Cell), c("x","y")]
    edge_end_coords = data.plot[match(edges$end, data.plot$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="")

}

dev.off()

```

###Pseudotime

```{r}

######Calc pseudo time 

harmony.embedding = as.data.frame(Embeddings(Tail_all_int_pax, "harmony"),stringsAsFactors = F)

meta.data = FetchData(object = Tail_all_int_pax, vars = c("celltype",'ident', 'orig.ident','UMAP_1','UMAP_2'))

cells_SP = rownames(meta.data)[meta.data$celltype %in% c("Somitic Progenitors")]
cells_P = rownames(meta.data)[meta.data$celltype %in% c("Committed Progenitors")]
cells_M = rownames(meta.data)[meta.data$celltype %in% c("Myogenic lineage")]


#calculate pseudo time separate on SM and P + M

harmony.embedding.muscle = as.data.frame(harmony.embedding[c(cells_P,cells_M),])
harmony.embedding.SP = as.data.frame(harmony.embedding[cells_SP,])

library("destiny")

#Pcalculate pseudotime for muscle branch
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.muscle),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.muscle = dpt[random_root(dpt), ]

harmony.embedding.muscle$pt = as.numeric(pseudotime.muscle)
harmony.embedding.muscle = harmony.embedding.muscle[order(harmony.embedding.muscle$pt),]
harmony.embedding.muscle$rank = rev(1:nrow(harmony.embedding.muscle))

#SM branch

harmony.embedding.SP$pt = 0
harmony.embedding.SP$rank = 0

#merge
pseudotime = rbind(harmony.embedding.SP,harmony.embedding.muscle,stringsAsFactors = F)

plot.pt = cbind(coords,rank = pseudotime[match(rownames(coords),rownames(pseudotime)),"rank"], stringsAsFactors = F)


# cells
cellcex = 1
#color
colramp = colorRampPalette(c("darkgoldenrod3","violetred2"))
colgrad = colramp(101)


pdf("./Tail_0_1_2wpa_sc_pax_SPRING_PT_Rank.pdf",width=8,height=8)

  # normalize expression [0-1]
  exp = plot.pt[,"rank"]/max(plot.pt[,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt[,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$x, plot.pt$y, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime")

    edge_start_plot.pt  = plot.pt[match(edges$start, plot.pt$Cell), c("x","y")]
    edge_end_plot.pt  = plot.pt[match(edges$end, plot.pt$Cell), c("x","y")]
    segments(x0=edge_start_plot.pt[,"x"], y0=edge_start_plot.pt[,"y"],
    	x=edge_end_plot.pt[,"x"], y=edge_end_plot.pt[,"y"], lwd=linewidth, col=linecol)
		
    points(plot.pt$x, plot.pt$y, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="")

dev.off()


#Plot genes on top of SPRING embedding


tmp=c("AMEX60DD027495",
"AMEX60DD010148",
"AMEX60DD052476",
"AMEX60DD050893",
"AMEX60DD020388",
"AMEX60DD052322",
"AMEX60DD007644",
"AMEX60DD024484",
"AMEX60DD055040",
"AMEX60DD034577",
"AMEX60DD031139",
"AMEX60DD017092",
"AMEX60DD008545",
"AMEX60DD024851",
"AMEX60DD056342",
"AMEX60DD055382",
"AMEX60DD027986",
"AMEX60DD055570",
"AMEX60DD025304")

meta.data = FetchData(object = Tail_all_int_pax, vars = c('ident', 'orig.ident',"celltype",tmp))


tmp.name=c("UBE2C",
"TOP2A",
"CDK1",
"ENTPD2",
"NTN3-NTN1",
"PAX7",
"MYF5",
"AXL",
"NRP2",
"SMOC2",
"CACNG1",
"CDH15",
"MYOG",
"CKM",
"DES",
"MYL1",
"MYLPF",
"TTN",
"ACTC1")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))


colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 1
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

##########

pdf("./Tail_0_1_2wpa_sc_pax_SPRING_MyogenicLineageMarker.pdf",width=8,height=8)

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i])

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

}

dev.off()

#ntn3 seems to be earlier than pax7
#Find correlating genes

norm.data = GetAssayData(object = Tail_all_int_pax, slot = "data")
norm.data = as.matrix(norm.data)

gene<-as.numeric(norm.data["AMEX60DD020388",])
correlations<-apply(norm.data,1,function(x){cor(gene,x)})
correlations[is.na(correlations)] = 0

correlations.anno = as.data.frame(correlations)
correlations.anno$gene = rownames(correlations.anno)

correlations.anno = merge(correlations.anno,anno, by.x="gene" , by.y="V1")
correlations.anno$ID = correlations.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
correlations.anno = correlations.anno  %>% arrange(desc(correlations))

write.csv(correlations.anno,"Tail_0_1_2wpa_sc_pax_Harmony_CorrelationMarker.csv")

#Plot genes on top of SPRING embedding


tmp=c("AMEX60DD020388",
"AMEX60DD055040",
"AMEX60DD009987",
"AMEX60DD043119",
"AMEX60DD001909",
"AMEX60DD018225",
"AMEX60DD023964",
"AMEX60DD000336",
"AMEX60DD013919",
"AMEX60DD025227",
"AMEX60DD055954",
"AMEX60DD045083",
"AMEX60DD003483",
"AMEX60DD002950",
"AMEX60DD040573",
"AMEX60DD008914",
"AMEX60DD041795",
"AMEX60DD016542")

meta.data = FetchData(object = Tail_all_int_pax, vars = c('ident', 'orig.ident',"celltype",tmp))


tmp.name=c("NTN3-NTN1",
"NRP2",
"MEOX1*tf",
"DMRT2*tf",
"IGSF10",
"TNN",
"ASPN",
"MMP11",
"CILP2",
"EFEMP2",
"CXCR4",
"SFRP2*wnt",
"NR2F2*tf",
"SFRP1*wnt",
"KHDRBS3",
"WNT2B*wnt",
"TCF4*tf",
"NTN1")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))


colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 1
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

##########

pdf("./Tail_0_1_2wpa_sc_pax_SPRING_EarlyMyoCorrelationMarker.pdf",width=8,height=8)

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i])

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

}

dev.off()



```

##Metacell 1wpa 2wpa

###Run Seurat
```{r}

setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

Int_col_BL_CT_int = readRDS(file = "Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")

#keep UMAP but cluster crazily high

#Cluster cells
Int_col_BL_CT_meta<- FindNeighbors(Int_col_BL_CT_int, dims = 1:100, reduction = "harmony")
Int_col_BL_CT_meta <- FindClusters(Int_col_BL_CT_meta, resolution = 300)

#plot on UMAP

pdf("Int_col_onlyBL_CT_HarmonyTimeInt_UMAP_PC50_resMeta.pdf",width=120,height=10)
DimPlot(object = Int_col_BL_CT_meta, raster = T, reduction = 'umap', pt.size = 1)
dev.off()

saveRDS(Int_col_BL_CT_meta,"Int_col_onlyBL_CT_HarmonyTimeInt_MetaRes_SeuratObj.RDS")

tmp = as.data.frame(melt(table(FetchData(Int_col_BL_CT_meta,c("seurat_clusters","orig.ident")))))
test = cbind(tmp[1:2959,],tmp[2960:5918,])
colnames(test) = c("seurat_clusters","1wpa","value_1wpa","ignore","2wpa","value_2wpa")
test$ignore = NULL
test$fraction_1wpa = test$value_1wpa / (test$value_1wpa+test$value_2wpa)

#merge clusters to meta cells

Int_col_BL_CT_meta_mean = AverageExpression(
  Int_col_BL_CT_meta,
  return.seurat = T,
  group.by = "ident",
  slot = "counts")

Int_col_BL_CT_meta_mean@meta.data$fraction_1wpa = test$fraction_1wpa

#Normalize
Int_col_BL_CT_meta_mean = NormalizeData(Int_col_BL_CT_meta_mean)

mito.genes <- c("ND2","ND1","ND3","ND4","ND4L","ND5","ND6")
mito.contig <- intersect(c(rownames(anno)[anno$V2 %in% mito.genes ] , rownames(anno)[anno$V2 %in% anno$V2[grep("^COX",anno$V2)]] ) , rownames(Tail_col_0wpa_v3))

Int_col_BL_CT_meta_mean[["percent.mt"]] <- PercentageFeatureSet(Int_col_BL_CT_meta_mean, features = mito.contig)

#get cell cycle genes

g2m.genes <- cc.genes$g2m.genes
g2m.contig <- intersect(rownames(anno)[anno$V2 %in% g2m.genes ] , rownames(Int_col_BL_CT_meta_mean) )

s.genes <- cc.genes$s.genes
s.contig <- intersect(rownames(anno)[anno$V2 %in% s.genes ] , rownames(Int_col_BL_CT_meta_mean) )

#cell cycle scoring
Int_col_BL_CT_meta_mean <- CellCycleScoring(Int_col_BL_CT_meta_mean, s.features = s.contig, g2m.features = g2m.contig, set.ident = TRUE)


#scale
#ScaleData(  object,  features = NULL,  assay = NULL,  vars.to.regress = NULL,  split.by = NULL,  model.use = "linear",  use.umi = FALSE,  do.scale = TRUE,  do.center = TRUE,  scale.max = 10,  block.size = 1000,  min.cells.to.block = 3000,  verbose = TRUE)

all.genes <- rownames(Int_col_BL_CT_meta_mean)

Int_col_BL_CT_meta_mean = ScaleData(  Int_col_BL_CT_meta_mean, features = all.genes ,vars.to.regress = c("nCount_RNA","percent.mt"))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

Int_col_BL_CT_meta_mean = RunPCA(Int_col_BL_CT_meta_mean,features = all.genes,npcs = 100)

#plot PCA heatmaps
#DimHeatmap(pbmc, dims = 1, cells = 500, balanced = TRUE)

library(RColorBrewer)
library(ggplot2)

pdf("Int_col_onlyBL_CT_Meta_PCAheatmaps.pdf",width=30,height=20)
DimHeatmap(Int_col_BL_CT_meta_mean, dims = 1:16, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT_meta_mean, dims = 17:32, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT_meta_mean, dims = 33:48, cells = 1000, balanced = TRUE,  ncol = 4, fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
dev.off()

#elbow plot
pdf("Int_col_onlyBL_CT_Meta_PCelbow.pdf",width=20,height=10)
ElbowPlot(Int_col_BL_CT_meta_mean, ndims = 100)
dev.off()

#Cluster cells
Int_col_BL_CT_meta_mean<- FindNeighbors(Int_col_BL_CT_meta_mean, dims = 1:100)
Int_col_BL_CT_meta_mean <- FindClusters(Int_col_BL_CT_meta_mean, resolution = 1)

#run UMAP
Int_col_BL_CT_meta_mean = RunUMAP(Int_col_BL_CT_meta_mean,dims = 1:100, min.dist = 0.08)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_onlyBL_CT_Meta_UMAP_PC100_res1.pdf",width=15,height=15)
DimPlot(object = Int_col_BL_CT_meta_mean, reduction = 'umap', pt.size = 3)
DimPlot(object = Int_col_BL_CT_meta_mean, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
DimPlot(object = Int_col_BL_CT_meta_mean, reduction = 'umap',group.by = "fraction_1wpa", pt.size = 3)
FeaturePlot(Int_col_BL_CT_meta_mean, pt.size = 1, features = c("nFeature_RNA","nCount_RNA","percent.mt","G2M.Score"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


gene_ids = c("AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD023726","AMEX60DD054625","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD022355","AMEX60DD009984","AMEX60DD009989","AMEX60DD009968","AMEX60DD044091")
	
gene_names =c("PRRX1","RSPO3","FGF8","WNT5A","TBX2","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","DLX5","SOST","ETV4","PHOSPHO1","Blastema5dpaSpecific")

gg_Fig <- FeaturePlot(Int_col_BL_CT_meta_mean, pt.size = 2, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend()})

pdf("Int_col_onlyBL_CT_Meta_feature_SpecificMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()



markers <- FindAllMarkers(Int_col_BL_CT_meta_mean, only.pos = TRUE,  logfc.threshold = 0.3)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Int_col_onlyBL_CT_Meta_allMarker.csv")

saveRDS(Int_col_BL_CT_meta_mean, file = "Int_col_onlyBL_CT_Meta_SeuratObj.RDS")


gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	

gg_Fig <- FeaturePlot(Int_col_BL_CT_meta_mean, pt.size = 1, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_Meta_feature_GeneralMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()

gene_names = c("UNC5B",
"BMP2",
"DLX3",
"DLX5",
"MSX2",
"IL11",
"MMP11",
"HOXA13",
"SLITRK6",
"SPRY2",
"MYH11",
"BOX5-DLX6",
"MKX",
"CILP2",
"CILP",
"ASPN",
"THBS4",
"CDH15",
"ZFHX3",
"DR999_PMT18929-SCX",
"DIO3")
gene_ids = c("AMEX60DD035718",
"AMEX60DD009952",
"AMEX60DD022355",
"AMEX60DD028758",
"AMEX60DD016524",
"AMEX60DD000336",
"AMEX60DD021953",
"AMEX60DD048713",
"AMEX60DD048778",
"AMEX60DD020580",
"AMEX60DD009950",
"AMEX60DD051903",
"AMEX60DD021849",
"AMEX60DD013919",
"AMEX60DD003432",
"AMEX60DD023964",
"AMEX60DD042297",
"AMEX60DD017092",
"AMEX60DD016992",
"AMEX60DD040818",
"AMEX60DD011352")

gg_Fig <- FeaturePlot(Int_col_BL_CT_meta_mean, pt.size = 3, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_Meta_feature_InterestingMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()


#Score blastema for uninjured

#get list for scoring 

markers.0dpa = read.csv("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_0wpa_CoarseCellTypeMarker.csv")
markers.0dpa  = markers.0dpa %>% arrange(desc(avg_log2FC)) %>% group_by(cluster)  %>% slice(1:30)


markers.0dpa = as.data.frame(markers.0dpa)

naive_cart <- list(markers.0dpa$gene[markers.0dpa$cluster == "Cartilage"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = naive_cart,
  name = 'naive_cart'
)
naive_psm <- list(markers.0dpa$gene[markers.0dpa$cluster == "PSMlike"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = naive_psm,
  name = 'naive_psm'
)
naive_mesen <- list(markers.0dpa$gene[markers.0dpa$cluster == "Mesenchyme"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = naive_mesen,
  name = 'naive_mesen'
)
naive_dermis <- list(markers.0dpa$gene[markers.0dpa$cluster == "Dermis"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = naive_dermis,
  name = 'naive_dermis'
)
naive_prog <- list(markers.0dpa$gene[markers.0dpa$cluster == "Progenitors"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = naive_prog,
  name = 'naive_prog'
)

pdf("Int_col_onlyBL_CT_Meta_UninjuredScores.pdf",width=10,height=10)
FeaturePlot(Int_col_BL_CT_meta_mean,c("naive_cart1","naive_prog1","naive_dermis1","naive_psm1","naive_mesen1"), pt.size = 2, cols = c("grey",rev(viridis_pal(option = "viridis")(12))))
dev.off()





#Score blastema for embryonic tail bud

#get list for scoring 

markers.embryo = read.csv("~/Projects/SingleCellGroup/Axolotl/EmbryonicTailData_SS2/Tail_LB_int_clean_allMarker.csv")
markers.embryo  = markers.embryo %>% arrange(desc(avg_logFC)) %>% group_by(cluster)  %>% slice(1:30)


markers.embryo = as.data.frame(markers.embryo)

embryo_cart <- list(markers.embryo$gene[markers.embryo$cluster == "3"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = embryo_cart,
  name = 'embryo_cart'
)
embryo_psm <- list(markers.embryo$gene[markers.embryo$cluster == "1"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = embryo_psm,
  name = 'embryo_psm'
)
embryo_mesen <- list(markers.embryo$gene[markers.embryo$cluster == "0"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = embryo_mesen,
  name = 'embryo_mesen'
)

embryo_prog <- list(markers.embryo$gene[markers.embryo$cluster == "2"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = embryo_prog,
  name = 'embryo_prog'
)

pdf("Int_col_onlyBL_CT_Meta_EmbryoScores.pdf",width=10,height=10)
FeaturePlot(Int_col_BL_CT_meta_mean,c("embryo_cart1","embryo_prog1","embryo_psm1","embryo_mesen1"), pt.size = 2, cols = c("grey",rev(viridis_pal(option = "viridis")(12))))
dev.off()




#Score blastema for uninjured limb data

#get list for scoring 

markers.limb = read.csv("~/Projects/SingleCellGroup/Axolotl/Limb_BL_0dpa_Genome/limbBL_0dpa_CT_allMarker.csv")
markers.limb  = markers.limb %>% arrange(desc(avg_log2FC)) %>% group_by(cluster)  %>% slice(1:30)


markers.limb = as.data.frame(markers.limb)

fCT <- list(markers.limb$gene[markers.limb$cluster %in% c(0,1,2)])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = fCT,
  name = 'fCT'
)

cart <- list(markers.limb$gene[markers.limb$cluster == "5"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = cart,
  name = 'cart'
)


peri <- list(markers.limb$gene[markers.limb$cluster == "7"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = peri,
  name = 'peri'
)


tendon <- list(markers.limb$gene[markers.limb$cluster == "3"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = tendon,
  name = 'tendon'
)


fCT2 <- list(markers.limb$gene[markers.limb$cluster == "6"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = fCT2,
  name = 'fCT2'
)

dermis <- list(markers.limb$gene[markers.limb$cluster == "4"])

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = dermis,
  name = 'dermis'
)


pdf("Int_col_onlyBL_CT_Meta_LimbScores.pdf",width=10,height=10)
FeaturePlot(Int_col_BL_CT_meta_mean,c("fCT1","fCT21","tendon1","cart1","dermis1","peri1"), pt.size = 2, cols = c("grey",rev(viridis_pal(option = "viridis")(12))))
dev.off()







#get dermis score based on 0dpa Prayag data (cluster 4 with avgFC >= 1 and diff >= 0.1)
#"COL7A1","COL24A1","TAT","APCDD1","LAMC3","NPNT","LOC115460157","TWIST3","COL6A6","F2R","GJA1","LOC110077393","XELAEV_18003999MG"
dermis_list <- list(c("AMEX60DD023361","AMEX60DD019005","AMEX60DD016957","AMEX60DD038203","AMEX60DD050389","AMEX60DD044651","AMEX60DD043179","AMEX60DD029436","AMEX60DD022617","AMEX60DD042352","AMEX60DD034120","AMEX60DD042699","AMEX60DD024182"))
  
Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = dermis_list,
  name = 'Dermis'
)

#get fCT2 score based on 0dpa Prayag data (cluster 4 with avgFC >= 1 and diff >= 0.1)
#"EPPK1","F5","KLF5","OLFML2A","NR2F2","LOC115076211-SSTR2","N337_11780-KRT5","AB205_0156120-KRT19","LOC115459871-ANPEP","EMP1","PCDH19","ARAP1","APOE","ANXA1","ALCAM","TGFBI","ITIH5","CAV2","TFPI","EHD2","CAVIN1","BAMBI","APOC1","LOC101955705","LOC106706246","OLFML1","JAG1","CLDN1","TNFAIP6","FOXD2-FOXD1","CRABP2","CAVIN2","TNFSF12","DACT1","CYP4B1","DHRS3","AFAP1","SULF2","COL6A6","LMCD1","PDGFRL","JUP","LOC102353832-LMO4","CAV1","SFRP4","CDH2","COL28A1","HEXB","TINAGL1","KRT5","KRT18","GPC4"

fCT2_list <- list(c("AMEX60DD040762","AMEX60DD047565","AMEX60DD048840","AMEX60DD050542","AMEX60DD003483","AMEX60DD020737","AMEX60DD029832","AMEX60DD010100","AMEX60DD003314","AMEX60DD029166","AMEX60DD037682","AMEX60DD049857","AMEX60DD018143","AMEX60DD043190","AMEX60DD047325","AMEX60DDU001002594","AMEX60DD008143","AMEX60DD006318","AMEX60DD055299","AMEX60DD024434","AMEX60DD010031","AMEX60DD021836","AMEX60DD018145","AMEX60DD034470","AMEX60DD004576","AMEX60DD004862","AMEX60DD035775","AMEX60DD003189","AMEX60DD055863","AMEX60DD019707","AMEX60DD015718","AMEX60DD028077","AMEX60DD013007","AMEX60DDU001000645","AMEX60DD019712","AMEX60DD052111","AMEX60DD045942","AMEX60DD027586","AMEX60DD022617","AMEX60DD023550","AMEX60DD045459","AMEX60DD010084","AMEX60DD051000","AMEX60DD006319","AMEX60DD038079","AMEX60DD039579","AMEX60DD022331","AMEX60DD042398","AMEX60DD006028","AMEX60DD029823","AMEX60DD029807","AMEX60DD037492"))

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = fCT2_list,
  name = 'fCT2'
)

#get tendon score based on 0dpa Prayag data (cluster 3 with avgFC >= 1 and diff >= 0.1)
#"TNMD","COL11A2","FMOD","COL11A1","ASPN","COL2A1","THBS4","COMP","CILP","CILP2","CHODL","PRSS35","FBLN1","OLFM3","COL16A1","CORIN","PTX4","LOC115077920-PCOLCE","IYD","DR999_PMT18929-SCX","TNC","TGFBI","NEXN","GFRA1","BMP5","PDGFRL","LOC107116286-VWDE","COL13A1","MKX","TIMP1","LOC109101717-SIX1","F5","LOC115482310-TMEM176B","SMOC2","PAMR1","CSPG5","ITGA8","DPP10","THBS2","MFAP4","LOC113406022","AGRN","ADAMTSL3","MATN1","SDC2","SPON2",

tendon_list <- list(c("AMEX60DD037674","AMEX60DD010831","AMEX60DD009113","AMEX60DD018809","AMEX60DD023964","AMEX60DD029426","AMEX60DD042297","AMEX60DD014166","AMEX60DD003432","AMEX60DD013919","AMEX60DD047447","AMEX60DD033732","AMEX60DD006493","AMEX60DD018815","AMEX60DD006034","AMEX60DD045587","AMEX60DD020283","AMEX60DD012610","AMEX60DD035566","AMEX60DD040818","AMEX60DD050822","AMEX60DDU001002594","AMEX60DD018963","AMEX60DD053205","AMEX60DD033480","AMEX60DD045459","AMEX60DD022491","AMEX60DD051881","AMEX60DD021849","AMEX60DD020992","AMEX60DD011856","AMEX60DD047565","AMEX60DD007259","AMEX60DD034577","AMEX60DD005111","AMEX60DD026060","AMEX60DD022164","AMEX60DD055999","AMEX60DD034565","AMEX60DD028164","AMEX60DD048549","AMEX60DD051219","AMEX60DD003564","AMEX60DD005459","AMEX60DD040162","AMEX60DD046082"))

Int_col_BL_CT_meta_mean <- AddModuleScore(
  object = Int_col_BL_CT_meta_mean,
  features = tendon_list,
  name = 'tendon'
)

pdf("Int_col_onlyBL_CT_Meta_LimbScores.pdf",width=10,height=10)
FeaturePlot(Int_col_BL_CT_meta_mean,c("Dermis1","tendon1","fCT21"), pt.size = 2, cols = c("grey",rev(viridis_pal(option = "viridis")(12))))
dev.off()


```

####Progenitors

```{r}

setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

Int_col_BL_CT_meta_mean = readRDS(file = "Int_col_onlyBL_CT_Meta_SeuratObj.RDS")


#select progenitors

Int_col_BL_CT_meta_mean_prog = subset(Int_col_BL_CT_meta_mean,idents = c(0,2,3,5,7,10))

pdf("Int_col_onlyBL_CT_Meta_UMAP_PC100_res1_ProgenitorSubset.pdf",width=15,height=15)
DimPlot(object = Int_col_BL_CT_meta_mean_prog, reduction = 'umap', pt.size = 3)
dev.off()

Int_col_BL_CT_meta_mean_prog@meta.data$original.ident = Int_col_BL_CT_meta_mean_prog@active.ident

#scale
#ScaleData(  object,  features = NULL,  assay = NULL,  vars.to.regress = NULL,  split.by = NULL,  model.use = "linear",  use.umi = FALSE,  do.scale = TRUE,  do.center = TRUE,  scale.max = 10,  block.size = 1000,  min.cells.to.block = 3000,  verbose = TRUE)

all.genes <- rownames(Int_col_BL_CT_meta_mean_prog)

Int_col_BL_CT_meta_mean_prog = ScaleData(  Int_col_BL_CT_meta_mean_prog, features = all.genes ,vars.to.regress = c("nCount_RNA","percent.mt"))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

Int_col_BL_CT_meta_mean_prog = RunPCA(Int_col_BL_CT_meta_mean_prog,features = all.genes,npcs = 100)

#plot PCA heatmaps
#DimHeatmap(pbmc, dims = 1, cells = 500, balanced = TRUE)

library(RColorBrewer)
library(ggplot2)

pdf("Int_col_onlyBL_CT_Meta_Prog_PCAheatmaps.pdf",width=30,height=20)
DimHeatmap(Int_col_BL_CT_meta_mean_prog, dims = 1:16, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT_meta_mean_prog, dims = 17:32, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT_meta_mean_prog, dims = 33:48, cells = 1000, balanced = TRUE,  ncol = 4, fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
dev.off()

#elbow plot
pdf("Int_col_onlyBL_CT_Meta_Prog_PCelbow.pdf",width=20,height=10)
ElbowPlot(Int_col_BL_CT_meta_mean_prog, ndims = 100)
dev.off()

#Cluster cells
Int_col_BL_CT_meta_mean_prog<- FindNeighbors(Int_col_BL_CT_meta_mean_prog, dims = 1:100)
Int_col_BL_CT_meta_mean_prog <- FindClusters(Int_col_BL_CT_meta_mean_prog, resolution = 1)

#run UMAP
Int_col_BL_CT_meta_mean_prog = RunUMAP(Int_col_BL_CT_meta_mean_prog,dims = 1:100, min.dist = 0.1)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_onlyBL_CT_Meta_Prog_UMAP_PC100_res1.pdf",width=15,height=15)
DimPlot(object = Int_col_BL_CT_meta_mean_prog, reduction = 'umap', pt.size = 3)
DimPlot(object = Int_col_BL_CT_meta_mean_prog, reduction = 'umap',group.by = "original.ident", pt.size = 3)
DimPlot(object = Int_col_BL_CT_meta_mean_prog, reduction = 'umap',group.by = "orig.ident", pt.size = 3)

DimPlot(object = Int_col_BL_CT_meta_mean_prog, reduction = 'umap',group.by = "fraction_1wpa", pt.size = 3)
FeaturePlot(Int_col_BL_CT_meta_mean_prog, pt.size = 1, features = c("nFeature_RNA","nCount_RNA","percent.mt","G2M.Score"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


gene_ids = c("AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD023726","AMEX60DD054625","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD022355","AMEX60DD009984","AMEX60DD009989","AMEX60DD009968")
	
gene_names =c("PRRX1","RSPO3","FGF8","WNT5A","TBX2","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","DLX5","SOST","ETV4","PHOSPHO1")

gg_Fig <- FeaturePlot(Int_col_BL_CT_meta_mean_prog, pt.size = 1, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_Meta_Prog_feature_SpecificMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()



markers <- FindAllMarkers(Int_col_BL_CT_meta_mean_prog, only.pos = TRUE,  logfc.threshold = 0.5)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Int_col_onlyBL_CT_Meta_Prog_allMarker.csv")

saveRDS(Int_col_BL_CT_meta_mean_prog, file = "Int_col_onlyBL_CT_Meta_Prog_SeuratObj.RDS")


gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	

gg_Fig <- FeaturePlot(Int_col_BL_CT_meta_mean_prog, pt.size = 1, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_Meta_Prog_feature_GeneralMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()

gene_names = c("UNC5B",
"BMP2",
"DLX3",
"DLX5",
"MSX2",
"IL11",
"MMP11",
"HOXA13",
"SLITRK6",
"SPRY2",
"MYH11",
"BOX5-DLX6",
"MKX",
"CILP2",
"CILP",
"ASPN",
"THBS4",
"CDH15",
"ZFHX3",
"DR999_PMT18929-SCX")
gene_ids = c("AMEX60DD035718",
"AMEX60DD009952",
"AMEX60DD022355",
"AMEX60DD028758",
"AMEX60DD016524",
"AMEX60DD000336",
"AMEX60DD021953",
"AMEX60DD048713",
"AMEX60DD048778",
"AMEX60DD020580",
"AMEX60DD009950",
"AMEX60DD051903",
"AMEX60DD021849",
"AMEX60DD013919",
"AMEX60DD003432",
"AMEX60DD023964",
"AMEX60DD042297",
"AMEX60DD017092",
"AMEX60DD016992",
"AMEX60DD040818")


gg_Fig <- FeaturePlot(Int_col_BL_CT_meta_mean_prog, pt.size = 1, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_onlyBL_CT_Meta_Prog_feature_InterestingMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()


```



###Run Velocity

```{r}

setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

Int_col_BL_CT_meta = readRDS("Int_col_onlyBL_CT_HarmonyTimeInt_MetaRes_SeuratObj.RDS")

Int_col_BL_CT_meta_mean = readRDS(file = "Int_col_onlyBL_CT_Meta_SeuratObj.RDS")

Tail_col_1wpa_spliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/Col1a2_tail_BL_10x/count_123456_114342_1wpa_col1a2_Full_Solo.out/Velocyto/raw/spliced")
Tail_col_1wpa_spliced = Tail_col_1wpa_spliced[rownames(anno), ,drop = FALSE]
Tail_col_1wpa_spliced <- CreateSeuratObject(counts = Tail_col_1wpa_spliced, project = "Tail_col_1wpa_spliced", min.cells = 1, min.features = 1)

Tail_col_1wpa_unspliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/Col1a2_tail_BL_10x/count_123456_114342_1wpa_col1a2_Full_Solo.out/Velocyto/raw/unspliced")
Tail_col_1wpa_unspliced = Tail_col_1wpa_unspliced[rownames(anno), ,drop = FALSE]
Tail_col_1wpa_unspliced <- CreateSeuratObject(counts = Tail_col_1wpa_unspliced, project = "Tail_col_1wpa_unspliced", min.cells = 1, min.features = 1)


Tail_col_2wpa_spliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/HLFYTDRXX_R9843/count_123456_2wpa_col1a2_Full_Solo.out/Velocyto/raw/spliced")
Tail_col_2wpa_spliced = Tail_col_2wpa_spliced[rownames(anno), ,drop = FALSE]
Tail_col_2wpa_spliced <- CreateSeuratObject(counts = Tail_col_2wpa_spliced, project = "Tail_col_2wpa_spliced", min.cells = 1, min.features = 1)

Tail_col_2wpa_unspliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/HLFYTDRXX_R9843/count_123456_2wpa_col1a2_Full_Solo.out/Velocyto/raw/unspliced")
Tail_col_2wpa_unspliced = Tail_col_2wpa_unspliced[rownames(anno), ,drop = FALSE]
Tail_col_2wpa_unspliced <- CreateSeuratObject(counts = Tail_col_2wpa_unspliced, project = "Tail_col_2wpa_unspliced", min.cells = 1, min.features = 1)



Int_col_BL_CT_spliced = merge(Tail_col_1wpa_spliced, y = Tail_col_2wpa_spliced, add.cell.ids = c("wpa1", "wpa2"), project = "Tail_col_Integration_onlyBL_spliced")

Int_col_BL_CT_unspliced = merge(Tail_col_1wpa_unspliced, y = Tail_col_2wpa_unspliced, add.cell.ids = c("wpa1", "wpa2"), project = "Tail_col_Integration_onlyBL_unspliced")


#subset

library(velocyto.R)

Int_col_BL_CT_spliced = subset(Int_col_BL_CT_spliced,cells = colnames(Int_col_BL_CT_int))
Idents(Int_col_BL_CT_spliced) = Int_col_BL_CT_meta@meta.data$seurat_clusters

#merge clusters to meta cells

Int_col_BL_CT_spliced_mean = AverageExpression(
  Int_col_BL_CT_spliced,
  return.seurat = T,
  group.by = "ident",
  slot = "counts")

Int_col_BL_CT_spliced_mean@meta.data$fraction_1wpa = Int_col_BL_CT_meta_mean$fraction_1wpa


Int_col_BL_CT_unspliced = subset(Int_col_BL_CT_unspliced,cells = colnames(Int_col_BL_CT_int))
Idents(Int_col_BL_CT_unspliced) = Int_col_BL_CT_meta@meta.data$seurat_clusters

#merge clusters to meta cells

Int_col_BL_CT_unspliced_mean = AverageExpression(
  Int_col_BL_CT_unspliced,
  return.seurat = T,
  group.by = "ident",
  slot = "counts")

Int_col_BL_CT_unspliced_mean@meta.data$fraction_1wpa = Int_col_BL_CT_meta_mean$fraction_1wpa

#Velocity 

emat <- GetAssayData(Int_col_BL_CT_spliced_mean, slot = "counts")
nmat <- GetAssayData(Int_col_BL_CT_unspliced_mean, slot = "counts")

# take cluster labels
cluster.label <- Int_col_BL_CT_meta_mean@meta.data$seurat_cluster
names(cluster.label) = colnames(Int_col_BL_CT_meta_mean)

gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

cell.colors = gg_color_hue(length(unique(Int_col_BL_CT_meta_mean@meta.data$seurat_cluster)))
#cell.colors <-  colorRampPalette(brewer.pal(12,"Set1"))(length(unique(BLct_CSDct_filter@meta.data$orig.ident)))

names(cell.colors) <- unique(cluster.label)

cell.cols <- cell.colors[cluster.label]
names(cell.cols) = colnames(Int_col_BL_CT_meta_mean)

# take embedding
emb <- Embeddings(Int_col_BL_CT_meta_mean, "umap")

# filter genes
emat <- filter.genes.by.cluster.expression(emat,cluster.label,min.max.cluster.average = 0.5)
nmat <- filter.genes.by.cluster.expression(nmat,cluster.label,min.max.cluster.average = 0.05)

length(intersect(rownames(emat),rownames(emat)))

#Estimate RNA velocity (using gene-relative model with k=20 cell kNN pooling and using top/bottom 2% quantiles for gamma fit):

fit.quantile <- 0.2
rvel.cd <- gene.relative.velocity.estimates(emat,nmat,deltaT=1,kCells=10,fit.quantile=fit.quantile)
#cell.dist=cell.dist,

#Visualize velocity on the t-SNE embedding, using velocity vector fields:
pdf("Int_col_onlyBL_CT_Meta_UMAP_Velocity.pdf",width=10,height=10)
show.velocity.on.embedding.cor(emb,rvel.cd,n=500,scale='sqrt',cell.colors=cell.cols,cex=0.8,arrow.scale=0.5,show.grid.flow=TRUE,min.grid.cell.mass=0.1,grid.n=50,do.par=F,n.cores=50,cell.border.alpha = 0)
dev.off()

#Zhisong example
#show.velocity.on.embedding.cor(spring.coor,rvel.cd,n=200,scale='sqrt',cell.colors=cell.cols,cex=2,arrow.scale=1000,show.grid.flow=TRUE,min.grid.cell.mass=0.1,grid.n=80,do.par=T,cell.border.alpha = 0.1,n.cores=25, arrow.lwd = 2, bty="n", xaxt="n", yaxt="n")

```



Run in bash
Check possible environments
conda info --envs

conda activate /home/tobias_gerber/.conda/envs/axoloom

python

Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scvelo as scv




```{r}
#make anndata object for scVelo


library("reticulate")
use_condaenv("/home/tobias_gerber/.conda/envs/axoloom", required = TRUE)
scv <- import("scvelo")

ad <- import("anndata", convert = FALSE)
dfobs <- data.frame(seurat_cluster = Int_col_BL_CT_meta_mean@meta.data$seurat_cluster,fraction_1wpa =  Int_col_BL_CT_meta_mean@meta.data$fraction_1wpa)
rownames(dfobs) <- colnames(Int_col_BL_CT_meta_mean)
#dfvar <- data.frame(genes_attributes)
#rownames(dfvar) <- genes

#bring all matrices to the same gene number

subsample = unique(intersect(intersect(intersect(rownames(Int_col_BL_CT_spliced_mean),rownames(Int_col_BL_CT_meta_mean)),intersect(rownames(Int_col_BL_CT_unspliced_mean),rownames(Int_col_BL_CT_meta_mean))),intersect(rownames(Int_col_BL_CT_spliced_mean),rownames(Int_col_BL_CT_unspliced_mean))))

subsample_mat <- subset(Int_col_BL_CT_meta_mean, features = subsample)
subsample_mat = GetAssayData(subsample_mat, slot = "counts")

subsample_emat <- subset(Int_col_BL_CT_spliced_mean, features = subsample)
subsample_emat = GetAssayData(subsample_emat, slot = "counts")

subsample_nmat <- subset(Int_col_BL_CT_unspliced_mean, features = subsample)
subsample_nmat = GetAssayData(subsample_nmat, slot = "counts")

adata <- ad$AnnData(
  X=t(subsample_mat),
  obs=dfobs,
 # var=dfvar,
  layers=list('spliced'=t(subsample_emat), 'unspliced'=t(subsample_nmat)),
  obsm=list('X_umap'=emb, 'X_pca'=Embeddings(Int_col_BL_CT_meta_mean, "pca")) 
  )

## run scvelo dynamic model
scv$pp$filter_genes(adata) ## filter
scv$pp$moments(adata,n_pcs=100,n_neighbors=as.integer(10)) ## normalize and compute moments
scv$tl$recover_dynamics(adata) ## model

## takes awhile, so uncomment to save
#adata$write('data/pancreas.h5ad', compression='gzip')
#adata = scv$read('data/pancreas.h5ad')

## plot (creates pop up window)

scv$tl$velocity(adata, mode='dynamical')
scv$tl$velocity_graph(adata)
scv$pl$velocity_embedding_stream(adata,layer=c('velocity'),color=c('seurat_cluster'), basis='umap',dpi = 300, save = 'Int_col_onlyBL_CT_Meta_UMAP_scVelo.svg')



```


###Run TEI

```{r}
# example to add TEI (transcriptom evolutionary index) values to seurat object

## load libray
library(dplyr)
library(data.table)
library(Matrix)
library(ggplot2)
library(viridis)

## define function
TEIsparse <- function(counts, ps){
  counts_ps_common_ids <- sort(Reduce(intersect, list(rownames(counts), names(ps))))
  counts <- counts[rownames(counts) %in% counts_ps_common_ids,,drop=FALSE]
  counts <- counts[order(rownames(counts)),]
  ps <- ps[names(ps) %in% counts_ps_common_ids]
  ps <- ps[order(names(ps))]
  m2 <- data.table::as.data.table(summary(counts))
  m2$ps <- ps[m2$i]
  sumx <- (m2 %>% dplyr::group_by(j) %>% dplyr::summarise(sumx=sum(x)))
  teisum <- (m2 %>% dplyr::group_by(j) %>% dplyr::summarise(teisum=sum(x*ps)))
  tei <- teisum$teisum/sumx$sumx
  names(tei) <- colnames(counts)
  return(tei)
}

## load PSmap
load("./AMBMEX_PS.RData")

## create named ps vector (mandatory that names correspond to gene names of seurat object)
ps <- setNames(AMBMEX_PS$minPS$B62e3, AMBMEX_PS$minPS$GeneID)

## load seurat object
#limb <- readRDS("limb_ALL_Axo_SeuratObj.RDS")

## add TEI values
Int_col_BL_CT_meta_mean@meta.data["TEI"] <- TEIsparse(Int_col_BL_CT_meta_mean@assays$RNA@counts, ps)



## boxplot
p <- ggplot(Int_col_BL_CT_meta_mean@meta.data, aes(TEI, seurat_clusters)) + geom_boxplot()


pdf("Int_col_onlyBL_CT_Meta_UMAP_TAI.pdf",width=10,height=10)

p

## RidgePlot
RidgePlot(Int_col_BL_CT_meta_mean, "TEI")

## FeaturePlot
FeaturePlot(Int_col_BL_CT_meta_mean,pt.size= 3, "TEI")
FeaturePlot(Int_col_BL_CT_meta_mean, "TEI", min.cutoff='q05', max.cutoff='q95',pt.size= 3, cols = c(rev(viridis_pal(option = "turbo")(12))))

dev.off()

```


##Combine all meta

```{r}
library(Seurat)

#load data sets

Tail_0wpa_all = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_0wpa_CT_harm_SeuratObj.RDS")
Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_Meta_SeuratObj.RDS")

Tail_0wpa_all@meta.data$origin = Tail_0wpa_all@meta.data$orig.ident
Tail_1_2wpa@meta.data$origin = "blastema"


#merge ojects
#merge(x = NULL, y = NULL, add.cell.ids = NULL, merge.data = TRUE, project = "SeuratProject", ...)

Tail_all = merge(Tail_0wpa_all, y = Tail_1_2wpa, add.cell.ids = c("0wpa", "Blastema"), project = "Tail_0_1_2wpa_Integration")

Tail_all = NormalizeData(Tail_all)

#parallize scaling
library(future)
plan("multiprocess", workers = 8)
plan()

#set maximum to 10GB for each worker
options(future.globals.maxSize = 50000 * 1024^2)


#get cell cycle genes

g2m.genes <- cc.genes$g2m.genes
g2m.contig <- intersect(rownames(anno)[anno$V2 %in% g2m.genes ] , rownames(Tail_all) )

s.genes <- cc.genes$s.genes
s.contig <- intersect(rownames(anno)[anno$V2 %in% s.genes ] , rownames(Tail_all) )

#cell cycle scoring
Tail_all <- CellCycleScoring(Tail_all, s.features = s.contig, g2m.features = g2m.contig, set.ident = TRUE)



#scale
#ScaleData(  object,  features = NULL,  assay = NULL,  vars.to.regress = NULL,  split.by = NULL,  model.use = "linear",  use.umi = FALSE,  do.scale = TRUE,  do.center = TRUE,  scale.max = 10,  block.size = 1000,  min.cells.to.block = 3000,  verbose = TRUE)

all.genes <- rownames(Tail_all)

Tail_all = ScaleData(  Tail_all, features = all.genes ,vars.to.regress = c("nCount_RNA","percent.mt"))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

Tail_all = RunPCA(Tail_all,features = all.genes,npcs = 100)

#plot PCA heatmaps
#DimHeatmap(pbmc, dims = 1, cells = 500, balanced = TRUE)

library(RColorBrewer)

#elbow plot
pdf("Tail_0_1_2wpa_PCelbow.pdf",width=20,height=10)
ElbowPlot(Tail_all, ndims = 100)
dev.off()

#Cluster cells
Tail_all<- FindNeighbors(Tail_all, dims = 1:50)
Tail_all <- FindClusters(Tail_all, resolution = 0.6)

#run UMAP
Tail_all = RunUMAP(Tail_all,dims = 1:50)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Tail_0_1_2wpa_UMAP_PC50_res0.6.pdf",width=20,height=20)
DimPlot(object = Tail_all, reduction = 'umap', pt.size = 3)
DimPlot(object = Tail_all, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
FeaturePlot(Tail_all, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD002658","AMEX60DD052322","eGFP","mCherry"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


#Harmony
library(harmony)

Tail_all_int = RunHarmony(object = Tail_all,group.by.vars= "origin", assay.use = "RNA", dims.use = 1:100, plot_convergence = TRUE, max.iter.cluster = 30 ,max.iter.harmony=50,theta=2,lambda=3)


#Cluster cells
Tail_all_int<- FindNeighbors(Tail_all_int, dims = 1:100, reduction = "harmony")
Tail_all_int <- FindClusters(Tail_all_int, resolution = 0.6)


#run UMAP
Tail_all_int = RunUMAP(Tail_all_int,dims = 1:100, reduction = "harmony", min.dist = 0.1)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Tail_0_1_2wpa_Harmony_UMAP_PC100_res0.6.pdf",width=20,height=20)
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap', pt.size = 3)
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap',group.by = "orig.ident", pt.size = 3)
DimPlot(object = Tail_all_int, shuffle= T, reduction = 'umap',group.by = "origin", pt.size = 3)
FeaturePlot(Tail_all_int, pt.size = 1, features = c("AMEX60DD018705","AMEX60DD023361","AMEX60DD029436","AMEX60DD009987","AMEX60DD052322","AMEX60DD008407","AMEX60DD005746","AMEX60DD048713","AMEX60DD001056"), raster = T, order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))

FeaturePlot(Tail_all_int,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
	
dev.off()

saveRDS(Tail_all_int, file = "Tail_0_1_2wpa_Harmony_SeuratObj.RDS")


```

###SPRING

```{r}

Int_col_BL_CT_int = readRDS(file = "Int_col_BL_CT_HarmonyTimeInt_final_SeuratObj.RDS")

###Prepare data

harmony.embedding = Embeddings(Int_col_BL_CT_int, "harmony")


#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}


#use only cluster marker to run SPRING

markers <- FindAllMarkers(Int_col_BL_CT_int, only.pos = TRUE,  logfc.threshold = 0.5)
regeneration.marker = as.character(unique(markers$gene[ markers$avg_logFC > 0.5]))

#extract norm data from Seurat object

norm.data = GetAssayData(object = Int_col_BL_CT_int, slot = "data")
norm.data = as.matrix(norm.data)


meta.data = FetchData(object = Int_col_BL_CT_int, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA"))

##############################
#SPRING
##############################

### export for SPRING (export norma and not scaled data)

#cutoff 0.8
y=as.data.frame(t(meta.data[,c("orig.ident","ident")]),stringsAsFactors=F)

write.table(y,sep = ",",col.names = F,file = "SPRING_Export_Int_col_BL_CT_meta.csv")

x = as.data.frame(t(harmony.embedding[,c(1:50)]))

write.table(x,sep = ",",col.names = F,file ="SPRING_Export_Int_col_BL_CT_HarmonyMatrix.csv")

z=rownames(x)

write.table(z,col.names = F,row.names = F,sep = ",",file = "SPRING_Export_Int_col_BL_CT_GeneList.csv")

###Run with : 50 PCs gene / 15 neighbors / gene variability percentile 20.0

#does not look promising... get pseudo time estimates onto UMAP
```

##Combine all sc

```{r}
library(Seurat)

#load data sets
setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

#load data sets

Tail_0wpa_all = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_0wpa_CT_harm_SeuratObj.RDS")
Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")

Tail_0wpa_all@meta.data$tissue = "uninjured" 
Tail_1_2wpa@meta.data$tissue = "blastema"

#merge ojects
#merge(x = NULL, y = NULL, add.cell.ids = NULL, merge.data = TRUE, project = "SeuratProject", ...)

Tail_all = merge(Tail_0wpa_all, y = Tail_1_2wpa, add.cell.ids = c("Uninjured", "Blastema"), project = "Tail_0_1_2wpa_Integration")

Tail_all = NormalizeData(Tail_all)

#parallize scaling
library(future)
plan("multiprocess", workers = 10)
plan()

#set maximum to 10GB for each worker
options(future.globals.maxSize = 10000 * 1024^2)


#get cell cycle genes

g2m.genes <- cc.genes$g2m.genes
g2m.contig <- intersect(rownames(anno)[anno$V2 %in% g2m.genes ] , rownames(Tail_all) )

s.genes <- cc.genes$s.genes
s.contig <- intersect(rownames(anno)[anno$V2 %in% s.genes ] , rownames(Tail_all) )

#cell cycle scoring
Tail_all <- CellCycleScoring(Tail_all, s.features = s.contig, g2m.features = g2m.contig, set.ident = TRUE)



#scale
#ScaleData(  object,  features = NULL,  assay = NULL,  vars.to.regress = NULL,  split.by = NULL,  model.use = "linear",  use.umi = FALSE,  do.scale = TRUE,  do.center = TRUE,  scale.max = 10,  block.size = 1000,  min.cells.to.block = 3000,  verbose = TRUE)

all.genes <- rownames(Tail_all)

Tail_all = ScaleData(  Tail_all, features = all.genes ,vars.to.regress = c("nCount_RNA","percent.mt"))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

Tail_all = RunPCA(Tail_all,features = all.genes,npcs = 100)

#plot PCA heatmaps
#DimHeatmap(pbmc, dims = 1, cells = 500, balanced = TRUE)

library(RColorBrewer)

#elbow plot
pdf("Tail_0_1_2wpa_sc_PCelbow.pdf",width=20,height=10)
ElbowPlot(Tail_all, ndims = 100)
dev.off()

#Cluster cells
Tail_all<- FindNeighbors(Tail_all, dims = 1:50)
Tail_all <- FindClusters(Tail_all, resolution = 0.6)

#run UMAP
Tail_all = RunUMAP(Tail_all,dims = 1:50)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Tail_0_1_2wpa_sc_UMAP_PC50_res0.6.pdf",width=20,height=20)
DimPlot(object = Tail_all, reduction = 'umap', pt.size = 3)
DimPlot(object = Tail_all, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
FeaturePlot(Tail_all, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD002658","AMEX60DD052322","eGFP","mCherry"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


#Harmony
library(harmony)

Tail_all_int = RunHarmony(object = Tail_all,group.by.vars= "orig.ident", assay.use = "RNA", dims.use = 1:50, plot_convergence = TRUE, max.iter.cluster = 30 ,max.iter.harmony=50,theta=2,lambda=9)


#Cluster cells
Tail_all_int<- FindNeighbors(Tail_all_int, dims = 1:50, reduction = "harmony")
Tail_all_int <- FindClusters(Tail_all_int, resolution = 3)

library(tidyverse)
library(factoextra)
Data.hc <- Tail_all_int@reductions$harmony@cell.embeddings[,c(1:50)] %>% get_dist(method = "pearson") %>% hclust(method = 'ward.D') ## PCs
Hcls <- data.frame(Clusters = as.factor(cutree(Data.hc, k = 35)))
Hcls <- Hcls %>% mutate(Wells = colnames(Tail_all_int))

Tail_all_int@meta.data$Hclust = Hcls$Clusters

#run UMAP
Tail_all_int = RunUMAP(Tail_all_int,dims = 1:50, reduction = "harmony", min.dist = 0.3)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Tail_0_1_2wpa_sc_Harmony_UMAP_PC50_res3.pdf",width=20,height=20)
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap', pt.size = 3,label = T, group.by = "Hclust")
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3, cols = c("firebrick","firebrick1","deepskyblue1","dodgerblue","firebrick3","firebrick4"))
FeaturePlot(Tail_all_int,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


gene_ids = c("AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD023726","AMEX60DD054625","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD022355","AMEX60DD009984","AMEX60DD009989","AMEX60DD009968","AMEX60DD021181")
	
gene_names =c("PRRX1","RSPO3","FGF8","WNT5A","TBX2","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","DLX5","SOST","ETV4","PHOSPHO1","LFNG")

gg_Fig <- FeaturePlot(Tail_all_int, pt.size = 2, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_0_1_2wpa_sc_Harmony_feature_SpecificMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()



Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(16,20,45,41)] = "Asomitic cells"
Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(0,2,8,9,7,22,15,39,40,36,23,32,27,25,13)] = "Intermediate Progenitors"
Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(14,1,6,38,24,18)] = "Committed Chondrocytes"
Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(4,5,42,43,33)] = "Chondrocytes"
Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(26,28,44,46)] = "Myogenic lineage"
Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(3,30,31,10,21,35,34)] = "Fin Mesenchyme"
Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(19,29,17,12,11,37)] = "Dermis"


Tail_all_int@meta.data$celltype <- factor(Tail_all_int@meta.data$celltype, levels = c("Asomitic cells","Intermediate Progenitors","Committed Chondrocytes","Chondrocytes","Fin Mesenchyme","Dermis","Myogenic lineage"))

#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$Hclust %in% c(12,13,20)] = "Asomitic cells"
#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$Hclust %in% c(3,5,8,16,18,31,33,35,24,29,1,14,21,7,2)] = "Intermediate Progenitors"
#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$Hclust %in% c(19,28,30)] = "Committed Chondrocytes"
#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$Hclust %in% c(22,27,32,34)] = "Chondrocytes"
#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$Hclust %in% c(11,25)] = "Myogenic lineage"
#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$Hclust %in% c(10,23,26,17)] = "Fin Mesenchyme"
#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$Hclust %in% c(4,6,9,15)] = "Dermis"
#Tail_all_int@meta.data$celltype[Tail_all_int@meta.data$seurat_clusters %in% c(4)] = "Committed Fibroblasts"


Tail_all_int@meta.data$timepoint[Tail_all_int@meta.data$tissue %in% "uninjured"] = "BL_0wpa"
Tail_all_int@meta.data$timepoint[Tail_all_int@meta.data$orig.ident %in% "Tail_col_1wpa"] = "BL_1wpa"
Tail_all_int@meta.data$timepoint[Tail_all_int@meta.data$orig.ident %in% "Tail_col_2wpa"] = "BL_2wpa"



library(grafify)
my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[1:9])

pdf("Tail_0_1_2wpa_sc_Harmony_UMAP_PC50_res3.pdf",width=20,height=20)
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap', pt.size = 3,label = T,group.by = "seurat_clusters", label.size = 10, cols = my_spectral_palette(length(unique(Tail_all_int@meta.data$seurat_clusters))))

my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[1:6])
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap',label = F, group.by = "orig.ident", pt.size = 3, cols = my_spectral_palette(length(unique(Tail_all_int@meta.data$orig.ident))))

DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap',label = F, group.by = "tissue", pt.size = 3, cols = c("goldenrod3","#1f78b4"))

my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[1:6])
DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap', group.by = "celltype", pt.size = 3, cols = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00")) 

FeaturePlot(Tail_all_int,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA","G2M.Score"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))


DimPlot(object = Tail_all_int,  shuffle= T,reduction = 'umap',label = F, group.by = "timepoint", pt.size = 3, cols = c("#E2DE66","#98D3AA","#0FA39B"))


dev.off()

saveRDS(Tail_all_int, file = "Tail_0_1_2wpa_sc_Harmony_SeuratObj.RDS")

Idents(Tail_all_int) = Tail_all_int@meta.data$celltype

#Find Markers
markers <- FindAllMarkers(Tail_all_int, only.pos = TRUE,  logfc.threshold = 0.1)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Tail_0_1_2wpa_sc_celltype_Marker.csv")




gene_names = c(
#Asomitic
"MEOX1",
"LFNG",
"MKX",
"SCX",
"TNMD",
#Progenitors 1/2wpa
"KIF26A",
"PCLAF",
"TOP2A",
"IL11",
"LEP",
#Progenitors 0wpa
#"CRABP2",
#"NGFR",
#"OTOR",
#"CDO1",
#"VWC2L",
#Committed chondrocytes
"CHRDL1",
"CHRDL2",
"FAT4",
"C17ORF67",
"FGFR2",
#Cartilage
"CNMD",
"ACAN",
"SOX9",
"MATN1",
"FGFBP2",
#Mesenchyme
"SOST",
"LTBP4",
"DLX5",
"DKK2",
"F13A1",
#Dermis
"DPT",
"TWIST3",
"PAH",
"FAM180A",
"OLFML2A",
#Muscle
"PAX7",
"MYF5",
"MYL1",
"MYLPF",
"ACTC1")

gene_ids = c(
#Asomitic
"AMEX60DD009987",
"AMEX60DD021181",
"AMEX60DD021849",
"AMEX60DD040818",
"AMEX60DD037674",
#Progenitors 1/2wpa
"AMEX60DD011082",
"AMEX60DD003844",
"AMEX60DD010148",
"AMEX60DD016524",
"AMEX60DD007977",
#Progenitors 0wpa
#"AMEX60DD015718",
#"AMEX60DD009964",
#"AMEX60DD035833",
#"AMEX60DD042801",
#"AMEX60DD019831",
#Committed chondrocytes
"AMEX60DD037123",
"AMEX60DD049819",
"AMEX60DD044847",
"AMEX60DD031033",
"AMEX60DD053280",
#Cartilage
"AMEX60DD048972",
"AMEX60DD004178",
"AMEX60DD031236",
"AMEX60DD005459",
"AMEX60DD045825",
#Mesenchyme
"AMEX60DD009984",
"AMEX60DD024616",
"AMEX60DD022355",
"AMEX60DD044644",
"AMEX60DD038300",
#Dermis
"AMEX60DD047546",
"AMEX60DD029436",
"AMEX60DD008421",
"AMEX60DD007033",
"AMEX60DD050542",
#Muscle
"AMEX60DD052322",
"AMEX60DD007644",
"AMEX60DD055382",
"AMEX60DD027986",
"AMEX60DD025304")

gg_Fig <- VlnPlot(Tail_all_int, pt.size = 0, features = gene_ids, group.by = "celltype", cols = c("plum4","darkseagreen2","turquoise1","coral2","darkgoldenrod3","turquoise4","violetred2","darkgoldenrod3")) 
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_0_1_2wpa_sc_Harmony_Vln_PaperMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()


pdf("Tail_0_1_2wpa_sc_Harmony_Dot_PaperMarker.pdf",width=5,height=10)
DotPlot(Tail_all_int, features = gene_ids, group.by = "celltype") + scale_size(range = c(1,8)) + RotatedAxis() + coord_flip() + theme(axis.text.x=element_text(size=7),axis.text.y=element_text(size=7)) + scale_x_discrete(labels=c(gene_names)) + scale_colour_gradientn(colors = c(rev(viridis_pal(option = "viridis")(12))))
dev.off()


gene_ids = c("AMEX60DD017690",
"AMEX60DD030096",  
"AMEX60DD043904", 
"AMEX60DD039871", 
"AMEX60DD027586", 
"AMEX60DD033627",
"AMEX60DD005267")
gene_names = c("SALL1",
"SPARC",
"SPP1",
"SULF1",
"SULF2",
"COL12A1",
"MDK1")

gg_Fig <- FeaturePlot(Tail_all_int, pt.size = 2, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_0_1_2wpa_sc_Harmony_feature_PrayagRequestMarker.pdf",width=15,height=10)
CombinePlots( gg_Fig )
dev.off()


#Estimate contribution to clusters

#Calculate celltype contribution normalized to total number of cells

plot.df = FetchData(Tail_all_int, c("celltype","tissue"))

plot.df = as.data.frame(table(plot.df))

bl.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$tissue == "blastema"]))[1:7]
un.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$tissue == "uninjured"]))[8:14]

plot.df$norm = c(bl.norm,un.norm)

plot.df.sub = plot.df[plot.df$celltype == "Asomitic cells",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_Asomitic.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = tissue)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("goldenrod3","#1f78b4"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

plot.df.sub = plot.df[plot.df$celltype == "Chondrocytes",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_Chondrocytes.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = tissue)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("goldenrod3","#1f78b4"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

plot.df.sub = plot.df[plot.df$celltype == "Committed Chondrocytes",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_CommittedChondrocytes.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = tissue)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("goldenrod3","#1f78b4"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()


plot.df.sub = plot.df[plot.df$celltype == "Intermediate Progenitors",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_IntermediateProgenitors.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = tissue)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("goldenrod3","#1f78b4"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()


plot.df.sub = plot.df[plot.df$celltype == "Fin Mesenchyme",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_FinMesenchyme.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = tissue)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("goldenrod3","#1f78b4"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

plot.df.sub = plot.df[plot.df$celltype == "Dermis",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_Dermis.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = tissue)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("goldenrod3","#1f78b4"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

plot.df.sub = plot.df[plot.df$celltype == "Myogenic lineage",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_MyogenicLineage.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = tissue)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("goldenrod3","#1f78b4"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()




###Estimate cell type ratios across time points


plot.df = FetchData(Tail_all_int, c("celltype","orig.ident"))

plot.df = as.data.frame(table(plot.df))

plot.df = plot.df[plot.df$orig.ident %in% c("Tail_col_0wpa_v3","Tail_col_1wpa","Tail_col_2wpa"),]

un.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$orig.ident == "Tail_col_0wpa_v3"]))[1:7]
bl1.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$orig.ident == "Tail_col_1wpa"]))[8:14]
bl2.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$orig.ident == "Tail_col_2wpa"]))[15:21]

plot.df$norm = c(un.norm,bl1.norm,bl2.norm)

#uninjured

plot.df.sub = plot.df[plot.df$orig.ident == "Tail_col_0wpa_v3",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_celltypeRatio_Uninjured.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = celltype)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

#wpa1
plot.df.sub = plot.df[plot.df$orig.ident == "Tail_col_1wpa",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_celltypeRatio_wpa1.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = celltype)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

#wpa2
plot.df.sub = plot.df[plot.df$orig.ident == "Tail_col_2wpa",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_celltypeRatio_wpa2.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = celltype)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

##Unite cartilage lineage

Tail_all_int@meta.data$celltype.merge.cart = Tail_all_int@meta.data$celltype

Tail_all_int@meta.data$celltype.merge.cart[Tail_all_int@meta.data$celltype.merge.cart %in% "Committed Chondrocytes"] = "Chondrocytes"

plot.df = FetchData(Tail_all_int, c("celltype.merge.cart","orig.ident"))

plot.df = as.data.frame(table(plot.df))

plot.df = plot.df[plot.df$orig.ident %in% c("Tail_col_0wpa_v3","Tail_col_1wpa","Tail_col_2wpa"),]


un.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$orig.ident == "Tail_col_0wpa_v3"]))[1:7]
bl1.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$orig.ident == "Tail_col_1wpa"]))[8:14]
bl2.norm = c(plot.df$Freq / sum(plot.df$Freq[plot.df$orig.ident == "Tail_col_2wpa"]))[15:21]

plot.df$norm = c(un.norm,bl1.norm,bl2.norm)

#uninjured

plot.df.sub = plot.df[plot.df$orig.ident == "Tail_col_0wpa_v3",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_celltypeRatio_cartMerge_Uninjured.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = celltype.merge.cart)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

#wpa1
plot.df.sub = plot.df[plot.df$orig.ident == "Tail_col_1wpa",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_celltypeRatio_cartMerge_wpa1.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = celltype.merge.cart)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()

#wpa2
plot.df.sub = plot.df[plot.df$orig.ident == "Tail_col_2wpa",]

pdf("./Tail_0_1_2wpa_sc_Harmony_Pie_celltypeRatio_cartMerge_wpa2.pdf",  width = 10, height = 10)
ggplot(data=plot.df.sub, aes(x="", y=norm, fill = celltype.merge.cart)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + scale_fill_manual(values = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00"))+ theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
dev.off()


```

###Marker comparison

```{r}
Tail_0wpa_all = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_0wpa_CT_harm_SeuratObj.RDS")



Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(6)] = "Asomitic cells"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(1,2,9,4,5,10,12,13,14)] = "Dermis"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(0)] = "Fin Mesenchyme"
#Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(7)] = "Developing Mesenchyme"
#Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(8)] = "Developing Dermis"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(3,7,8)] = "Intermediate Progenitors"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(11)] = "Chondrocytes"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(15)] = "Periskeletal"

library(grafify)
my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[1:9])

pdf("Tail_0wpa_sc_UMAP_PC50_res0.8.pdf",width=10,height=10)

DimPlot(object = Tail_0wpa_all, reduction = 'umap',label = T, pt.size = 2,shuffle = T, label.size = 5, group.by = 'seurat_clusters', cols = my_spectral_palette(length(unique(Tail_0wpa_all@meta.data$seurat_clusters))))

my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[1:6])
DimPlot(object = Tail_0wpa_all, reduction = 'umap',label = T, pt.size = 2,shuffle = T, group.by = 'orig.ident',cols = my_spectral_palette(length(unique(Tail_0wpa_all@meta.data$orig.ident))))

DimPlot(object = Tail_0wpa_all, reduction = 'umap',label = T, pt.size = 2,shuffle = T, group.by = 'celltype',cols = c("#E69F00","#1CA59A","#6EB0C2","#236E94","#A0BE67","#77C15A","#D55E00"))

FeaturePlot(Tail_0wpa_all,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA","G2M.Score"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))

dev.off()


Idents(Tail_0wpa_all) = Tail_0wpa_all@meta.data$celltype

#Find Markers
markers <- FindAllMarkers(Tail_0wpa_all, only.pos = TRUE,  logfc.threshold = 0.1)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Tail_0wpa_sc_celltype_Marker.csv")

#find genes that are specific to progenitors and not also found in chondrocytes or other differentiated cells

prog.genes = markers$gene[markers$cluster %in% "Intermediate Progenitors"]
other.genes = markers$gene[!markers$cluster %in% "Intermediate Progenitors"]
prog.genes = prog.genes[!prog.genes %in% other.genes]

prog.anno = markers.anno[markers.anno$gene %in% prog.genes,]

write.csv(prog.anno,"Tail_0wpa_sc_celltype_uniqueProgentorMarker.csv")



#Tail_0wpa_sub = subset(Tail_0wpa_all,subset = celltype %in% c("Intermediate Progenitors"),invert = T)

Tail_0wpa_all@meta.data$celltype <- factor(Tail_0wpa_all@meta.data$celltype, levels = c("Asomitic cells","Intermediate Progenitors","Chondrocytes","Fin Mesenchyme","Dermis","Periskeletal"))


#plot the markers for uncut only data set

gene_names = c(
#Asomitic
"MEOX1",
"LFNG",
"MKX",
"SCX",
"TNMD",
#Progenitors 1/2wpa
#"KIF26A",
#"PCLAF",
#"TOP2A",
#"IL11",
#"LEP",
#Progenitors 0wpa
"CRABP2",
"NGFR",
"OTOR",
"CDO1",
"VWC2L",
#Committed chondrocytes
#"CHRDL1",
#"CHRDL2",
#"FAT4",
#"C17ORF67",
#"FGFR2",
#Cartilage
"CHRDL1",
"CNMD",
"ACAN",
"SOX9",
"MATN1",
"FGFBP2",
#Mesenchyme
"PRRX1",
"SOST",
"LTBP4",
"DLX5",
"DKK2",
"F13A1",
#Dermis
"DPT",
"TWIST3",
"PAH",
"FAM180A",
"OLFML2A",
#Periskeletal 
"DKK3",
"MATN3",
"CLEC3A",
"COL21A1",
"PKDCC")

gene_ids = c(
#Asomitic
"AMEX60DD009987",
"AMEX60DD021181",
"AMEX60DD021849",
"AMEX60DD040818",
"AMEX60DD037674",
#Progenitors 1/2wpa
#"AMEX60DD011082",
#"AMEX60DD003844",
#"AMEX60DD010148",
#"AMEX60DD016524",
#"AMEX60DD007977",
#Progenitors 0wpa
"AMEX60DD015718",
"AMEX60DD009964",
"AMEX60DD035833",
"AMEX60DD042801",
"AMEX60DD019831",
#Committed chondrocytes
#"AMEX60DD037123",
#"AMEX60DD049819",
#"AMEX60DD044847",
#"AMEX60DD031033",
#"AMEX60DD053280",
#Cartilage
"AMEX60DD037123",
"AMEX60DD048972",
"AMEX60DD004178",
"AMEX60DD031236",
"AMEX60DD005459",
"AMEX60DD045825",
#Mesenchyme
"AMEX60DD018450",
"AMEX60DD009984",
"AMEX60DD024616",
"AMEX60DD022355",
"AMEX60DD044644",
"AMEX60DD038300",
#Dermis
"AMEX60DD047546",
"AMEX60DD029436",
"AMEX60DD008421",
"AMEX60DD007033",
"AMEX60DD050542",
#Periskeletal
"AMEX60DD004826",
"AMEX60DD033105",
"AMEX60DD015281",
"AMEX60DD033490",
"AMEX60DD036008")


gg_Fig <- VlnPlot(Tail_0wpa_all, pt.size = 0, features = gene_ids, group.by = "celltype", cols = c("plum4","darkseagreen2","turquoise1","coral2","darkgoldenrod3","turquoise4","violetred2","darkgoldenrod3")) 
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_0wpa_sc_Harmony_Vln_PaperMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()


pdf("Tail_0wpa_sc_Harmony_Dot_PaperMarker.pdf",width=5,height=10)
DotPlot(Tail_0wpa_all, features = gene_ids, group.by = "celltype") + scale_size(range = c(1,8)) + RotatedAxis() + coord_flip() + theme(axis.text.x=element_text(size=7),axis.text.y=element_text(size=7)) + scale_x_discrete(labels=c(gene_names)) + scale_colour_gradientn(colors = c(rev(viridis_pal(option = "viridis")(12))))
dev.off()

gg_Fig <- FeaturePlot(Tail_0wpa_all, pt.size = 2, features = gene_ids,order = T, raster = F,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T) & NoLegend()
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_0wpa_sc_Harmony_Feature_PaperMarker.pdf",width=17,height=20)
CombinePlots( gg_Fig )
dev.off()










#Run Harmony for limb blastema using same conditions

Limb_all = readRDS("/mnt/SingleCellGenomics/scg_projects/Xenopus/Correlation_Axolotl_BL_LB/limb_ALL_Harmony_SeuratObj.RDS")

Limb_all = subset(Limb_all, subset = origin %in% c("BL0dpa","BL11dpa","BL5dpa"))


#Harmony
library(harmony)

Limb_all_int = RunHarmony(object = Limb_all,group.by.vars= "origin", assay.use = "RNA", dims.use = 1:50, plot_convergence = TRUE, max.iter.cluster = 30 ,max.iter.harmony=50,theta=2,lambda=9)


#Cluster cells
Limb_all_int<- FindNeighbors(Limb_all_int, dims = 1:50, reduction = "harmony")
Limb_all_int <- FindClusters(Limb_all_int, resolution = 1)


#run UMAP
Limb_all_int = RunUMAP(Limb_all_int,dims = 1:50, reduction = "harmony", min.dist = 0.3)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Limb_0_5_11dpa_sc_Harmony_UMAP_PC50_res1.pdf",width=15,height=15)
DimPlot(object = Limb_all_int,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Limb_all_int,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3)
DimPlot(object = Limb_all_int,  shuffle= T,reduction = 'umap',label = T, group.by = "origin", pt.size = 3, cols = c("firebrick2","darkslategray","dodgerblue"))
FeaturePlot(Limb_all_int,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


#compare embryonic markers with blastema markers


gene_ids = c("AMEX60DD001640","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD021249","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD051448","AMEX60DD046291","AMEX60DD034944","AMEX60DD028372","AMEX60DD013332","AMEX60DD033213","AMEX60DD020503","AMEX60DDU001008266","AMEX60DD056488","AMEX60DD044865")
	
gene_names =c("SOX2","PRRX1","RSPO3","FGF8","TBX18","UNCX","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","PKDCC","ACAN","CNMD","COL2A1","HES3","EMX2","TBXT","TBX6","HES7","GREB1","WNT3A","CDX2","CDX4","FGF2")




gg_Fig <- FeaturePlot(Tail_all_int, pt.size = 0.3, features = gene_ids,order = T, cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T, min.cutoff = "q5", max.cutoff = "q95") & NoLegend()
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_0_1_2wpa_sc_Harmony_UMAP_feature_ManualMarker.pdf",width=13,height=10)
CombinePlots( gg_Fig )
dev.off()
	


####Plot some markers as histogram per in silico bulk

Tail_all_int@meta.data$time = Tail_all_int@meta.data$tissue
Tail_all_int@meta.data$time[Tail_all_int@meta.data$orig.ident == "Tail_col_1wpa"] = "blastema1wpa"
Tail_all_int@meta.data$time[Tail_all_int@meta.data$orig.ident == "Tail_col_2wpa"] = "blastema2wpa"


#remove col2a1 because of very high expression levels
gene_ids = c("AMEX60DD001640","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD021249","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD051448","AMEX60DD046291","AMEX60DD034944","AMEX60DD028372","AMEX60DD013332","AMEX60DD033213","AMEX60DD020503","AMEX60DDU001008266","AMEX60DD056488","AMEX60DD044865")
	
gene_names =c("SOX2","PRRX1","RSPO3","FGF8","TBX18","UNCX","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","PKDCC","ACAN","CNMD","HES3","EMX2","TBXT","TBX6","HES7","GREB1","WNT3A","CDX2","CDX4","FGF2")



plot.df = AverageExpression(
  Tail_all_int,
  features = c(gene_ids),
  return.seurat = FALSE,
#  slot = "scale.data",
  group.by = "time"
)

plot.df = melt(plot.df)

pdf("Tail_0_1_2wpa_sc_Harmony_Histogram_feature_ManualMarker.pdf",width=13,height=10)
ggplot(plot.df, aes(x = Var2 , y = value, fill = Var2)) + geom_col() + facet_wrap(.~ Var1, scales = "free")
dev.off()


```

###Hallmark genes

```{r}
#intersect with var genes


Tail_all_int <- FindVariableFeatures(Tail_all_int, nfeatures = 10000)

EMT.ha = read.delim("HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION.v2023.1.Hs.txt",sep = "\t",header = F)[,1]

anno.emt = vector()

for (i in 1:length(EMT.ha)) {
  anno.emt = rbind(anno.emt,anno[grep(pattern = paste(EMT.ha[i],"$",sep = ""),anno$V2),])
}

anno.emt = intersect(anno.emt$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.emt),
  name = 'EMT.score',
  ctrl = length(anno.mig)
)


Motility.GObp = read.delim("GObp_CellMotility.txt",sep = "\t",header = F)[,1]

anno.mot = vector()

for (i in 1:length(Motility.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.mot),
  name = 'Mot.score',
  ctrl = length(anno.mig)
)


AmeboidMig.GObp = read.delim("GObp_AmeboidalMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(AmeboidMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(AmeboidMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.mig),
  name = 'AmeboidMig.score',
  ctrl = length(anno.mig)
)



SmoothMuscleMig.GObp = read.delim("GObp_SmoothMuscleMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(SmoothMuscleMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(SmoothMuscleMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.mig),
  name = 'SmoothMuscleMig.score',
  ctrl = length(anno.mig)
)


PosFibMig.GObp = read.delim("GObp_PosFibMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(PosFibMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(PosFibMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.mig),
  name = 'PosFibMig.score',
  ctrl = length(anno.mig)
)


Motility.Actin.GObp = c("ACTR3","ARPC","ENAH","EPS8","JMY","NCKAP1L")

anno.mot = vector()

for (i in 1:length(Motility.Actin.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.Actin.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.mot),
  name = 'Mot.Act.score',
  ctrl = length(anno.mot)
)


Motility.Meso.GObp = c("APELA","EPB41L5","FGF8","LRP5","MESP1")

anno.mot = vector()

for (i in 1:length(Motility.Meso.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.Meso.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.mot),
  name = 'Mot.Meso.score',
  ctrl = length(anno.mot)
)


Wu.mig = read.delim("Wu_CellMigration.txt",sep = "\t",header = F)[,1]

anno.wu = vector()

for (i in 1:length(Wu.mig)) {
  anno.wu = rbind(anno.wu,anno[grep(pattern = paste(Wu.mig[i],"$",sep = ""),anno$V2),])
}

anno.wu = intersect(anno.wu$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.wu),
  name = 'Wu.mig.score',
  ctrl = length(anno.wu)
)




GO.ERK = read.delim("GObp_ERK.txt",sep = "\t",header = F)[,1]

anno.ERK = vector()

for (i in 1:length(GO.ERK)) {
  anno.ERK = rbind(anno.ERK,anno[grep(pattern = paste(GO.ERK[i],"$",sep = ""),anno$V2),])
}

anno.ERK = intersect(anno.ERK$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.ERK),
  name = 'GObp.ERK.score',
  ctrl = length(anno.ERK)
)

BioCart.ERK = read.delim("BioCart_ERK.txt",sep = "\t",header = F)[,1]

anno.ERK = vector()

for (i in 1:length(BioCart.ERK)) {
  anno.ERK = rbind(anno.ERK,anno[grep(pattern = paste(BioCart.ERK[i],"$",sep = ""),anno$V2),])
}

anno.ERK = intersect(anno.ERK$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.ERK),
  name = 'BioCart.ERK.score',
  ctrl = length(anno.ERK)
)

#cell proliferation

g2m.genes = cc.genes$g2m.genes
s.genes = cc.genes$s.genes

anno.g2m = vector()

for (i in 1:length(g2m.genes)) {
  anno.g2m = rbind(anno.g2m,anno[grep(pattern = paste(g2m.genes[i],"$",sep = ""),anno$V2),])
}

anno.g2m = intersect(anno.g2m$V1,VariableFeatures(Tail_all_int))

anno.s = vector()
for (i in 1:length(s.genes)) {
  anno.s = rbind(anno.s,anno[grep(pattern = paste(s.genes[i],"$",sep = ""),anno$V2),])
}

anno.s = intersect(anno.s$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- CellCycleScoring(Tail_all_int, s.features = anno.s, g2m.features = anno.g2m, set.ident = F)



GO.PosFibProl = read.delim("GObp_PosCellProl.txt",sep = "\t",header = F)[,1]

anno.prol = vector()

for (i in 1:length(GO.PosFibProl)) {
  anno.prol = rbind(anno.prol,anno[grep(pattern = paste(GO.PosFibProl[i],"$",sep = ""),anno$V2),])
}

anno.prol = intersect(anno.prol$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.prol),
  name = 'GObp.PosFibProl.score',
  ctrl = length(anno.prol)
)



GO.PosCellPopProl = read.delim("GObp_PosCellPopProl.txt",sep = "\t",header = F)[,1]

anno.prol = vector()

for (i in 1:length(GO.PosCellPopProl)) {
  anno.prol = rbind(anno.prol,anno[grep(pattern = paste(GO.PosCellPopProl[i],"$",sep = ""),anno$V2),])
}

anno.prol = intersect(anno.prol$V1,VariableFeatures(Tail_all_int))

Tail_all_int <- AddModuleScore(
  object = Tail_all_int,
  features = list(anno.prol),
  name = 'GObp.PosCellPopProl.score',
  ctrl = length(anno.prol)
)


pdf("Tail_0_1_2wpa_sc_Harmony_UMAP_FeatureMigScores.pdf",width=20,height=20)
FeaturePlot(Tail_all_int,raster = F,  pt.size = 1, features = c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1"), order = T, min.cutoff = "q5", cols = c("grey",rev(viridis_pal(option = "viridis")(12)))) & NoLegend() & NoAxes()
FeaturePlot(Tail_all_int,raster = F,  pt.size = 1, features = c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1"), order = F, cols = c("grey",rev(viridis_pal(option = "viridis")(12)))) & NoLegend()  & NoAxes()
dev.off()

```


###Pseudotime

```{r}

######Calc pseudo time 

harmony.embedding = as.data.frame(Embeddings(Tail_all_int, "harmony"),stringsAsFactors = F)

meta.data = FetchData(object = Tail_all_int, vars = c("celltype",'seurat_clusters', 'orig.ident','UMAP_1','UMAP_2',"tissue"))

cells_SP = rownames(meta.data)[meta.data$celltype %in% c("Asomitic Progenitors")]
cells_P = rownames(meta.data)[meta.data$celltype %in% c("Committed Progenitors")]
cells_M = rownames(meta.data)[meta.data$celltype %in% c("Myogenic lineage")]


#calculate pseudo time separate on SM and P + M

harmony.embedding.muscle = as.data.frame(harmony.embedding[c(cells_P,cells_M),])
harmony.embedding.SP = as.data.frame(harmony.embedding[cells_SP,])

library("destiny")

#Pcalculate pseudotime for muscle branch
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.muscle),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.muscle = dpt[random_root(dpt), ]

harmony.embedding.muscle$pt = as.numeric(pseudotime.muscle)
harmony.embedding.muscle = harmony.embedding.muscle[order(harmony.embedding.muscle$pt),]
harmony.embedding.muscle$rank = rev(1:nrow(harmony.embedding.muscle))

#SM branch

harmony.embedding.SP$pt = 0
harmony.embedding.SP$rank = 0

#merge
pseudotime = rbind(harmony.embedding.SP,harmony.embedding.muscle,stringsAsFactors = F)

plot.pt = cbind(coords,rank = pseudotime[match(rownames(coords),rownames(pseudotime)),"rank"], stringsAsFactors = F)


# cells
cellcex = 1
#color
colramp = colorRampPalette(c("darkgoldenrod3","violetred2"))
colgrad = colramp(101)


pdf("./Tail_0_1_2wpa_sc_pax_SPRING_PT_Rank.pdf",width=8,height=8)

  # normalize expression [0-1]
  exp = plot.pt[,"rank"]/max(plot.pt[,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt[,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$x, plot.pt$y, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime")

    edge_start_plot.pt  = plot.pt[match(edges$start, plot.pt$Cell), c("x","y")]
    edge_end_plot.pt  = plot.pt[match(edges$end, plot.pt$Cell), c("x","y")]
    segments(x0=edge_start_plot.pt[,"x"], y0=edge_start_plot.pt[,"y"],
    	x=edge_end_plot.pt[,"x"], y=edge_end_plot.pt[,"y"], lwd=linewidth, col=linecol)
		
    points(plot.pt$x, plot.pt$y, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="")

dev.off()


#Plot genes on top of SPRING embedding


tmp=c("AMEX60DD027495",
"AMEX60DD010148",
"AMEX60DD052476",
"AMEX60DD050893",
"AMEX60DD020388",
"AMEX60DD052322",
"AMEX60DD007644",
"AMEX60DD024484",
"AMEX60DD055040",
"AMEX60DD034577",
"AMEX60DD031139",
"AMEX60DD017092",
"AMEX60DD008545",
"AMEX60DD024851",
"AMEX60DD056342",
"AMEX60DD055382",
"AMEX60DD027986",
"AMEX60DD055570",
"AMEX60DD025304")

meta.data = FetchData(object = Tail_all_int_pax, vars = c('ident', 'orig.ident',"celltype",tmp))


tmp.name=c("UBE2C",
"TOP2A",
"CDK1",
"ENTPD2",
"NTN3-NTN1",
"PAX7",
"MYF5",
"AXL",
"NRP2",
"SMOC2",
"CACNG1",
"CDH15",
"MYOG",
"CKM",
"DES",
"MYL1",
"MYLPF",
"TTN",
"ACTC1")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))


colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 1
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

##########

pdf("./Tail_0_1_2wpa_sc_pax_SPRING_MyogenicLineageMarker.pdf",width=8,height=8)

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i])

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

}

dev.off()

#ntn3 seems to be earlier than pax7
#Find correlating genes

norm.data = GetAssayData(object = Tail_all_int_pax, slot = "data")
norm.data = as.matrix(norm.data)

gene<-as.numeric(norm.data["AMEX60DD020388",])
correlations<-apply(norm.data,1,function(x){cor(gene,x)})
correlations[is.na(correlations)] = 0

correlations.anno = as.data.frame(correlations)
correlations.anno$gene = rownames(correlations.anno)

correlations.anno = merge(correlations.anno,anno, by.x="gene" , by.y="V1")
correlations.anno$ID = correlations.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
correlations.anno = correlations.anno  %>% arrange(desc(correlations))

write.csv(correlations.anno,"Tail_0_1_2wpa_sc_pax_Harmony_CorrelationMarker.csv")

#Plot genes on top of SPRING embedding


tmp=c("AMEX60DD020388",
"AMEX60DD055040",
"AMEX60DD009987",
"AMEX60DD043119",
"AMEX60DD001909",
"AMEX60DD018225",
"AMEX60DD023964",
"AMEX60DD000336",
"AMEX60DD013919",
"AMEX60DD025227",
"AMEX60DD055954",
"AMEX60DD045083",
"AMEX60DD003483",
"AMEX60DD002950",
"AMEX60DD040573",
"AMEX60DD008914",
"AMEX60DD041795",
"AMEX60DD016542")

meta.data = FetchData(object = Tail_all_int_pax, vars = c('ident', 'orig.ident',"celltype",tmp))


tmp.name=c("NTN3-NTN1",
"NRP2",
"MEOX1*tf",
"DMRT2*tf",
"IGSF10",
"TNN",
"ASPN",
"MMP11",
"CILP2",
"EFEMP2",
"CXCR4",
"SFRP2*wnt",
"NR2F2*tf",
"SFRP1*wnt",
"KHDRBS3",
"WNT2B*wnt",
"TCF4*tf",
"NTN1")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))


colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 1
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

##########

pdf("./Tail_0_1_2wpa_sc_pax_SPRING_EarlyMyoCorrelationMarker.pdf",width=8,height=8)

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i])

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

}

dev.off()



```

###SPRING

```{r}

Int_col_BL_CT_int = readRDS(file = "Int_col_BL_CT_HarmonyTimeInt_final_SeuratObj.RDS")

###Prepare data

harmony.embedding = Embeddings(Tail_all_int, "harmony")


#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}

#use only cluster marker to run SPRING

markers <- FindAllMarkers(Int_col_BL_CT_int, only.pos = TRUE,  logfc.threshold = 0.5)
regeneration.marker = as.character(unique(markers$gene[ markers$avg_logFC > 0.5]))

#extract norm data from Seurat object

norm.data = GetAssayData(object = Tail_all_int, slot = "data")
norm.data = as.matrix(norm.data)

meta.data = FetchData(object = Tail_all_int, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA"))

##############################
#SPRING
##############################

### export for SPRING (export norma and not scaled data)

#cutoff 0.8
y=as.data.frame(t(meta.data[,c("orig.ident","ident")]),stringsAsFactors=F)

write.table(y,sep = ",",col.names = F,file = "SPRING_Export_Tail_0_1_2wpa_sc_meta.csv")

x = as.data.frame(t(harmony.embedding[,c(1:50)]))

write.table(x,sep = ",",col.names = F,file ="SPRING_Export_Tail_0_1_2wpa_sc_HarmonyMatrix.csv")

z=rownames(x)

write.table(z,col.names = F,row.names = F,sep = ",",file = "SPRING_Export_Tail_0_1_2wpa_sc_GeneList.csv")

###Run with : 50 PCs gene / 15 neighbors / gene variability percentile 20.0

#does not look promising... get pseudo time estimates onto UMAP
```


###Run Velocity

```{r}

setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")


Tail_all_int = readRDS(file = "Tail_0_1_2wpa_sc_Harmony_SeuratObj.RDS")

Tail_col_1wpa_spliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/Col1a2_tail_BL_10x/count_123456_114342_1wpa_col1a2_Full_Solo.out/Velocyto/raw/spliced")
Tail_col_1wpa_spliced = Tail_col_1wpa_spliced[rownames(anno), ,drop = FALSE]
Tail_col_1wpa_spliced <- CreateSeuratObject(counts = Tail_col_1wpa_spliced, project = "Tail_col_1wpa_spliced", min.cells = 1, min.features = 1)

Tail_col_1wpa_unspliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/Col1a2_tail_BL_10x/count_123456_114342_1wpa_col1a2_Full_Solo.out/Velocyto/raw/unspliced")
Tail_col_1wpa_unspliced = Tail_col_1wpa_unspliced[rownames(anno), ,drop = FALSE]
Tail_col_1wpa_unspliced <- CreateSeuratObject(counts = Tail_col_1wpa_unspliced, project = "Tail_col_1wpa_unspliced", min.cells = 1, min.features = 1)


Tail_col_2wpa_spliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/HLFYTDRXX_R9843/count_123456_2wpa_col1a2_Full_Solo.out/Velocyto/raw/spliced")
Tail_col_2wpa_spliced = Tail_col_2wpa_spliced[rownames(anno), ,drop = FALSE]
Tail_col_2wpa_spliced <- CreateSeuratObject(counts = Tail_col_2wpa_spliced, project = "Tail_col_2wpa_spliced", min.cells = 1, min.features = 1)

Tail_col_2wpa_unspliced <- Read10X(data.dir = "/mnt/SingleCellGenomics/scg_projects/10xGenomics/HLFYTDRXX_R9843/count_123456_2wpa_col1a2_Full_Solo.out/Velocyto/raw/unspliced")
Tail_col_2wpa_unspliced = Tail_col_2wpa_unspliced[rownames(anno), ,drop = FALSE]
Tail_col_2wpa_unspliced <- CreateSeuratObject(counts = Tail_col_2wpa_unspliced, project = "Tail_col_2wpa_unspliced", min.cells = 1, min.features = 1)



Int_col_BL_CT_spliced = merge(Tail_col_1wpa_spliced, y = Tail_col_2wpa_spliced, add.cell.ids = c("wpa1", "wpa2"), project = "Tail_col_Integration_onlyBL_spliced")

Int_col_BL_CT_unspliced = merge(Tail_col_1wpa_unspliced, y = Tail_col_2wpa_unspliced, add.cell.ids = c("wpa1", "wpa2"), project = "Tail_col_Integration_onlyBL_unspliced")


#subset

library(velocyto.R)

Int_col_BL_CT_spliced = subset(Int_col_BL_CT_spliced,cells = colnames(Int_col_BL_CT_int))
Idents(Int_col_BL_CT_spliced) = Int_col_BL_CT_meta@meta.data$seurat_clusters

#merge clusters to meta cells

Int_col_BL_CT_spliced_mean = AverageExpression(
  Int_col_BL_CT_spliced,
  return.seurat = T,
  group.by = "ident",
  slot = "counts")

Int_col_BL_CT_spliced_mean@meta.data$fraction_1wpa = Int_col_BL_CT_meta_mean$fraction_1wpa


Int_col_BL_CT_unspliced = subset(Int_col_BL_CT_unspliced,cells = colnames(Int_col_BL_CT_int))
Idents(Int_col_BL_CT_unspliced) = Int_col_BL_CT_meta@meta.data$seurat_clusters

#merge clusters to meta cells

Int_col_BL_CT_unspliced_mean = AverageExpression(
  Int_col_BL_CT_unspliced,
  return.seurat = T,
  group.by = "ident",
  slot = "counts")

Int_col_BL_CT_unspliced_mean@meta.data$fraction_1wpa = Int_col_BL_CT_meta_mean$fraction_1wpa

#Velocity 

emat <- GetAssayData(Int_col_BL_CT_spliced_mean, slot = "counts")
nmat <- GetAssayData(Int_col_BL_CT_unspliced_mean, slot = "counts")

# take cluster labels
cluster.label <- Int_col_BL_CT_meta_mean@meta.data$seurat_cluster
names(cluster.label) = colnames(Int_col_BL_CT_meta_mean)

gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

cell.colors = gg_color_hue(length(unique(Int_col_BL_CT_meta_mean@meta.data$seurat_cluster)))
#cell.colors <-  colorRampPalette(brewer.pal(12,"Set1"))(length(unique(BLct_CSDct_filter@meta.data$orig.ident)))

names(cell.colors) <- unique(cluster.label)

cell.cols <- cell.colors[cluster.label]
names(cell.cols) = colnames(Int_col_BL_CT_meta_mean)

# take embedding
emb <- Embeddings(Int_col_BL_CT_meta_mean, "umap")

# filter genes
emat <- filter.genes.by.cluster.expression(emat,cluster.label,min.max.cluster.average = 0.5)
nmat <- filter.genes.by.cluster.expression(nmat,cluster.label,min.max.cluster.average = 0.05)

length(intersect(rownames(emat),rownames(emat)))

#Estimate RNA velocity (using gene-relative model with k=20 cell kNN pooling and using top/bottom 2% quantiles for gamma fit):

fit.quantile <- 0.2
rvel.cd <- gene.relative.velocity.estimates(emat,nmat,deltaT=1,kCells=10,fit.quantile=fit.quantile)
#cell.dist=cell.dist,

#Visualize velocity on the t-SNE embedding, using velocity vector fields:
pdf("Int_col_onlyBL_CT_Meta_UMAP_Velocity.pdf",width=10,height=10)
show.velocity.on.embedding.cor(emb,rvel.cd,n=500,scale='sqrt',cell.colors=cell.cols,cex=0.8,arrow.scale=0.5,show.grid.flow=TRUE,min.grid.cell.mass=0.1,grid.n=50,do.par=F,n.cores=50,cell.border.alpha = 0)
dev.off()

#Zhisong example
#show.velocity.on.embedding.cor(spring.coor,rvel.cd,n=200,scale='sqrt',cell.colors=cell.cols,cex=2,arrow.scale=1000,show.grid.flow=TRUE,min.grid.cell.mass=0.1,grid.n=80,do.par=T,cell.border.alpha = 0.1,n.cores=25, arrow.lwd = 2, bty="n", xaxt="n", yaxt="n")

```



Run in bash
Check possible environments
conda info --envs

conda activate /home/tobias_gerber/.conda/envs/axoloom

python

Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scvelo as scv




```{r}
#make anndata object for scVelo


library("reticulate")
use_condaenv("/home/tobias_gerber/.conda/envs/axoloom", required = TRUE)
scv <- import("scvelo")

ad <- import("anndata", convert = FALSE)
dfobs <- data.frame(seurat_cluster = Int_col_BL_CT_meta_mean@meta.data$seurat_cluster,fraction_1wpa =  Int_col_BL_CT_meta_mean@meta.data$fraction_1wpa)
rownames(dfobs) <- colnames(Int_col_BL_CT_meta_mean)
#dfvar <- data.frame(genes_attributes)
#rownames(dfvar) <- genes

#bring all matrices to the same gene number

subsample = unique(intersect(intersect(intersect(rownames(Int_col_BL_CT_spliced_mean),rownames(Int_col_BL_CT_meta_mean)),intersect(rownames(Int_col_BL_CT_unspliced_mean),rownames(Int_col_BL_CT_meta_mean))),intersect(rownames(Int_col_BL_CT_spliced_mean),rownames(Int_col_BL_CT_unspliced_mean))))

subsample_mat <- subset(Int_col_BL_CT_meta_mean, features = subsample)
subsample_mat = GetAssayData(subsample_mat, slot = "counts")

subsample_emat <- subset(Int_col_BL_CT_spliced_mean, features = subsample)
subsample_emat = GetAssayData(subsample_emat, slot = "counts")

subsample_nmat <- subset(Int_col_BL_CT_unspliced_mean, features = subsample)
subsample_nmat = GetAssayData(subsample_nmat, slot = "counts")

adata <- ad$AnnData(
  X=t(subsample_mat),
  obs=dfobs,
 # var=dfvar,
  layers=list('spliced'=t(subsample_emat), 'unspliced'=t(subsample_nmat)),
  obsm=list('X_umap'=emb, 'X_pca'=Embeddings(Int_col_BL_CT_meta_mean, "pca")) 
  )

## run scvelo dynamic model
scv$pp$filter_genes(adata) ## filter
scv$pp$moments(adata,n_pcs=100,n_neighbors=as.integer(10)) ## normalize and compute moments
scv$tl$recover_dynamics(adata) ## model

## takes awhile, so uncomment to save
#adata$write('data/pancreas.h5ad', compression='gzip')
#adata = scv$read('data/pancreas.h5ad')

## plot (creates pop up window)

scv$tl$velocity(adata, mode='dynamical')
scv$tl$velocity_graph(adata)
scv$pl$velocity_embedding_stream(adata,layer=c('velocity'),color=c('seurat_cluster'), basis='umap',dpi = 300, save = 'Int_col_onlyBL_CT_Meta_UMAP_scVelo.svg')



```

###Add Tail bud

```{r}
Tail_LB_int = readRDS(file = "../EmbryonicTailData_SS2/Tail_LB_int_clean_CPM_SeuratObj.RDS")

Tail_all_LB = merge(Tail_all_int,Tail_LB_int)

Tail_all_LB@meta.data$timepoint = Tail_all_LB@meta.data$Stage
Tail_all_LB@meta.data$timepoint[Tail_all_LB@meta.data$tissue %in% "uninjured"] = "BL_0wpa"
Tail_all_LB@meta.data$timepoint[Tail_all_LB@meta.data$orig.ident %in% "Tail_col_1wpa"] = "BL_1wpa"
Tail_all_LB@meta.data$timepoint[Tail_all_LB@meta.data$orig.ident %in% "Tail_col_2wpa"] = "BL_2wpa"


Tail_all_LB@meta.data$timepoint <- factor(Tail_all_LB@meta.data$timepoint, levels = c("25","28","30","35","BL_0wpa","BL_1wpa","BL_2wpa"))


#somitogenesis
gene_ids = c("AMEX60DD021181",
"AMEX60DD013332",
"AMEX60DD034517",
"AMEX60DD050689",
"AMEX60DD010485",
"AMEX60DD050943")
gene_names = c("LFNG",
"HES7",
"DLL1",
"NOTCH1",
"NOTCH1like",
"NRARP")


pdf("Tail_BL_LB_sc_Harmony_Dot_SomitogenesisMarker.pdf",width=5,height=7)
DotPlot(Tail_all_LB, features = gene_ids, group.by = "timepoint") + scale_size(range = c(1,8)) + RotatedAxis() + coord_flip() + theme(axis.text.x=element_text(size=7),axis.text.y=element_text(size=7)) + scale_x_discrete(labels=c(gene_names)) + scale_colour_gradientn(colors = c(rev(viridis_pal(option = "viridis")(12))))
dev.off()






gene_ids = c("AMEX60DD021249",
"AMEX60DD033766",
"AMEX60DD034944",
"AMEX60DD028372",
"AMEX60DD037807",
"AMEX60DD029640",
"AMEX60DD047214",
"AMEX60DD054124",
"AMEX60DD004140",
"AMEX60DD004139",
"AMEX60DD035450",
"AMEX60DD014003",
"AMEX60DD011133",
"AMEX60DD024449",
"AMEX60DD021621",
"AMEX60DD009987")

gene_names = c("UNCX",
"TBX18",
"TBXT",
"TBX6",
"RIPPLY1",
"RIPPLY2",
"RIPPLY3",
"LHX1",
"MESP1",
"MESP2",
"FOXA2",
"GDF3",
"OTX2",
"OTX2like",
"SHH",
"MEOX1")


pdf("Tail_BL_LB_sc_Harmony_Dot_ResegmentationMarker.pdf",width=5,height=7)
DotPlot(Tail_all_LB, features = gene_ids, group.by = "timepoint") + scale_size(range = c(1,8)) + RotatedAxis() + coord_flip() + theme(axis.text.x=element_text(size=7),axis.text.y=element_text(size=7)) + scale_x_discrete(labels=c(gene_names)) + scale_colour_gradientn(colors = c(rev(viridis_pal(option = "viridis")(12))))
dev.off()


#Plot markers grouped by celltype to show that asomitic cells have it



#somitogenesis
gene_ids = c("AMEX60DD021181",
"AMEX60DD013332",
"AMEX60DD034517",
"AMEX60DD050689",
"AMEX60DD010485",
"AMEX60DD050943")
gene_names = c("LFNG",
"HES7",
"DLL1",
"NOTCH1",
"NOTCH1like",
"NRARP")


pdf("Tail_BL_sc_Harmony_Vln_SomitogenesisMarker.pdf",width=10,height=10)
VlnPlot(Tail_all_int, pt.size = 0, features = gene_ids, group.by = "celltype") 
dev.off()

#+ scale_colour_gradientn(colors = c(rev(viridis_pal(option = "viridis")(12))))

#visualize average values

av.Tail_all_int <- as.matrix(AverageExpression(Tail_all_int, group.by = "celltype")$RNA)
av.Tail_all_int = av.Tail_all_int[intersect(gene_ids,rownames(av.Tail_all_int)),]
av.Tail_all_int = melt(av.Tail_all_int)


pdf("Tail_BL_sc_Harmony_Bar_SomitogenesisMarker.pdf",width=15,height=10)
ggplot(av.Tail_all_int,aes(x = Var2, y = value)) + geom_col()  + theme_bw() + facet_wrap(.~ Var1, scales= "free")
dev.off()



gene_ids = c("AMEX60DD021249",
"AMEX60DD033766",
"AMEX60DD034944",
"AMEX60DD028372",
"AMEX60DD037807",
"AMEX60DD029640",
"AMEX60DD047214",
"AMEX60DD054124",
"AMEX60DD004140",
"AMEX60DD004139",
"AMEX60DD035450",
"AMEX60DD014003",
"AMEX60DD000572",
"AMEX60DD011133",
"AMEX60DD024449",
"AMEX60DD021621",
"AMEX60DD009987")

gene_names = c("UNCX",
"TBX18",
"TBXT",
"TBX6",
"RIPPLY1",
"RIPPLY2",
"RIPPLY3",
"LHX1",
"MESP1",
"MESP2",
"FOXA2",
"GDF3",
"TMED2",
"OTX2",
"OTX2like",
"SHH",
"MEOX1")


pdf("Tail_BL_sc_Harmony_Vln_ResegmentationMarker.pdf",width=15,height=20)
VlnPlot(Tail_all_int, pt.size = 0, features = gene_ids, group.by = "celltype") 
dev.off()


#visualize average values

av.Tail_all_int <- as.matrix(AverageExpression(Tail_all_int, group.by = "celltype")$RNA)
av.Tail_all_int = av.Tail_all_int[intersect(gene_ids,rownames(av.Tail_all_int)),]
av.Tail_all_int = melt(av.Tail_all_int)


pdf("Tail_BL_sc_Harmony_Bar_ResegmentationMarker.pdf",width=15,height=10)
ggplot(av.Tail_all_int,aes(x = Var2, y = value)) + geom_col()  + theme_bw() + facet_wrap(.~ Var1, scales= "free")
dev.off()

#+ ylim(0,0.5) + geom_hline(yintercept=mean(plot.df$Cor_mean), linetype="dashed", color = "blue")


#Dotplot with all togetger


gene_ids = c("AMEX60DD021181",
"AMEX60DD013332",
"AMEX60DD034517",
"AMEX60DD050689",
"AMEX60DD010485",
"AMEX60DD050943",

"AMEX60DD021249",
"AMEX60DD033766",
"AMEX60DD034944",
"AMEX60DD028372",
"AMEX60DD037807",
"AMEX60DD029640",
"AMEX60DD047214",
"AMEX60DD054124",
"AMEX60DD004140",
"AMEX60DD004139",
"AMEX60DD035450",
"AMEX60DD014003",
"AMEX60DD000572",
"AMEX60DD011133",
"AMEX60DD024449",
"AMEX60DD021621",
"AMEX60DD009987")

gene_names = c("LFNG",
"HES7",
"DLL1",
"NOTCH1",
"NOTCH1like",
"NRARP",

"UNCX",
"TBX18",
"TBXT",
"TBX6",
"RIPPLY1",
"RIPPLY2",
"RIPPLY3",
"LHX1",
"MESP1",
"MESP2",
"FOXA2",
"GDF3",
"TMED2",
"OTX2",
"OTX2like",
"SHH",
"MEOX1")


pdf("Tail_BL_LB_sc_Harmony_Dot_AllMarker.pdf",width=5,height=7)
DotPlot(Tail_all_LB, features = gene_ids, group.by = "timepoint") + scale_size(range = c(1,8)) + RotatedAxis() + coord_flip() + theme(axis.text.x=element_text(size=7),axis.text.y=element_text(size=7)) + scale_x_discrete(labels=c(gene_names)) + scale_colour_gradientn(colors = c(rev(viridis_pal(option = "viridis")(12))))
dev.off()

```


##Extract Pax7 lineage based on combined data set clusters

```{r}
setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")
Tail_all_int = readRDS(file = "Tail_0_1_2wpa_sc_Harmony_SeuratObj.RDS")


#subset asomitic cells and muscle lineage cells

Tail_1_2wpa_pax = subset(Tail_all_int,subset = celltype %in% c("Myogenic lineage","Asomitic cells"))
Tail_1_2wpa_pax = subset(Tail_1_2wpa_pax,subset = tissue == "blastema")


#Cluster cells
Tail_1_2wpa_pax<- FindNeighbors(Tail_1_2wpa_pax, dims = 1:100, reduction = "harmony")
Tail_1_2wpa_pax <- FindClusters(Tail_1_2wpa_pax, resolution = 2)


#run UMAP
Tail_1_2wpa_pax = RunUMAP(Tail_1_2wpa_pax,dims = 1:100, reduction = "harmony", min.dist = 0.3)
#,n.components = 3L

#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Tail_1_2wpa_pax_full_Harmony_UMAP_PC100_res2.pdf",width=20,height=20)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3, cols = c("deepskyblue1","dodgerblue","firebrick3","firebrick4"))
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', group.by = "celltype", pt.size = 3, cols = c("turquoise1","turquoise3","plum4","turquoise4","indianred","coral2","violetred2","darkgoldenrod3"))
FeaturePlot(Tail_1_2wpa_pax,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()

#look for markers to clean out progenitors that are on the way to other lineages


gene_ids = c("AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD023726","AMEX60DD054625","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD022355","AMEX60DD009984","AMEX60DD009989","AMEX60DD009968")
	
gene_names =c("PRRX1","RSPO3","FGF8","WNT5A","TBX2","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","DLX5","SOST","ETV4","PHOSPHO1")

gg_Fig <- FeaturePlot(Tail_1_2wpa_pax, pt.size = 2, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T) & NoLegend()
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_1_2wpa_pax_full_Harmony_feature_SpecificMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()




gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	

gg_Fig <- FeaturePlot(Tail_1_2wpa_pax, pt.size = 2, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T) & NoLegend()
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_1_2wpa_pax_full_Harmony_feature_GeneralMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()

#remove mesenchyme and cartilage lineage cells and intermediate progenitors

Tail_1_2wpa_pax = subset(Tail_1_2wpa_pax,idents = c(1,6,10),invert = T)

#Cluster cells
Tail_1_2wpa_pax<- FindNeighbors(Tail_1_2wpa_pax, dims = 1:20, reduction = "harmony")
Tail_1_2wpa_pax <- FindClusters(Tail_1_2wpa_pax, resolution = 2)


#run UMAP
Tail_1_2wpa_pax = RunUMAP(Tail_1_2wpa_pax,dims = 1:20, reduction = "harmony", min.dist = 0.01)



pdf("Tail_1_2wpa_pax_full_Harmony_UMAP_PC20_res2.pdf",width=20,height=20)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3, cols = c("deepskyblue1","dodgerblue","firebrick3","firebrick4"))
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', group.by = "celltype", pt.size = 3, cols = c("turquoise1","turquoise3","plum4","turquoise4","indianred","coral2","violetred2","darkgoldenrod3"))
FeaturePlot(Tail_1_2wpa_pax,raster = T,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA","G2M.Score"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()



gene_ids = c("AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD023726","AMEX60DD054625","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD052322","AMEX60DD007644","AMEX60DD055512","AMEX60DD029436","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD022355","AMEX60DD009984","AMEX60DD009989","AMEX60DD009968")
	
gene_names =c("PRRX1","RSPO3","FGF8","WNT5A","TBX2","TCF15","MEOX1","MEOX2","SCX","TNMD","PAX7","MYF5","TWIST2","TWIST3","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","DLX5","SOST","ETV4","PHOSPHO1")

gg_Fig <- FeaturePlot(Tail_1_2wpa_pax, pt.size = 2, features = gene_ids,order = T, raster = T,cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T) & NoLegend()
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_1_2wpa_pax_full_Harmony_feature_SpecificMarker.pdf",width=20,height=17)
CombinePlots( gg_Fig )
dev.off()




gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	

gg_Fig <- FeaturePlot(Tail_1_2wpa_pax, pt.size = 2, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T) & NoLegend()
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_1_2wpa_pax_full_Harmony_feature_GeneralMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()


#markers

markers <- FindAllMarkers(Tail_1_2wpa_pax, only.pos = TRUE,  logfc.threshold = 0.3)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Tail_1_2wpa_pax_full_Harmony_ClusterMarker.csv")

saveRDS(Tail_1_2wpa_pax, file = "Tail_1_2wpa_pax_full_Harmony_SeuratObj.RDS")

```

###Hallmark genes

```{r}
#intersect with var genes


Tail_1_2wpa_pax <- FindVariableFeatures(Tail_1_2wpa_pax, nfeatures = 10000)

EMT.ha = read.delim("HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION.v2023.1.Hs.txt",sep = "\t",header = F)[,1]

anno.emt = vector()

for (i in 1:length(EMT.ha)) {
  anno.emt = rbind(anno.emt,anno[grep(pattern = paste(EMT.ha[i],"$",sep = ""),anno$V2),])
}

anno.emt = intersect(anno.emt$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.emt),
  name = 'EMT.score',
  ctrl = 10
)


Motility.GObp = read.delim("GObp_CellMotility.txt",sep = "\t",header = F)[,1]

anno.mot = vector()

for (i in 1:length(Motility.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.mot),
  name = 'Mot.score',
  ctrl = 10
)


AmeboidMig.GObp = read.delim("GObp_AmeboidalMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(AmeboidMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(AmeboidMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.mig),
  name = 'AmeboidMig.score',
  ctrl = 10
)



SmoothMuscleMig.GObp = read.delim("GObp_SmoothMuscleMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(SmoothMuscleMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(SmoothMuscleMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.mig),
  name = 'SmoothMuscleMig.score',
  ctrl = 10
)


PosFibMig.GObp = read.delim("GObp_PosFibMigration.txt",sep = "\t",header = F)[,1]

anno.mig = vector()

for (i in 1:length(PosFibMig.GObp)) {
  anno.mig = rbind(anno.mig,anno[grep(pattern = paste(PosFibMig.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mig = intersect(anno.mig$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.mig),
  name = 'PosFibMig.score',
  ctrl = 10
)


Motility.Actin.GObp = c("ACTR3","ARPC","ENAH","EPS8","JMY","NCKAP1L")

anno.mot = vector()

for (i in 1:length(Motility.Actin.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.Actin.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.mot),
  name = 'Mot.Act.score',
  ctrl = 10
)


Motility.Meso.GObp = c("APELA","EPB41L5","FGF8","LRP5","MESP1")

anno.mot = vector()

for (i in 1:length(Motility.Meso.GObp)) {
  anno.mot = rbind(anno.mot,anno[grep(pattern = paste(Motility.Meso.GObp[i],"$",sep = ""),anno$V2),])
}

anno.mot = intersect(anno.mot$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.mot),
  name = 'Mot.Meso.score',
  ctrl = 10
)


Wu.mig = read.delim("Wu_CellMigration.txt",sep = "\t",header = F)[,1]

anno.wu = vector()

for (i in 1:length(Wu.mig)) {
  anno.wu = rbind(anno.wu,anno[grep(pattern = paste(Wu.mig[i],"$",sep = ""),anno$V2),])
}

anno.wu = intersect(anno.wu$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.wu),
  name = 'Wu.mig.score',
  ctrl = 10
)




GO.ERK = read.delim("GObp_ERK.txt",sep = "\t",header = F)[,1]

anno.ERK = vector()

for (i in 1:length(GO.ERK)) {
  anno.ERK = rbind(anno.ERK,anno[grep(pattern = paste(GO.ERK[i],"$",sep = ""),anno$V2),])
}

anno.ERK = intersect(anno.ERK$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.ERK),
  name = 'GObp.ERK.score',
  ctrl = 10
)

BioCart.ERK = read.delim("BioCart_ERK.txt",sep = "\t",header = F)[,1]

anno.ERK = vector()

for (i in 1:length(BioCart.ERK)) {
  anno.ERK = rbind(anno.ERK,anno[grep(pattern = paste(BioCart.ERK[i],"$",sep = ""),anno$V2),])
}

anno.ERK = intersect(anno.ERK$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.ERK),
  name = 'BioCart.ERK.score',
  ctrl = 10
)

#cell proliferation

g2m.genes = cc.genes$g2m.genes
s.genes = cc.genes$s.genes

anno.g2m = vector()

for (i in 1:length(g2m.genes)) {
  anno.g2m = rbind(anno.g2m,anno[grep(pattern = paste(g2m.genes[i],"$",sep = ""),anno$V2),])
}

anno.g2m = intersect(anno.g2m$V1,VariableFeatures(Tail_1_2wpa_pax))

anno.s = vector()
for (i in 1:length(s.genes)) {
  anno.s = rbind(anno.s,anno[grep(pattern = paste(s.genes[i],"$",sep = ""),anno$V2),])
}

anno.s = intersect(anno.s$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- CellCycleScoring(Tail_1_2wpa_pax, s.features = anno.s, g2m.features = anno.g2m, set.ident = F)



GO.PosFibProl = read.delim("GObp_PosCellProl.txt",sep = "\t",header = F)[,1]

anno.prol = vector()

for (i in 1:length(GO.PosFibProl)) {
  anno.prol = rbind(anno.prol,anno[grep(pattern = paste(GO.PosFibProl[i],"$",sep = ""),anno$V2),])
}

anno.prol = intersect(anno.prol$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.prol),
  name = 'GObp.PosFibProl.score',
  ctrl = 10
)



GO.PosCellPopProl = read.delim("GObp_PosCellPopProl.txt",sep = "\t",header = F)[,1]

anno.prol = vector()

for (i in 1:length(GO.PosCellPopProl)) {
  anno.prol = rbind(anno.prol,anno[grep(pattern = paste(GO.PosCellPopProl[i],"$",sep = ""),anno$V2),])
}

anno.prol = intersect(anno.prol$V1,VariableFeatures(Tail_1_2wpa_pax))

Tail_1_2wpa_pax <- AddModuleScore(
  object = Tail_1_2wpa_pax,
  features = list(anno.prol),
  name = 'GObp.PosCellPopProl.score',
  ctrl = 10
)

pdf("Tail_1_2wpa_pax_full_Harmony_UMAP_PC20_res2.pdf",width=20,height=20)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', pt.size = 3,label = T)
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap',label = T, group.by = "orig.ident", pt.size = 3, cols = c("deepskyblue1","dodgerblue","firebrick3","firebrick4"))
DimPlot(object = Tail_1_2wpa_pax,  shuffle= T,reduction = 'umap', group.by = "celltype", pt.size = 3, cols = c("turquoise1","turquoise3","plum4","turquoise4","indianred","coral2","violetred2","darkgoldenrod3"))
FeaturePlot(Tail_1_2wpa_pax,raster = F,  pt.size = 1, features = c("nCount_RNA","percent.mt","nFeature_RNA"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
FeaturePlot(Tail_1_2wpa_pax,raster = F,  pt.size = 2, features = c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1"), order = T, cols = c("grey",rev(viridis_pal(option = "viridis")(12)))) & NoLegend()
dev.off()



```


###SPRING

```{r}

###Prepare data

harmony.embedding = Embeddings(Tail_1_2wpa_pax, "harmony")


#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}



#extract norm data from Seurat object

norm.data = GetAssayData(object = Tail_1_2wpa_pax, slot = "data")
norm.data = as.matrix(norm.data)


meta.data = FetchData(object = Tail_1_2wpa_pax, vars = c('ident', 'orig.ident','celltype',"nFeature_RNA","nCount_RNA"))

##############################
#SPRING
##############################

### export for SPRING (export norma and not scaled data)

#cutoff 0.8
y=as.data.frame(t(meta.data[,c("orig.ident","ident","celltype")]),stringsAsFactors=F)

write.table(y,sep = ",",col.names = F,file = "SPRING_Export_Tail_1_2wpa_pax_meta.csv")

x = as.data.frame(t(harmony.embedding[,c(1:50)]))

write.table(x,sep = ",",col.names = F,file ="SPRING_Export_Tail_1_2wpa_pax_HarmonyMatrix.csv")

z=rownames(x)

write.table(z,col.names = F,row.names = F,sep = ",",file = "SPRING_Export_Tail_1_2wpa_pax_GeneList.csv")

###Run with : 10 PCs gene / 5 neighbors 

#import SPRING tables

coords = read.csv("SPRING_Import_Tail_1_2wpa_pax_coordinates_May23.txt",header = F)
colnames(coords) = c("Cell","x","y")
edges = read.csv("SPRING_Import_Tail_1_2wpa_pax_edge_list_May23.txt",header = F)
colnames(edges) = c("start","end")

rownames(coords) = rownames(harmony.embedding)



### set some parameters
# cells
colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 2
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5


coords$celltype=Tail_1_2wpa_pax@meta.data$celltype

coords$color[coords$celltype=="Asomitic cells"]= "#E69F00"
coords$color[coords$celltype=="Myogenic lineage"]="#D55E00"
plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]

pdf("./Tail_1_2wpa_pax_full_SPRING_celltype.pdf",width=12,height=8)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()

coords$seurat_clusters=Tail_1_2wpa_pax@meta.data$seurat_clusters
coords$color[coords$seurat_clusters==0]= "#a6cee3"
coords$color[coords$seurat_clusters==1]= "#1f78b4"
coords$color[coords$seurat_clusters==2]="#b2df8a"
coords$color[coords$seurat_clusters==3]= "#33a02c"
coords$color[coords$seurat_clusters==4]="#fb9a99"
coords$color[coords$seurat_clusters==5]= "#e31a1c"
coords$color[coords$seurat_clusters==6]="#fdbf6f"
coords$color[coords$seurat_clusters==7]= "#ff7f00"
coords$color[coords$seurat_clusters==8]="#cab2d6"
coords$color[coords$seurat_clusters==9]="#6a3d9a"
coords$color[coords$seurat_clusters==10]= "#ffff99"
coords$color[coords$seurat_clusters==11]="#b15928"

plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]

pdf("./Tail_1_2wpa_pax_full_SPRING_cluster.pdf",width=12,height=8)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()


########## plot many genes at once

#set genes to plot - cluster markers



tmp = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD054625","AMEX60DD007033","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
names(tmp) =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","TBX2","FAM180A","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")
	

data.plot=as.data.frame(t(norm.data))
#data.plot = cbind(data.plot,FetchData(Tail_1_2wpa_pax,c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")))

# create 200 colors (colstart --> colend)
colramp = colorRampPalette(c("grey",rev(viridis_pal(option = "viridis")(12))))
colgrad = colramp(101)

data.plot = data.plot[match(rownames(data.plot),rownames(coords)),]
data.plot = cbind(coords,data.plot)
##########

pdf("./Tail_1_2wpa_pax_full_SPRING_Features.pdf",width=12,height=8 )

for (i in colnames(data.plot[,tmp])) {

  # normalize expression [0-1]
  exp = data.plot[,i]/max(data.plot[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  data.plot[,i] = round(exp, digits=2) *100 +1
  data.plot = data.plot[order(data.plot[,i]),]
  
  #plot gradient

    plot(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="", main = names(tmp)[tmp %in% i])

    edge_start_coords = data.plot[match(edges$start, data.plot$Cell), c("x","y")]
    edge_end_coords = data.plot[match(edges$end, data.plot$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="")

}

dev.off()


tmp = c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")
names(tmp) =c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")
	

data.plot=as.data.frame(t(norm.data))
data.plot = cbind(data.plot,FetchData(Tail_1_2wpa_pax,c("EMT.score1","Mot.Act.score1","Mot.Meso.score1","Wu.mig.score1","Mot.score1","AmeboidMig.score1","SmoothMuscleMig.score1","PosFibMig.score1","GObp.ERK.score1","BioCart.ERK.score1","G2M.Score","GObp.PosFibProl.score1","GObp.PosCellPopProl.score1")))

# create 200 colors (colstart --> colend)
colramp = colorRampPalette(c((viridis_pal(option = "magma")(12))))
colgrad = colramp(101)

data.plot = data.plot[match(rownames(data.plot),rownames(coords)),]
data.plot = cbind(coords,data.plot)
##########

pdf("./Tail_1_2wpa_pax_SPRING_Scores.pdf",width=15,height=8 )

for (i in colnames(data.plot[,tmp])) {

  # normalize expression [0-1]
  exp = data.plot[,i]/max(data.plot[,i])
  exp = exp+abs(min(exp))
  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  data.plot[,i] = round(exp, digits=2) *100 +1
  data.plot = data.plot[order(data.plot[,i]),]
  
  #plot gradient

    plot(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="", main = names(tmp)[tmp %in% i])

    edge_start_coords = data.plot[match(edges$start, data.plot$Cell), c("x","y")]
    edge_end_coords = data.plot[match(edges$end, data.plot$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(data.plot$x, data.plot$y, cex=cellcex, pch=16, col=colgrad[data.plot[,i]], xlab="", ylab="")

}

dev.off()

```

###Pseudotime

```{r}
Tail_1_2wpa_pax = readRDS(file = "Tail_1_2wpa_pax_full_Harmony_SeuratObj.RDS")

coords = read.csv("SPRING_Import_Tail_1_2wpa_pax_coordinates_May23.txt",header = F)
colnames(coords) = c("Cell","x","y")
edges = read.csv("SPRING_Import_Tail_1_2wpa_pax_edge_list_May23.txt",header = F)
colnames(edges) = c("start","end")

rownames(coords) = rownames(harmony.embedding)

coords$celltype=Tail_1_2wpa_pax@meta.data$celltype

######Calc pseudo time 

harmony.embedding = as.data.frame(Embeddings(Tail_1_2wpa_pax, "harmony"),stringsAsFactors = F)

meta.data = FetchData(object = Tail_1_2wpa_pax, vars = c("celltype",'ident', 'orig.ident','UMAP_1','UMAP_2'))

cells_myo = rownames(meta.data)[meta.data$celltype %in% c("Myogenic lineage")]
cells_aso = rownames(meta.data)[meta.data$celltype %in% c("Asomitic cells")]


#calculate pseudo time separate on aso and myo

harmony.embedding.aso = as.data.frame(harmony.embedding[c(cells_aso),])
harmony.embedding.myo = as.data.frame(harmony.embedding[c(cells_myo),])

library("destiny")

#Pcalculate pseudotime for muscle branch
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.aso),stringsAsFactors = F))

dpt = DPT(DatasetDM )

pseudotime.aso = dpt[random_root(dpt), ]

harmony.embedding.aso$pt = as.numeric(pseudotime.aso)
harmony.embedding.aso = harmony.embedding.aso[order(harmony.embedding.aso$pt),]
harmony.embedding.aso$rank = (1:nrow(harmony.embedding.aso))

#myo branch

DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.myo),stringsAsFactors = F))

dpt = DPT(DatasetDM, tips = c(1,2) )

pseudotime.myo = dpt[random_root(dpt), ]

harmony.embedding.myo$pt = as.numeric(pseudotime.myo)
harmony.embedding.myo = harmony.embedding.myo[order(harmony.embedding.myo$pt),]
harmony.embedding.myo$rank = rev(1:nrow(harmony.embedding.myo)) + nrow(harmony.embedding.aso)

#merge
pseudotime = rbind(harmony.embedding.aso,harmony.embedding.myo,stringsAsFactors = F)

plot.pt = cbind(coords,rank = pseudotime[match(rownames(coords),rownames(pseudotime)),"rank"], stringsAsFactors = F)


# cells
cellcex = 2
linecol = adjustcolor("gray10", alpha.f=0.4)
linewidth = 0.5
#color
colramp = colorRampPalette(c("darkgoldenrod3","violetred2","blue4"))
colgrad = colramp(101)


pdf("./Tail_1_2wpa_pax_full_SPRING_PT_Rank.pdf",width=12,height=8)

  # normalize expression [0-1]
  exp = plot.pt[,"rank"]/max(plot.pt[,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt[,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$x, plot.pt$y, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime")

    edge_start_plot.pt  = plot.pt[match(edges$start, plot.pt$Cell), c("x","y")]
    edge_end_plot.pt  = plot.pt[match(edges$end, plot.pt$Cell), c("x","y")]
    segments(x0=edge_start_plot.pt[,"x"], y0=edge_start_plot.pt[,"y"],
    	x=edge_end_plot.pt[,"x"], y=edge_end_plot.pt[,"y"], lwd=linewidth, col=linecol)
		
    points(plot.pt$x, plot.pt$y, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="")

dev.off()


#Plot genes on top of SPRING embedding


tmp=c("AMEX60DD027495",
"AMEX60DD010148",
"AMEX60DD052476",
"AMEX60DD050893",
"AMEX60DD020388",
"AMEX60DD052322",
"AMEX60DD007644",
"AMEX60DD024484",
"AMEX60DD055040",
"AMEX60DD034577",
"AMEX60DD031139",
"AMEX60DD017092",
"AMEX60DD008545",
"AMEX60DD024851",
"AMEX60DD056342",
"AMEX60DD055382",
"AMEX60DD027986",
"AMEX60DD055570",
"AMEX60DD025304")

meta.data = FetchData(object = Tail_all_int_pax, vars = c('ident', 'orig.ident',"celltype",tmp))


tmp.name=c("UBE2C",
"TOP2A",
"CDK1",
"ENTPD2",
"NTN3-NTN1",
"PAX7",
"MYF5",
"AXL",
"NRP2",
"SMOC2",
"CACNG1",
"CDH15",
"MYOG",
"CKM",
"DES",
"MYL1",
"MYLPF",
"TTN",
"ACTC1")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))


colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 1
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

##########

pdf("./Tail_0_1_2wpa_sc_pax_SPRING_MyogenicLineageMarker.pdf",width=8,height=8)

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i])

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

}

dev.off()

#ntn3 seems to be earlier than pax7
#Find correlating genes

norm.data = GetAssayData(object = Tail_all_int_pax, slot = "data")
norm.data = as.matrix(norm.data)

gene<-as.numeric(norm.data["AMEX60DD020388",])
correlations<-apply(norm.data,1,function(x){cor(gene,x)})
correlations[is.na(correlations)] = 0

correlations.anno = as.data.frame(correlations)
correlations.anno$gene = rownames(correlations.anno)

correlations.anno = merge(correlations.anno,anno, by.x="gene" , by.y="V1")
correlations.anno$ID = correlations.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
correlations.anno = correlations.anno  %>% arrange(desc(correlations))

write.csv(correlations.anno,"Tail_0_1_2wpa_sc_pax_Harmony_CorrelationMarker.csv")

#Plot genes on top of SPRING embedding


tmp=c("AMEX60DD020388",
"AMEX60DD055040",
"AMEX60DD009987",
"AMEX60DD043119",
"AMEX60DD001909",
"AMEX60DD018225",
"AMEX60DD023964",
"AMEX60DD000336",
"AMEX60DD013919",
"AMEX60DD025227",
"AMEX60DD055954",
"AMEX60DD045083",
"AMEX60DD003483",
"AMEX60DD002950",
"AMEX60DD040573",
"AMEX60DD008914",
"AMEX60DD041795",
"AMEX60DD016542")

meta.data = FetchData(object = Tail_all_int_pax, vars = c('ident', 'orig.ident',"celltype",tmp))


tmp.name=c("NTN3-NTN1",
"NRP2",
"MEOX1*tf",
"DMRT2*tf",
"IGSF10",
"TNN",
"ASPN",
"MMP11",
"CILP2",
"EFEMP2",
"CXCR4",
"SFRP2*wnt",
"NR2F2*tf",
"SFRP1*wnt",
"KHDRBS3",
"WNT2B*wnt",
"TCF4*tf",
"NTN1")
names(tmp.name) = tmp

#color.palette = c(colorRampPalette(brewer.pal(9,"Greys")[9:3])(n = 30),colorRampPalette(brewer.pal(9,"Reds")[3:9])(n = 69))

# create 200 colors (colstart --> colend)
colramp1 = colorRampPalette(brewer.pal(9,"Greys")[9:2])
colramp2 = colorRampPalette(brewer.pal(9,"Reds")[2:9])

colgrad = c(colramp1(50),colramp2(51))


colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 1
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

##########

pdf("./Tail_0_1_2wpa_sc_pax_SPRING_EarlyMyoCorrelationMarker.pdf",width=8,height=8)

for (i in colnames(meta.data[,tmp])) {

  # normalize expression [0-1]
  exp = meta.data[,i]/max(meta.data[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = tmp.name[i])

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

}

dev.off()



```

##Correlation Analysis
###0wpa - blastema

```{r}
Tail_0wpa_all = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_0wpa_CT_harm_SeuratObj.RDS")
#Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_Meta_SeuratObj.RDS")
Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")

#Correlate cluster of different objects
av.exp.bl <- AverageExpression(Tail_1_2wpa)$RNA
colnames(av.exp.bl) = paste("bl",colnames(av.exp.bl),sep = "_")
av.exp.0wpa <- AverageExpression(Tail_0wpa_all)$RNA
colnames(av.exp.0wpa) = paste("0wpa",colnames(av.exp.0wpa),sep = "_")

#get similar gene lists
tmp = unique(intersect(rownames(av.exp.bl),rownames(av.exp.0wpa)),intersect(rownames(av.exp.bl),rownames(av.exp.0wpa)))
av.exp.bl = av.exp.bl[tmp,]
av.exp.0wpa = av.exp.0wpa[tmp,]

shared_genes = Reduce(intersect, list(rownames(av.exp.bl),rownames(av.exp.0wpa)))

av.exp.bl = av.exp.bl[rownames(av.exp.bl) %in% shared_genes,]
av.exp.0wpa = av.exp.0wpa[rownames(av.exp.0wpa) %in% shared_genes,]

data.means.combined = cbind(av.exp.bl,av.exp.0wpa)


rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:12) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,13:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:12]

library(gplots)
library(RColorBrewer)
library(viridis)

col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

palette.breaks <- seq(min(cor.sc.bulk),max(cor.sc.bulk), 0.1)

hc <- hclust(as.dist(1-cor(cor.sc.bulk,method="spearman")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk),method="spearman")), method="ward")

pdf("COR_heatmap_BLmeta_0wpa.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="row")
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="col")
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none")
dev.off()


#ColSideColors=c("#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(colnames(data.means.combined))])



Tail_1_2wpa@meta.data$celltype = NULL

Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_clusters %in% c(10)] = "Asomitic cells"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_clusters %in% c(5)] = "Dermis"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_clusters %in% c(1,7)] = "Fin Mesenchyme"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_clusters %in% c(0,2,3,11,6,8)] = "Intermediate Progenitors"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_clusters %in% c(11,4)] = "Chondrocytes"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_clusters %in% c(9)] = "Myocytes"

pdf("Tail_1_2wpa_sc_UMAP_PC100_celltype.pdf",width=10,height=10)
DimPlot(object = Tail_1_2wpa, reduction = 'umap',label = T, pt.size = 2,shuffle = T, group.by = 'seurat_clusters')
DimPlot(object = Tail_1_2wpa, reduction = 'umap',label = T, pt.size = 2,shuffle = T, group.by = 'celltype')

dev.off()




Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(6)] = "Asomitic cells"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(1,2,9,4,5,10,12,13,14)] = "Dermal fibroblasts"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(0)] = "Fin Mesenchyme"
#Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(7)] = "Developing Mesenchyme"
#Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(8)] = "Developing Dermis"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(3,7,8)] = "Intermediate Progenitors"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(11)] = "Cartilage"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(15)] = "Periskeletal"


Tail_0wpa_all@meta.data$trans.line = "col1a2"
Tail_0wpa_all@meta.data$trans.line[Tail_0wpa_all@meta.data$orig.ident  %in% "Tail_twist_0wpa"] = "twist3"

pdf("Tail_0wpa_all_UMAP_PC100_celltype.pdf",width=10,height=10)

DimPlot(object = Tail_0wpa_all, reduction = 'umap',label = T, pt.size = 2,shuffle = T, group.by = 'celltype')
DimPlot(object = Tail_0wpa_all, reduction = 'umap',label = F, pt.size = 2,shuffle = T, group.by = 'trans.line')

dev.off()


#Correlate cell types of different objects (ignore myocytes (bl) and periskeletal (0wpa))

#bl
tmp = subset(Tail_1_2wpa, subset = celltype %in% c("Myocytes","Intermediate Progenitors"), invert = T)
Idents(tmp) = tmp@meta.data$celltype

#Find Markers
markers <- FindAllMarkers(tmp, only.pos = TRUE,  logfc.threshold = 0.5)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Tail_1_2wpa_celltype_noMyo_noProg_Marker.csv")

#average across coarse cell types 
av.exp.bl <- AverageExpression(tmp)$RNA
colnames(av.exp.bl) = paste("bl",colnames(av.exp.bl),sep = "_")


#0wpa

tmp = subset(Tail_0wpa_all,subset = celltype %in% c("Periskeletal","Intermediate Progenitors"), invert = T)
Idents(tmp) = tmp@meta.data$celltype

#Find Markers
markers <- FindAllMarkers(tmp, only.pos = TRUE,  logfc.threshold = 0.5)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Tail_0wpa_all_celltype_noProg_noPeri_Marker.csv")

#average across coarse cell types 


av.exp.0wpa <- AverageExpression(tmp)$RNA
colnames(av.exp.0wpa) = paste("0wpa",colnames(av.exp.0wpa),sep = "_")


#get similar gene lists
tmp = unique(intersect(rownames(av.exp.bl),rownames(av.exp.0wpa)),intersect(rownames(av.exp.bl),rownames(av.exp.0wpa)))
av.exp.bl = av.exp.bl[tmp,]
av.exp.0wpa = av.exp.0wpa[tmp,]

data.means.combined = cbind(av.exp.bl,av.exp.0wpa)


rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:5) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,6:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:5]

library(gplots)
library(RColorBrewer)
library(viridis)


#Do manual scaling over rows and then over cols

cor.sc.bulk.scale = cor.sc.bulk
#data.plot[,-1:-2][data.plot[,-1:-2]==0]<-NA
cor.sc.bulk.scale<-scale(cor.sc.bulk.scale)
#data.plot[data.plot>3]<-3
#data.plot[data.plot<(-3)]<-(-3)
#data.plot[is.na(data.plot)]<-(-3)
#cor.sc.bulk.scale = as.data.frame(cor.sc.bulk.scale)
cor.sc.bulk.scale = t(scale(t(cor.sc.bulk.scale)))
#cor.sc.bulk.scale[cor.sc.bulk.scale>2]<-2
#cor.sc.bulk.scale[cor.sc.bulk.scale<(-2)]<-(-2)

#col.regions=colorRampPalette(c(rep("black",1),brewer.pal(9,"Greys")[9:1],"white",brewer.pal(9,"Reds")[1:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)









col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

palette.breaks <- seq(min(cor.sc.bulk),max(cor.sc.bulk), 0.1)







hc <- hclust(as.dist(1-cor(cor.sc.bulk.scale,method="spearman")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk.scale),method="spearman")), method="ward")

pdf("COR_heatmap_BLmeta_0wpa_celltypes.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none")
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="row")
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="col")
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none")
dev.off()




#plot as heatmap to show how co-expression fades away in limb but not tail


markers.0dpa = read.csv("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Tail_0wpa_all_celltype_noProg_noPeri_Marker.csv")
markers.0dpa  = markers.0dpa %>% arrange(desc(avg_log2FC)) %>% group_by(cluster)  %>% slice(1:30)

markers.0dpa = as.data.frame(markers.0dpa)

tmp = subset(Tail_0wpa_all,subset = celltype %in% c("Periskeletal","Intermediate Progenitors"), invert = T)
Idents(tmp) = tmp@meta.data$celltype

### captures count of cells for the ident with the fewest
maxcells  <- min(table(Idents(tmp)))

pdf("Tail_0wpa_CT_Heatmap_0wpa_Marker.pdf",width=10,height=10)
DoHeatmap(
  subset(tmp, downsample = maxcells),
  features = markers.0dpa$gene,
  group.by = "celltype",
  group.bar = TRUE,
  group.colors = NULL,
  disp.min = -2.5,
  disp.max = NULL,
  slot = "scale.data",
  assay = NULL,
  label = TRUE,
  size = 5.5,
  hjust = 0,
  angle = 45,
  raster = TRUE,
  draw.lines = TRUE,
  lines.width = NULL,
  group.bar.height = 0.02,
  combine = TRUE
) + scale_fill_gradientn( colors = (colorRampPalette(colors = c("#a6611a","#dfc27d","#f5f5f5","#80cdc1","#018571")))(100))
#+ scale_fill_gradientn( colors = (c(viridis_pal(option = "inferno")(100))))
dev.off()

cor.mat = GetAssayData(tmp,slot = "data")
cor.mat = as.data.frame(t(as.matrix(cor.mat)),stringsAsFactors = F)
cor.mat = cor.mat[,markers.0dpa$gene]

cor.mat = cor(cor.mat)

#scale

cor.mat.scale = cor.mat
cor.mat.scale<-scale(cor.mat.scale)
cor.mat.scale[cor.mat.scale>2]<-2
cor.mat.scale[is.na(cor.mat.scale)]<-(-2)
cor.mat.scale[cor.mat.scale<(-2)]<-(-2)

palette.breaks <- seq(min(cor.mat),max(cor.mat), 0.1)

col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

hc <- hclust(as.dist(1-cor(cor.mat,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.mat),method="pearson")), method="ward")

pdf("Tail_0wpa_CT_Heatmap_Marker_Heatmap_GeneCor.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.mat.scale),trace="none",density="none",col=col.regions,Colv=F, dendrogram="none",Rowv=F, key=T,scale="none")
heatmap.celltypes<-heatmap.2(as.matrix(cor.mat),trace="none",density="none",col=col.regions,Colv=F, dendrogram="none",Rowv=F, key=T,scale="none")
dev.off()

#get correlation values within cell types

cor.mat.diag0 = cor.mat
diag(cor.mat.diag0) = NA

plot.df = as.data.frame(row.names = c("Asomitic","Cartilage","Dermal Fibroblasts","Fin Mesenchyme"), c(mean(cor.mat.diag0[1:30,1:30],na.rm = T), mean(cor.mat.diag0[31:60,31:60],na.rm = T), mean(cor.mat.diag0[61:90,61:90],na.rm = T),
mean(cor.mat.diag0[91:120,91:120],na.rm = T) ))
plot.df$Celltype = rownames(plot.df)
colnames(plot.df) = c("Cor_mean","Celltype")

pdf("Tail_0wpa_CT_Heatmap_Marker_Bar_GeneCor.pdf",width=15,height=10)
ggplot(plot.df,aes(x = Celltype, y = Cor_mean)) + geom_col(aes(fill = Celltype)) + ylim(0,0.5) + geom_hline(yintercept=mean(plot.df$Cor_mean), linetype="dashed", color = "blue") + theme_bw()
dev.off()


##### redo for blastema leaving out myocytes and progenitors

#only myo
#tmp = subset(Tail_1_2wpa,idents = c(0,1,2,3,4,5,6,7,9,10,11))

# myo and prog
tmp = subset(Tail_1_2wpa,idents = c(10,5,1,7,4,11))

### ensure ident to plot is active
Idents(tmp) <- "celltype"

### captures count of cells for the ident with the fewest
maxcells  <- min(table(Idents(tmp)))

pdf("Tail_1_2wpa_sc_Heatmap_0wpa_Marker.pdf",width=10,height=10)
DoHeatmap(
  subset(tmp, downsample = maxcells),
  features = markers.0dpa$gene,
  group.by = "celltype",
  group.bar = TRUE,
  group.colors = NULL,
  disp.min = -2.5,
  disp.max = NULL,
  slot = "scale.data",
  assay = NULL,
  label = TRUE,
  size = 5.5,
  hjust = 0,
  angle = 45,
  raster = TRUE,
  draw.lines = TRUE,
  lines.width = NULL,
  group.bar.height = 0.02,
  combine = TRUE
) + scale_fill_gradientn( colors = (colorRampPalette(colors = c("#a6611a","#dfc27d","#f5f5f5","#80cdc1","#018571")))(100))
dev.off()

cor.mat = GetAssayData(tmp,slot = "data")
cor.mat = as.data.frame(t(as.matrix(cor.mat)),stringsAsFactors = F)
cor.mat = cor.mat[,markers.0dpa$gene]

cor.mat = cor(cor.mat)

#scale

cor.mat.scale = cor.mat
cor.mat.scale<-scale(cor.mat.scale)
cor.mat.scale[cor.mat.scale>2]<-2
cor.mat.scale[is.na(cor.mat.scale)]<-(-2)
cor.mat.scale[cor.mat.scale<(-2)]<-(-2)

palette.breaks <- seq(min(cor.mat),max(cor.mat), 0.1)

col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

hc <- hclust(as.dist(1-cor(cor.mat,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.mat),method="pearson")), method="ward")

test = cor.mat.scale
test[lower.tri(test)] = 2.1

pdf("Tail_1_2wpa_sc_Heatmap_GeneCor.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.mat.scale),trace="none",density="none",col=col.regions,Colv=F, dendrogram="none",Rowv=F, key=T,scale="none")
heatmap.celltypes<-heatmap.2(as.matrix(cor.mat),trace="none",density="none",col=col.regions,Colv=F, dendrogram="none",Rowv=F, key=T,scale="none")
heatmap.celltypes<-heatmap.2(as.matrix(test),trace="none",density="none",col=col.regions,Colv=F, dendrogram="none",Rowv=F, key=T,scale="none")
dev.off()

#get correlation values within cell types

#get correlation values within cell types

cor.mat.diag0 = cor.mat
diag(cor.mat.diag0) = NA

plot.df = as.data.frame(row.names = c("Asomitic","Cartilage","Dermal Fibroblasts","Fin Mesenchyme"), c(mean(cor.mat.diag0[1:30,1:30],na.rm = T), mean(cor.mat.diag0[31:60,31:60],na.rm = T), mean(cor.mat.diag0[61:90,61:90],na.rm = T),
mean(cor.mat.diag0[91:120,91:120],na.rm = T) ))
plot.df$Celltype = rownames(plot.df)
colnames(plot.df) = c("Cor_mean","Celltype")


pdf("Tail_1_2wpa_sc_Bar_GeneCor.pdf",width=15,height=10)
ggplot(plot.df,aes(x = Celltype, y = Cor_mean)) + geom_col(aes(fill = Celltype)) + ylim(0,0.5) + geom_hline(yintercept=mean(plot.df$Cor_mean), linetype="dashed", color = "blue") + theme_bw()
dev.off()







cor.mat = GetAssayData(tmp,slot = "data")
cor.mat = as.data.frame(t(as.matrix(cor.mat)),stringsAsFactors = F)
cor.mat = cor.mat[colnames(cor.mat) %in% markers.0dpa$gene,]
cor.mat = cor.mat[rownames(cor.mat) %in% colnames(tmp)[tmp@meta.data$ident == "Cartilage"],]

cor.mat = cor(cor.mat)

```

###Embryonic data - 0wpa/blastema

```{r}
#against tail blastema

Tail_LB_int = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/EmbryonicTailData_SS2/Tail_LB_int_clean_CPM_SeuratObj.RDS")

tmp = subset(Tail_1_2wpa,idents = c(0,1,2,3,4,5,6,7,9,10,11))
Idents(tmp) = tmp@meta.data$celltype
av.exp.bl <- AverageExpression(tmp)$RNA
colnames(av.exp.bl) = paste("bl",colnames(av.exp.bl),sep = "_")

exp.embryo = as.data.frame(GetAssayData(Tail_LB_int[["RNA"]], slot = "data"))

#colnames(exp.embryo) = paste("em",colnames(exp.embryo),sep = "_")

#get similar gene lists
tmp = unique(intersect(rownames(av.exp.bl),rownames(exp.embryo)))

av.exp.bl = av.exp.bl[tmp,]
exp.embryo = exp.embryo[tmp,]

data.means.combined = cbind(av.exp.bl,exp.embryo)



rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:5) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,6:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:5]

library(gplots)
library(RColorBrewer)
library(viridis)


#Do manual scaling over rows and then over cols

cor.sc.bulk.scale = cor.sc.bulk
#data.plot[,-1:-2][data.plot[,-1:-2]==0]<-NA
cor.sc.bulk.scale<-scale(cor.sc.bulk.scale)
#data.plot[data.plot>3]<-3
#data.plot[data.plot<(-3)]<-(-3)
#data.plot[is.na(data.plot)]<-(-3)
#cor.sc.bulk.scale = as.data.frame(cor.sc.bulk.scale)
cor.sc.bulk.scale = t(scale(t(cor.sc.bulk.scale)))
cor.sc.bulk.scale[cor.sc.bulk.scale>2]<-2
cor.sc.bulk.scale[cor.sc.bulk.scale<(-2)]<-(-2)

#col.regions=colorRampPalette(c(rep("black",1),brewer.pal(9,"Greys")[9:1],"white",brewer.pal(9,"Reds")[1:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)


col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)







hc <- hclust(as.dist(1-cor(cor.sc.bulk.scale,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk.scale),method="pearson")), method="ward")

pdf("COR_heatmap_BLmeta_Embryo_celltypes.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@active.ident)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@meta.data$orig.ident)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@meta.data$Stage)])
#heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="row")
#heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="col",ColSideColors=c("dodgerblue","palegreen1","red3","purple","palegreen4")[as.factor(Tail_LB_int@active.ident)])
#heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none")
dev.off()

#against uninjured tail

Tail_LB_int = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/EmbryonicTailData_SS2/Tail_LB_int_clean_CPM_SeuratObj.RDS")

tmp = subset(Tail_0wpa_all,idents = c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,14))
Idents(tmp) = tmp@meta.data$celltype
av.exp.mat <- AverageExpression(tmp)$RNA
colnames(av.exp.mat) = paste("0wpa",colnames(av.exp.mat),sep = "_")

exp.embryo = as.data.frame(GetAssayData(Tail_LB_int[["RNA"]], slot = "data"))

#colnames(exp.embryo) = paste("em",colnames(exp.embryo),sep = "_")

#get similar gene lists
tmp = unique(intersect(rownames(av.exp.mat),rownames(exp.embryo)))

av.exp.mat = av.exp.mat[tmp,]
exp.embryo = exp.embryo[tmp,]

data.means.combined = cbind(av.exp.mat,exp.embryo)



rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:5) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,6:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:5]

library(gplots)
library(RColorBrewer)
library(viridis)


#Do manual scaling over rows and then over cols

cor.sc.bulk.scale = cor.sc.bulk
#data.plot[,-1:-2][data.plot[,-1:-2]==0]<-NA
cor.sc.bulk.scale<-scale(cor.sc.bulk.scale)
#data.plot[data.plot>3]<-3
#data.plot[data.plot<(-3)]<-(-3)
#data.plot[is.na(data.plot)]<-(-3)
#cor.sc.bulk.scale = as.data.frame(cor.sc.bulk.scale)
cor.sc.bulk.scale = t(scale(t(cor.sc.bulk.scale)))
cor.sc.bulk.scale[cor.sc.bulk.scale>2]<-2
cor.sc.bulk.scale[cor.sc.bulk.scale<(-2)]<-(-2)

#col.regions=colorRampPalette(c(rep("black",1),brewer.pal(9,"Greys")[9:1],"white",brewer.pal(9,"Reds")[1:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)


col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)







hc <- hclust(as.dist(1-cor(cor.sc.bulk.scale,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk.scale),method="pearson")), method="ward")

pdf("COR_heatmap_0wpa_Embryo_celltypes.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@active.ident)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@meta.data$orig.ident)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@meta.data$Stage)])
#heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="row")
#heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="col",ColSideColors=c("dodgerblue","palegreen1","red3","purple","palegreen4")[as.factor(Tail_LB_int@active.ident)])
#heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none")
dev.off()

#against all together


tmp = subset(Tail_1_2wpa,idents = c(0,1,2,3,4,5,6,7,9,10,11))
Idents(tmp) = tmp@meta.data$celltype
av.exp.bl <- AverageExpression(tmp)$RNA
colnames(av.exp.bl) = paste("bl",colnames(av.exp.bl),sep = "_")

tmp = subset(Tail_0wpa_all,idents = c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,14))
Idents(tmp) = tmp@meta.data$celltype
av.exp.mat <- AverageExpression(tmp)$RNA
colnames(av.exp.mat) = paste("0wpa",colnames(av.exp.mat),sep = "_")

exp.embryo = as.data.frame(GetAssayData(Tail_LB_int[["RNA"]], slot = "data"))

#colnames(exp.embryo) = paste("em",colnames(exp.embryo),sep = "_")

#get similar gene lists
tmp = unique(intersect(intersect(rownames(av.exp.mat),rownames(exp.embryo)),intersect(rownames(av.exp.bl),rownames(exp.embryo))))

av.exp.mat = av.exp.mat[tmp,]
av.exp.bl = av.exp.bl[tmp,]
exp.embryo = exp.embryo[tmp,]

data.means.combined = cbind(av.exp.mat,av.exp.bl,exp.embryo)



rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:10) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,11:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:10]

library(gplots)
library(RColorBrewer)
library(viridis)


#Do manual scaling over rows and then over cols

cor.sc.bulk.scale = cor.sc.bulk
#data.plot[,-1:-2][data.plot[,-1:-2]==0]<-NA
cor.sc.bulk.scale<-scale(cor.sc.bulk.scale)
#data.plot[data.plot>3]<-3
#data.plot[data.plot<(-3)]<-(-3)
#data.plot[is.na(data.plot)]<-(-3)
#cor.sc.bulk.scale = as.data.frame(cor.sc.bulk.scale)
cor.sc.bulk.scale = t(scale(t(cor.sc.bulk.scale)))
cor.sc.bulk.scale[cor.sc.bulk.scale>2]<-2
cor.sc.bulk.scale[cor.sc.bulk.scale<(-2)]<-(-2)

#col.regions=colorRampPalette(c(rep("black",1),brewer.pal(9,"Greys")[9:1],"white",brewer.pal(9,"Reds")[1:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)


col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)







hc <- hclust(as.dist(1-cor(cor.sc.bulk.scale,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk.scale),method="pearson")), method="ward")

pdf("COR_heatmap_all_Embryo_celltypes.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@active.ident)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@meta.data$orig.ident)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(Tail_LB_int@meta.data$Stage)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="row")
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="col",ColSideColors=c("dodgerblue","palegreen1","red3","purple","palegreen4")[as.factor(Tail_LB_int@active.ident)])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none")
dev.off()


#Score for blastema


#get list for scoring 

markers.bl = read.csv("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_Meta_CoarseCellTypeMarker.csv")
markers.bl  = markers.bl %>% arrange(desc(avg_log2FC)) %>% group_by(cluster)  %>% slice(1:30)


markers.bl = as.data.frame(markers.bl)

bl_cart <- list(markers.bl$gene[markers.bl$cluster == "Cartilage"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = bl_cart,
  name = 'bl_cart'
)
bl_psm <- list(markers.bl$gene[markers.bl$cluster == "PSMlike"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = bl_psm,
  name = 'bl_psm'
)
bl_mesen <- list(markers.bl$gene[markers.bl$cluster == "Mesenchyme"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = bl_mesen,
  name = 'bl_mesen'
)
bl_dermis <- list(markers.bl$gene[markers.bl$cluster == "Dermis"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = bl_dermis,
  name = 'bl_dermis'
)
bl_prog <- list(markers.bl$gene[markers.bl$cluster == "Progenitors"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = bl_prog,
  name = 'bl_prog'
)

pdf("Tail_Embryo_Harmony_BLscores.pdf",width=10,height=10)
FeaturePlot(Tail_LB_int,c("bl_cart1","bl_prog1","bl_dermis1","bl_psm1","bl_mesen1"), pt.size = 2, cols = c("grey",rev(viridis_pal(option = "viridis")(12))))
dev.off()


#Score for uninjured

#get list for scoring 

markers.0dpa = read.csv("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_0wpa_CoarseCellTypeMarker.csv")
markers.0dpa  = markers.0dpa %>% arrange(desc(avg_log2FC)) %>% group_by(cluster)  %>% slice(1:30)

markers.0dpa = as.data.frame(markers.0dpa)

naive_cart <- list(markers.0dpa$gene[markers.0dpa$cluster == "Cartilage"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = naive_cart,
  name = 'naive_cart'
)
naive_psm <- list(markers.0dpa$gene[markers.0dpa$cluster == "PSMlike"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = naive_psm,
  name = 'naive_psm'
)
naive_mesen <- list(markers.0dpa$gene[markers.0dpa$cluster == "Mesenchyme"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = naive_mesen,
  name = 'naive_mesen'
)
naive_dermis <- list(markers.0dpa$gene[markers.0dpa$cluster == "Dermis"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = naive_dermis,
  name = 'naive_dermis'
)
naive_prog <- list(markers.0dpa$gene[markers.0dpa$cluster == "Progenitors"])

Tail_LB_int <- AddModuleScore(
  object = Tail_LB_int,
  features = naive_prog,
  name = 'naive_prog'
)

pdf("Tail_Embryo_Harmony_UninjuredScores.pdf",width=10,height=10)
FeaturePlot(Tail_LB_int,c("naive_cart1","naive_prog1","naive_dermis1","naive_psm1","naive_mesen1"), pt.size = 2, cols = c("grey",rev(viridis_pal(option = "viridis")(12))))
dev.off()


```

###Limb blastema - blastema/0wpa



```{r}
#against tail blastema

Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_Meta_SeuratObj.RDS")


Tail_1_2wpa@meta.data$type = "tail_blastema"

Limb_BL = readRDS(file = "/mnt/SingleCellGenomics/scg_projects/Xenopus/Correlation_Axolotl_BL_LB/limb_ALL_bl_Harmony_SeuratObj.RDS")

tmp = subset(Tail_1_2wpa,idents = c(0,1,2,3,4,5,6,7,9,10,11))
Idents(tmp) = tmp@meta.data$celltype
av.exp.bl <- AverageExpression(tmp)$RNA
colnames(av.exp.bl) = paste("tail_bl",colnames(av.exp.bl),sep = "_")

tmp = Limb_BL
av.exp.bl.limb <- AverageExpression(tmp)$RNA
colnames(av.exp.bl.limb) = paste("limb_bl",colnames(av.exp.bl.limb),sep = "_")

#colnames(exp.embryo) = paste("em",colnames(exp.embryo),sep = "_")

#get similar gene lists
tmp = unique(intersect(rownames(av.exp.bl),rownames(av.exp.bl.limb)))

av.exp.bl = av.exp.bl[tmp,]
av.exp.bl.limb = av.exp.bl.limb[tmp,]

data.means.combined = cbind(av.exp.bl,av.exp.bl.limb)



rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:5) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,6:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:5]

library(gplots)
library(RColorBrewer)
library(viridis)


#Do manual scaling over rows and then over cols

cor.sc.bulk.scale = cor.sc.bulk
#data.plot[,-1:-2][data.plot[,-1:-2]==0]<-NA
cor.sc.bulk.scale<-scale(cor.sc.bulk.scale)
#data.plot[data.plot>3]<-3
#data.plot[data.plot<(-3)]<-(-3)
#data.plot[is.na(data.plot)]<-(-3)
#cor.sc.bulk.scale = as.data.frame(cor.sc.bulk.scale)
#cor.sc.bulk.scale = t(scale(t(cor.sc.bulk.scale)))
cor.sc.bulk.scale[cor.sc.bulk.scale>2]<-2
cor.sc.bulk.scale[cor.sc.bulk.scale<(-2)]<-(-2)

#col.regions=colorRampPalette(c(rep("black",1),brewer.pal(9,"Greys")[9:1],"white",brewer.pal(9,"Reds")[1:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)


col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)

hc <- hclust(as.dist(1-cor(cor.sc.bulk.scale,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk.scale),method="pearson")), method="ward")

pdf("COR_heatmap_BLmeta_BLlimb_celltypes.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod")[as.factor(colnames(av.exp.bl.limb))])
dev.off()



#against uninjured tail

Tail_0wpa_all = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_0wpa_CT_harm_SeuratObj.RDS")
Tail_0wpa_all@meta.data$type = "tail_0wpa"


tmp = subset(Tail_0wpa_all,idents = c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15))
Idents(tmp) = tmp@meta.data$celltype
av.exp.mat <- AverageExpression(tmp)$RNA
colnames(av.exp.mat) = paste("0wpa",colnames(av.exp.mat),sep = "_")


tmp = Limb_BL
av.exp.bl.limb <- AverageExpression(tmp)$RNA
colnames(av.exp.bl.limb) = paste("limb_bl",colnames(av.exp.bl.limb),sep = "_")

#colnames(exp.embryo) = paste("em",colnames(exp.embryo),sep = "_")

#get similar gene lists
tmp = unique(intersect(rownames(av.exp.mat),rownames(av.exp.bl.limb)))

av.exp.mat = av.exp.mat[tmp,]
av.exp.bl.limb = av.exp.bl.limb[tmp,]

data.means.combined = cbind(av.exp.mat,av.exp.bl.limb)




rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:6) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,7:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:6]

library(gplots)
library(RColorBrewer)
library(viridis)


#Do manual scaling over rows and then over cols

cor.sc.bulk.scale = cor.sc.bulk
#data.plot[,-1:-2][data.plot[,-1:-2]==0]<-NA
cor.sc.bulk.scale<-scale(cor.sc.bulk.scale)
#data.plot[data.plot>3]<-3
#data.plot[data.plot<(-3)]<-(-3)
#data.plot[is.na(data.plot)]<-(-3)
#cor.sc.bulk.scale = as.data.frame(cor.sc.bulk.scale)
#cor.sc.bulk.scale = t(scale(t(cor.sc.bulk.scale)))
cor.sc.bulk.scale[cor.sc.bulk.scale>2]<-2
cor.sc.bulk.scale[cor.sc.bulk.scale<(-2)]<-(-2)

#col.regions=colorRampPalette(c(rep("black",1),brewer.pal(9,"Greys")[9:1],"white",brewer.pal(9,"Reds")[1:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)


col.regions=colorRampPalette(viridis_pal(option = "turbo")(100))(100)

palette.breaks <- seq(min(cor.sc.bulk.scale),max(cor.sc.bulk.scale), 0.1)







hc <- hclust(as.dist(1-cor(cor.sc.bulk.scale,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk.scale),method="pearson")), method="ward")


pdf("COR_heatmap_Tail0wpa_BLlimb_celltypes.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk.scale),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="both",Rowv=as.dendrogram(hr), key=T,scale="none",ColSideColors=c("dodgerblue","palegreen1","red3","purple","goldenrod","yellow","forestgreen")[as.factor(colnames(av.exp.bl.limb))])
dev.off()



#against all together



#Score for blastema


#get list for scoring 

markers.bl = read.csv("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_Meta_CoarseCellTypeMarker.csv")
markers.bl  = markers.bl %>% arrange(desc(avg_log2FC)) %>% group_by(cluster)  %>% slice(1:30)


markers.bl = as.data.frame(markers.bl)

bl_cart <- list(markers.bl$gene[markers.bl$cluster == "Cartilage"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = bl_cart,
  name = 'bl_cart'
)
bl_psm <- list(markers.bl$gene[markers.bl$cluster == "PSMlike"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = bl_psm,
  name = 'bl_psm'
)
bl_mesen <- list(markers.bl$gene[markers.bl$cluster == "Mesenchyme"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = bl_mesen,
  name = 'bl_mesen'
)
bl_dermis <- list(markers.bl$gene[markers.bl$cluster == "Dermis"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = bl_dermis,
  name = 'bl_dermis'
)
bl_prog <- list(markers.bl$gene[markers.bl$cluster == "Progenitors"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = bl_prog,
  name = 'bl_prog'
)

pdf("LimbBL_Harmony_UMAP_TailBLscores.pdf",width=10,height=10)
FeaturePlot(Limb_BL,c("bl_cart1","bl_prog1","bl_dermis1","bl_psm1","bl_mesen1"),order = T, pt.size = 2, cols = c(rev(viridis_pal(option = "plasma")(12))))
FeaturePlot(Limb_BL,c("bl_cart1","bl_prog1","bl_dermis1","bl_psm1","bl_mesen1"),order = T, pt.size = 2)
dev.off()


#Score for uninjured

#get list for scoring 

markers.0dpa = read.csv("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_0wpa_CoarseCellTypeMarker.csv")
markers.0dpa  = markers.0dpa %>% arrange(desc(avg_log2FC)) %>% group_by(cluster)  %>% slice(1:30)


markers.0dpa = as.data.frame(markers.0dpa)

#score

naive_cart <- list(markers.0dpa$gene[markers.0dpa$cluster == "Cartilage"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = naive_cart,
  name = 'naive_cart'
)
naive_psm <- list(markers.0dpa$gene[markers.0dpa$cluster == "PSMlike"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = naive_psm,
  name = 'naive_psm'
)
naive_mesen <- list(markers.0dpa$gene[markers.0dpa$cluster == "Mesenchyme"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = naive_mesen,
  name = 'naive_mesen'
)
naive_dermis <- list(markers.0dpa$gene[markers.0dpa$cluster == "Dermis"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = naive_dermis,
  name = 'naive_dermis'
)
naive_prog <- list(markers.0dpa$gene[markers.0dpa$cluster == "Progenitors"])

Limb_BL <- AddModuleScore(
  object = Limb_BL,
  features = naive_prog,
  name = 'naive_prog'
)

pdf("LimbBL_Harmony_UMAP_TailUninjuredScores.pdf",width=10,height=10)
FeaturePlot(Limb_BL,c("naive_cart1","naive_prog1","naive_dermis1","naive_psm1","naive_mesen1"), order = T, pt.size = 2, cols = c(rev(viridis_pal(option = "plasma")(12))))
FeaturePlot(Limb_BL,c("naive_cart1","naive_prog1","naive_dermis1","naive_psm1","naive_mesen1"), order = T, pt.size = 2)
dev.off()


```


##QuadPro/Correlation 

###Prepare Data

```{r}
setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

#load data sets

Tail_0wpa_all = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_0wpa_CT_harm_SeuratObj.RDS")
Tail_1_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/Int_col_onlyBL_CT_HarmonyTimeInt_SeuratObj.RDS")
Tail_LB_int = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/EmbryonicTailData_SS2/Tail_LB_int_clean_CPM_SeuratObj.RDS")
Limb_BL = readRDS(file = "/mnt/SingleCellGenomics/scg_projects/Xenopus/Correlation_Axolotl_BL_LB/limb_ALL_bl_Harmony_SeuratObj.RDS")

#Import limb BL data

norm.data.bl = GetAssayData(object = Limb_BL, slot = "data")
norm.data.bl = as.matrix(norm.data.bl)
norm.data.bl = as.data.frame(t(norm.data.bl),stringsAsFactors=F)

meta.data.bl = FetchData(object = Limb_BL, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA"))
meta.data.bl$origin = "Limb_bl"
meta.data.bl$time = FetchData(object = Limb_BL, vars = "origin")[,1]



#Import tail LB data

norm.data.lb = GetAssayData(object = Tail_LB_int, slot = "data")
norm.data.lb = as.matrix(norm.data.lb)
norm.data.lb = as.data.frame(t(norm.data.lb),stringsAsFactors=F)


meta.data.lb = FetchData(object = Tail_LB_int, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA","Stage"))
meta.data.lb$origin = "Tail_em"
meta.data.lb$time = meta.data.lb$Stage
meta.data.lb$time = paste("em",meta.data.lb$time,sep = "_")
meta.data.lb$Stage = NULL

#Import tail BL data

norm.data.tail.bl = GetAssayData(object = Tail_1_2wpa, slot = "data")
norm.data.tail.bl = as.matrix(norm.data.tail.bl)
norm.data.tail.bl = as.data.frame(t(norm.data.tail.bl),stringsAsFactors=F)


meta.data.tail.bl = FetchData(object = Tail_1_2wpa, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA"))
meta.data.tail.bl$origin = "Tail_bl"
meta.data.tail.bl$time  = meta.data.tail.bl$orig.ident 

#Import tail uncut data

norm.data.tail.un = GetAssayData(object = Tail_0wpa_all, slot = "data")
norm.data.tail.un = as.matrix(norm.data.tail.un)
norm.data.tail.un = as.data.frame(t(norm.data.tail.un),stringsAsFactors=F)


meta.data.tail.un = FetchData(object = Tail_0wpa_all, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA"))
meta.data.tail.un$origin = "Tail_un"
meta.data.tail.un$time = "Tail_0wpa"

#combine

data.bl = cbind(meta.data.bl[,c("nCount_RNA","nFeature_RNA","ident","orig.ident","origin","time")], norm.data.bl, stringsAsFactors = F)
data.lb = cbind(meta.data.lb[,c("nCount_RNA","nFeature_RNA","ident","orig.ident","origin","time")], norm.data.lb, stringsAsFactors = F)
data.tail.bl = cbind(meta.data.tail.bl[,c("nCount_RNA","nFeature_RNA","ident","orig.ident","origin","time")], norm.data.tail.bl, stringsAsFactors = F)
data.tail.un = cbind(meta.data.tail.un[,c("nCount_RNA","nFeature_RNA","ident","orig.ident","origin","time")], norm.data.tail.un, stringsAsFactors = F)


#get the same genes for all the data frames

shared_genes = Reduce(intersect, list(colnames(data.tail.bl),colnames(data.tail.un),colnames(data.lb),colnames(data.bl)))

data.bl = data.bl[,colnames(data.bl) %in% shared_genes]
data.lb = data.lb[,colnames(data.lb) %in% shared_genes]
data.tail.bl = data.tail.bl[,colnames(data.tail.bl) %in% shared_genes]
data.tail.un = data.tail.un[,colnames(data.tail.un) %in% shared_genes]

#commbine all
data = rbind(data.bl, data.lb, stringsAsFactors = F)
data = rbind(data, data.tail.bl, stringsAsFactors = F)
data = rbind(data, data.tail.un, stringsAsFactors = F)

data.cor = data[,6:ncol(data)]

data.means <-as.data.frame(do.call("rbind", as.list(by(data.cor[,2:ncol(data.cor)], as.factor(data.cor$time), colMeans))))

data.means<-as.data.frame(t(data.means),stringsAsFactors = F)

data.means.combined <- cbind(data.means, as.data.frame( t (data.cor[,2:ncol(data.cor)]) ,stringsAsFactors = F), stringsAsFactors=F)

saveRDS(data.means.combined, "PseudobulkDF_time_allCells.RDS")
#data.means.combined = readRDS("Correlation_Axolotl_pseudobulk.RDS")


#combine only tail

data = rbind(data.tail.bl, data.lb, stringsAsFactors = F)
data = rbind(data, data.tail.un, stringsAsFactors = F)

data.cor = data[,6:ncol(data)]

data.means <-as.data.frame(do.call("rbind", as.list(by(data.cor[,2:ncol(data.cor)], as.factor(data.cor$time), colMeans))))

data.means<-as.data.frame(t(data.means),stringsAsFactors = F)

data.means.combined <- cbind(data.means, as.data.frame( t (data.cor[,2:ncol(data.cor)]) ,stringsAsFactors = F), stringsAsFactors=F)

saveRDS(data.means.combined, "PseudobulkDF_time_OnlyTailCells.RDS")
#data.means.combined = readRDS("Correlation_Axolotl_pseudobulk.RDS")



#combine only tail blastema and uncut on cell type level

#assign broad cell type levels

Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(10)] = "PSMlike_bl"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(5)] = "Dermis_bl"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(1,7)] = "Mesenchyme_bl"
#Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(2,10)] = "Developing Mesenchyme"
#Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(3)] = "Developing Dermis"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(0,2,3,6,8)] = "Progenitors_bl"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(4,11)] = "Cartilage_bl"
Tail_1_2wpa@meta.data$celltype[Tail_1_2wpa@meta.data$seurat_cluster %in% c(9)] = "Myocytes_bl"

library(grafify)
library(RColorBrewer)


Tail_1_2wpa@meta.data$celltype <- factor(Tail_1_2wpa@meta.data$celltype, levels = c("PSMlike_bl","Progenitors_bl","Mesenchyme_bl","Dermis_bl","Cartilage_bl","Myocytes_bl"))


pdf("Tail_1_2wpa_CT_UMAP_celltype.pdf",width=10,height=10)
my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[1:8])
DimPlot(object = Tail_1_2wpa, reduction = 'umap',label = T, pt.size = 2,shuffle = T, cols = my_spectral_palette(length(unique(Tail_1_2wpa@active.ident))))
my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[rev(1:5)])
DimPlot(object = Tail_1_2wpa, reduction = 'umap',label = F, pt.size = 2,shuffle = T, group.by = 'celltype' ,cols = my_spectral_palette(length(unique(Tail_1_2wpa@meta.data$celltype))))

dev.off()

#colors for all celltype pca comparison
#"#0072B2" "#579B89" "#AEC460" "#DADD46" "#82C458" "#2BAA6A" "#0FA288" "#2EAAB3" "#4EB2DE" "#7DAEA9" "#B1A654" "#E69F00"

gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD013332","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","HES7","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")


gg_Fig <- FeaturePlot(Tail_1_2wpa, pt.size = 5, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "viridis")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) & NoLegend() & NoAxes()})

pdf("Tail_1_2wpa_CT_UMAP_feature_GeneralMarker.pdf",width=18,height=15)
CombinePlots( gg_Fig )
dev.off()



Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(6)] = "PSMlike_0"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(1,2,9,4,5,10,12,13,14)] = "Dermis_0"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(0)] = "Mesenchyme_0"
#Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(7)] = "Developing Mesenchyme"
#Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(8)] = "Developing Dermis"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(3,7,8)] = "Progenitors_0"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(11,15)] = "Cartilage_0"
#Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(15)] = "Periskeletal"


library(grafify)
library(RColorBrewer)


Tail_0wpa_all@meta.data$celltype <- factor(Tail_0wpa_all@meta.data$celltype, levels = c("PSMlike_0","Progenitors_0","Mesenchyme_0","Dermis_0","Cartilage_0"))

pdf("Tail_All0dpa_CT_UMAP_celltype.pdf",width=10,height=10)
my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[1:8])
DimPlot(object = Tail_0wpa_all, reduction = 'umap',label = T, pt.size = 2,shuffle = T, cols = my_spectral_palette(length(unique(Tail_0wpa_all@active.ident))))

my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[rev(2:5)])
DimPlot(object = Tail_0wpa_all, reduction = 'umap',label = F, pt.size = 2,shuffle = T, group.by = 'celltype',cols = my_spectral_palette(length(unique(Tail_0wpa_all@meta.data$celltype))))

dev.off()


gene_ids = c("AMEX60DD021181","AMEX60DD052891","AMEX60DD009987","AMEX60DD040818","AMEX60DD037674","AMEX60DD031236","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD029436","AMEX60DD007033","AMEX60DD009984","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD018450")
	
gene_names =c("LFNG","FGF8","MEOX1","SCX","TNMD","SOX9","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","TWIST3","FAM180A","SOST","F13B","LTBP4","F13A1","PRRX1")

my_spectral_palette = colorRampPalette(colors = grafify:::graf_palettes$okabe_ito[rev(1:5)])


gg_Fig <- VlnPlot(Tail_0wpa_all,features = gene_ids,pt.size = 0, group.by = "celltype", cols = my_spectral_palette(length(unique(Tail_0wpa_all@meta.data$celltype))))
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Tail_All0dpa_CT_Vln_PaperMarker.pdf",width=22,height=15)
VlnPlot(Tail_0wpa_all,features = gene_ids,pt.size = 0, group.by = "celltype", cols = my_spectral_palette(length(unique(Tail_0wpa_all@meta.data$celltype))))
CombinePlots( gg_Fig )
dev.off()

pdf("Tail_All0dpa_CT_Dot_PaperMarker.pdf",width=9,height=5)
(Tail_0wpa_all,features = gene_ids, group.by = "celltype") +scale_colour_gradientn(colors =  c(viridis_pal(option = "plasma")(12))) 
dev.off()




#Import tail BL data

norm.data.tail.bl = GetAssayData(object = Tail_1_2wpa, slot = "data")
norm.data.tail.bl = as.matrix(norm.data.tail.bl)
norm.data.tail.bl = as.data.frame(t(norm.data.tail.bl),stringsAsFactors=F)

#Import tail uncut data

norm.data.tail.un = GetAssayData(object = Tail_0wpa_all, slot = "data")
norm.data.tail.un = as.matrix(norm.data.tail.un)
norm.data.tail.un = as.data.frame(t(norm.data.tail.un),stringsAsFactors=F)


#Import meta data

meta.data.tail.un = FetchData(object = Tail_0wpa_all, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA","celltype"))
meta.data.tail.un$origin = "Tail_un"
meta.data.tail.un$time = "Tail_0wpa"

meta.data.tail.bl = FetchData(object = Tail_1_2wpa, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA","celltype"))
meta.data.tail.bl$origin = "Tail_bl"
meta.data.tail.bl$time  = meta.data.tail.bl$orig.ident 

data.tail.bl = cbind(meta.data.tail.bl[,c("nCount_RNA","nFeature_RNA","ident","orig.ident","origin","time","celltype")], norm.data.tail.bl, stringsAsFactors = F)
data.tail.un = cbind(meta.data.tail.un[,c("nCount_RNA","nFeature_RNA","ident","orig.ident","origin","time","celltype")], norm.data.tail.un, stringsAsFactors = F)

shared_genes = Reduce(intersect, list(colnames(data.tail.bl),colnames(data.tail.un)))

data.tail.bl = data.tail.bl[,colnames(data.tail.bl) %in% shared_genes]
data.tail.un = data.tail.un[,colnames(data.tail.un) %in% shared_genes]

data = rbind(data.tail.bl, data.tail.un, stringsAsFactors = F)

data.cor = data[,7:ncol(data)]

data.means <-as.data.frame(do.call("rbind", as.list(by(data.cor[,2:ncol(data.cor)], as.factor(data.cor$celltype), colMeans))))

data.means<-as.data.frame(t(data.means),stringsAsFactors = F)

data.means.combined <- cbind(data.means, as.data.frame( t (data.cor[,2:ncol(data.cor)]) ,stringsAsFactors = F), stringsAsFactors=F)

saveRDS(data.means.combined, "PseudobulkDF_time_OnlyTail_0_1_2wpa_Cells.RDS")
#data.means.combined = readRDS("Correlation_Axolotl_pseudobulk.RDS")


#######Extract from data frame above only progenitor cells and combine with embryonic data


data = rbind(data.tail.bl, data.tail.un, stringsAsFactors = F)

data = data[data$celltype == "PSMlike_0" | data$celltype == "PSMlike_bl",]

#Import tail LB data

norm.data.lb = GetAssayData(object = Tail_LB_int, slot = "data")
norm.data.lb = as.matrix(norm.data.lb)
norm.data.lb = as.data.frame(t(norm.data.lb),stringsAsFactors=F)

meta.data.lb = FetchData(object = Tail_LB_int, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA","Stage"))
meta.data.lb$origin = "Tail_em"
meta.data.lb$time = meta.data.lb$Stage
meta.data.lb$time = paste("em",meta.data.lb$time,sep = "_")
meta.data.lb$Stage = NULL
meta.data.lb$celltype = meta.data.lb$time
 
data.lb = cbind(meta.data.lb[,c("nCount_RNA","nFeature_RNA","ident","orig.ident","origin","time","celltype")], norm.data.lb, stringsAsFactors = F)
           
shared_genes = Reduce(intersect, list(colnames(data),colnames(data.lb)))

data.lb = data.lb[,colnames(data.lb) %in% shared_genes]
data = data[,colnames(data) %in% shared_genes]
          
                     
data  = rbind(data,data.lb,stringsAsFactors = F)

data.cor = data[,7:ncol(data)]

data.means <-as.data.frame(do.call("rbind", as.list(by(data.cor[,2:ncol(data.cor)], as.factor(data.cor$celltype), colMeans))))

data.means<-as.data.frame(t(data.means),stringsAsFactors = F)

data.means.combined <- cbind(data.means, as.data.frame( t (data.cor[,2:ncol(data.cor)]) ,stringsAsFactors = F), stringsAsFactors=F)

saveRDS(data.means.combined, "PseudobulkDF_celltype_Tail_Bud_0_1_2wpa_Cells.RDS")
#data.means.combined = readRDS("Correlation_Axolotl_pseudobulk.RDS")



```

###Cor all cells

```{r}
rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:6) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,7:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:6]

#saveRDS(cor.sc.bulk,"COR_matrix_AllInterstitialCells_Transplant.RDS")

#cor.sc.bulk = readRDS("COR_matrix_AllInterstitialCells_Transplant.RDS")
library(gplots)
library(RColorBrewer)

col.regions=colorRampPalette(c(brewer.pal(9,"Greys")[2:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk),max(cor.sc.bulk), 0.1)

hc <- hclust(as.dist(1-cor(cor.sc.bulk,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk),method="pearson")), method="ward")

pdf("COR_heatmap_time_allCells.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="col",Rowv=FALSE, key=T,scale="col",ColSideColors=c("#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(data.cor[colnames(cor.sc.bulk),"time"])])
dev.off()

#not very informative repeat with time point resolution







pdf("COR_heatmap_AllCells_Transplant_noDend.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.3(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions, key=T,scale="col",Colv=NULL, dendrogram="row",Rowv=T,ColSideColors=c("#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(data[colnames(cor.sc.bulk),"origin"])])
dev.off()

pdf("test.pdf",width=15,height=10)
ggplot(data = cor.sc.bulk , mapping = aes(x = rownames(cor.sc.bulk),y = colnames(cor.sc.bulk))) +  geom_tile()
dev.off()


```
###Cor only tail

```{r}
rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:7) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,8:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:7]

#saveRDS(cor.sc.bulk,"COR_matrix_AllInterstitialCells_Transplant.RDS")

#cor.sc.bulk = readRDS("COR_matrix_AllInterstitialCells_Transplant.RDS")
library(gplots)
library(RColorBrewer)

col.regions=colorRampPalette(c(brewer.pal(9,"Greys")[2:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk),max(cor.sc.bulk), 0.1)

hc <- hclust(as.dist(1-cor(cor.sc.bulk,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk),method="pearson")), method="ward")

pdf("COR_heatmap_time_OnlyTailCells.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="col",Rowv=FALSE, key=T,scale="col",ColSideColors=c("#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(data.cor[colnames(cor.sc.bulk),"time"])])
dev.off()

#not very informative repeat with time point resolution







pdf("COR_heatmap_AllCells_Transplant_noDend.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.3(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions, key=T,scale="col",Colv=NULL, dendrogram="row",Rowv=T,ColSideColors=c("#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(data[colnames(cor.sc.bulk),"origin"])])
dev.off()

pdf("test.pdf",width=15,height=10)
ggplot(data = cor.sc.bulk , mapping = aes(x = rownames(cor.sc.bulk),y = colnames(cor.sc.bulk))) +  geom_tile()
dev.off()


```


###Cor tail cell types cells

```{r}
rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:11) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,12:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:11]

#saveRDS(cor.sc.bulk,"COR_matrix_AllInterstitialCells_Transplant.RDS")

#cor.sc.bulk = readRDS("COR_matrix_AllInterstitialCells_Transplant.RDS")
library(gplots)
library(RColorBrewer)

col.regions=colorRampPalette(c(brewer.pal(9,"Greys")[2:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk),max(cor.sc.bulk), 0.1)

hc <- hclust(as.dist(1-cor(cor.sc.bulk,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk),method="pearson")), method="ward")

pdf("COR_heatmap_celltype_OnlyTailCells.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="col",Rowv=FALSE, key=T,scale="col",ColSideColors=c("#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(data.cor[colnames(cor.sc.bulk),"celltype"])])
dev.off()

pdf("test.pdf",width=15,height=10)
ggplot(data = cor.sc.bulk , mapping = aes(x = rownames(cor.sc.bulk),y = colnames(cor.sc.bulk))) +  geom_tile()
dev.off()

#load in seurat

cor.sc.bulk.seu <- CreateSeuratObject(counts = cor.sc.bulk, project = "CelltypeCorrelation", min.cells = 0, min.features = 0)
cor.sc.bulk.seu

cor.sc.bulk.seu@meta.data$time = data$time
cor.sc.bulk.seu@meta.data$origin = data$origin
cor.sc.bulk.seu@meta.data$orig.ident = data$orig.ident
cor.sc.bulk.seu@meta.data$celltype = data$celltype


cor.sc.bulk.seu = ScaleData(  cor.sc.bulk.seu, features = rownames(cor.sc.bulk.seu))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

cor.sc.bulk.seu = RunPCA(cor.sc.bulk.seu,features = rownames(cor.sc.bulk.seu),npcs = 10)


pdf("COR_Seurat_celltype_OnlyTailCells_PCA.pdf",width=10,height=10)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'orig.ident',dims = 3:4)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'origin',dims = 3:4)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'celltype',dims = 3:4)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'time',dims = 3:4)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'orig.ident',dims = 5:6)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'origin',dims = 5:6)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'celltype',dims = 5:6)
DimPlot(object = cor.sc.bulk.seu, reduction = 'pca', pt.size = 2,shuffle = T,label = T,group.by = 'time',dims = 5:6)
dev.off()

cor.sc.bulk.seu<- FindNeighbors(cor.sc.bulk.seu, dims = 3:10)
cor.sc.bulk.seu <- FindClusters(cor.sc.bulk.seu, resolution = 0.6)

#run UMAP
cor.sc.bulk.seu = RunUMAP(cor.sc.bulk.seu,dims = 3:10)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("COR_Seurat_celltype_OnlyTailCells_UMAP.pdf",width=20,height=20)
DimPlot(object = cor.sc.bulk.seu,label = T, reduction = 'umap', pt.size = 3)
DimPlot(object = cor.sc.bulk.seu,label = T, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
DimPlot(object = cor.sc.bulk.seu,label = T, reduction = 'umap',group.by = "origin", pt.size = 3)
DimPlot(object = cor.sc.bulk.seu,label = T, reduction = 'umap',group.by = "celltype", pt.size = 3)
DimPlot(object = cor.sc.bulk.seu,label = T, reduction = 'umap',group.by = "time", pt.size = 3)
dev.off()


saveRDS(cor.sc.bulk.seu,"COR_Seurat_celltype_OnlyTailCells.RDS")
```

###Cor only progenitors and tail bud cells

```{r}
rm(cor)
rm(cor.sc.bulk)

cor.sc.bulk<-data.frame()

for(i in 1:6) {
        print(i)
        cor<-cor(data.means.combined[,i], data.means.combined[,7:ncol(data.means.combined)], method = c("spearman"))
        cor.sc.bulk<-rbind(cor.sc.bulk,cor)
}

rownames(cor.sc.bulk)<-colnames(data.means.combined)[1:6]

#saveRDS(cor.sc.bulk,"COR_matrix_AllInterstitialCells_Transplant.RDS")

#cor.sc.bulk = readRDS("COR_matrix_AllInterstitialCells_Transplant.RDS")
library(gplots)
library(RColorBrewer)

col.regions=colorRampPalette(c(brewer.pal(9,"Greys")[2:9]), space="Lab")

palette.breaks <- seq(min(cor.sc.bulk),max(cor.sc.bulk), 0.1)

hc <- hclust(as.dist(1-cor(cor.sc.bulk,method="pearson")), method="ward")
hr <- hclust(as.dist(1-cor(t(cor.sc.bulk),method="pearson")), method="ward")

pdf("COR_heatmap_celltype_Tail_Bud_0_1_2wpa_ProgCells.pdf",width=15,height=10)
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="col",Rowv=FALSE, key=T,scale="col",ColSideColors=c("#fe9929","#d95f0e","#993404","#ffffcc","#a1dab4","#41b6c4","#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(data[colnames(cor.sc.bulk),"celltype"])])
heatmap.celltypes<-heatmap.2(as.matrix(cor.sc.bulk),trace="none",density="none",col=col.regions,Colv=as.dendrogram(hc), dendrogram="col",Rowv=FALSE, key=T,scale="col",ColSideColors=c("#fed98e","#fe9929","#d95f0e","#993404","#a1dab4","#41b6c4","#2c7fb8","#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" ,"dodgerblue","palegreen1","palegreen2","palegreen3","palegreen4")[as.factor(data[colnames(cor.sc.bulk),"time"])])
dev.off()

```

###QP all cells

```{r}

################stage 35 as reference##################


### run with all cells

data.qp = data.means.combined

data.qp = cbind(fc_Tail0dpaVSst35 = data.qp[,"Tail_0wpa"] - data.qp[,"em_35"], data.qp, stringsAsFactors = F)

#genes.fc_0dpaVSst54 <- c(rownames(data.qp[which(data.qp$fc_0dpaVSst54>1),]),rownames(data.qp[which(data.qp$fc_0dpaVSst54<(-1)),]))
genes.fc_Tail0dpaVSst35 <-data.qp[data.qp$fc_Tail0dpaVSst35>0.5 | data.qp$fc_Tail0dpaVSst35<(-0.5),] 
genes.fc_Tail0dpaVSst35 = genes.fc_Tail0dpaVSst35[order(genes.fc_Tail0dpaVSst35$fc_0dpaVSst54),]
genes.fc_Tail0dpaVSst35 = unique(c(tail(rownames(genes.fc_Tail0dpaVSst35),100),rev(head(rownames(genes.fc_Tail0dpaVSst35),100))))


tmp = data.frame(Gene_id = genes.fc_Tail0dpaVSst35)
tmp = merge(tmp,anno,by.x = "Gene_id",by.y = "V1")
tmp = tmp[match(genes.fc_Tail0dpaVSst35,tmp$Gene_id),]
tmp$sample[1:100] = "matureTail" 
tmp$sample[101:200] = "stage35Tail" 

write.table(tmp,"./QP_genes.fc_Tail0dpaVSst35_AxolotlSet_raw.csv",row.names=TRUE,col.names=F,sep=",")

genes.fc_Tail0dpaVSst35 = genes.fc_Tail0dpaVSst35[!genes.fc_Tail0dpaVSst35 %in% c("eGFP","mCherry")]

write.table(genes.fc_Tail0dpaVSst35,"./QP_genes.fc_Tail0dpaVSst35.csv",row.names=TRUE,col.names=F,sep=",")

genes.fc_Tail0dpaVSst35 = read.csv("./QP_genes.fc_0dpaVSst54.csv",header = F)
genes.fc_Tail0dpaVSst35 = as.character(genes.fc_Tail0dpaVSst35$V2)

#keep only relevant genes

data.qp = data.qp[genes.fc_Tail0dpaVSst35,]

X <- as.matrix(data.qp[,c("Tail_0wpa","em_35")])

identity.all<-data.frame()

library("quadprog");
library("reshape")

for (i in 13:ncol(data.qp)){
  identity<-c()
  Y <- as.matrix(data.qp[,i])
  Rinv <- solve(chol(t(X) %*% X));
  C <- cbind(rep(1,2), diag(2))
  b <- c(1,rep(0,2))
  d <- t(Y) %*% X  
  QP<-solve.QP(Dmat = Rinv, factorized = TRUE, dvec = d, Amat = C, bvec = b, meq = 1)
  Error<-sum(abs(Y-X%*%QP$solution))
  identity<-cbind(colnames(data.qp[i]),QP$solution[1],QP$solution[2],QP$Lagrangian[1],Error)
  identity.all<-rbind(identity.all,identity)
  }

colnames(identity.all)<-c("cell_name","frxn_Tail0dpa","frxn_st35Tail","Lagrangian","Error")
rownames(identity.all)<-identity.all[,"cell_name"]
identity.all$frxn_Tail0dpa<-as.numeric(identity.all$frxn_Tail0dpa)
identity.all$frxn_st35Tail<-as.numeric(identity.all$frxn_st35Tail)
identity.all$Lagrangian<-as.numeric(identity.all$Lagrangian)
identity.all$Error<-as.numeric(identity.all$Error)


saveRDS(identity.all,"QP_IdentityResult_Tail0dpaVSst35Tail")
#identity.all = readRDS("QP_IdentityResult_0dpaVSst54")

identity.all.meta<-cbind(data[rownames(identity.all),c("nCount_RNA","nFeature_RNA","origin","time")],identity.all)
identity.all.meta.sort <- identity.all.meta[order(identity.all.meta$frxn_st35Tail),] 
identity.all.meta.sort$Num<-c(1:nrow(identity.all.meta.sort))
identity.all.meta.sort$NumJitter = NA
for (i in 1 : nrow(identity.all.meta.sort)) {
  identity.all.meta.sort[i,"NumJitter"] =  identity.all.meta.sort[i,"Num"] + sample(-1000:1000,1)
}
identity.all.meta.sort$order = NA

identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_0wpa"] = 1
identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_col_1wpa"] = 2
identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_col_2wpa"] = 3
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_25"] = 4
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_28"] = 5
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_30"] = 6
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_35"] = 7
identity.all.meta.sort$order[identity.all.meta.sort$time == "BL11dpa"] = 8
identity.all.meta.sort$order[identity.all.meta.sort$time == "BL18dpa"] = 9
identity.all.meta.sort$order[identity.all.meta.sort$time == "BL25dpa"] = 10
identity.all.meta.sort$order[identity.all.meta.sort$time == "BL38dpa"] = 11


#randomize plotting
identity.all.meta.sort$plot = sample(1:nrow(identity.all.meta.sort),replace = F)
identity.all.meta.sort = identity.all.meta.sort[order(identity.all.meta.sort$plot),]

#-----------------------------------------------
#plot each cell based on fractional identity with all cells
#-----------------------------------------------
library(ggplot2)
library(RColorBrewer)

pdf("QP_Tail0dpaVSst35Tail_Allcells__Num.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=NumJitter, y=frxn_st35Tail, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + xlab(expression(paste("Cell number"))) +ylab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaVSst35Tail_Allcells__frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=frxn_st35Tail, y=frxn_Tail0dpa, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + ylab(expression(paste("fraction 0dpa identity"))) +xlab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaVSst35Tail_Allcells_violin_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_violin(scale="width", trim=T) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("red3","darkslategray","gray20","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()

pdf("QP_Tail0dpaVSst35Tail_Allcells_box_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_boxplot(outlier.colour="black", outlier.shape=16,outlier.size=1, notch=FALSE) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("red3","darkslategray","gray20","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()

 
pdf("QP_Tail0dpaVSst35Tail_Allcells_violin_nUMI.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail, color = nCount_RNA)) + geom_jitter() + theme_bw() + ylim(0,1) + scale_colour_gradientn(colors = c(brewer.pal(9,"Greys")[3],brewer.pal(9,"Blues")[3:9]))
dev.off()

#####some mature cells are quite high on embryonic identity --> extract cells and perform DE on Seurat object

saveRDS(identity.all.meta.sort,"QP_Tail0dpaVSst35Tail_Allcells_data.RDS")

```

###QP only tail

```{r}

################stage 35 as reference##################


### run with all cells

data.qp = data.means.combined

data.qp = cbind(fc_Tail0dpaVSst35 = data.qp[,"Tail_0wpa"] - data.qp[,"em_35"], data.qp, stringsAsFactors = F)

#genes.fc_0dpaVSst54 <- c(rownames(data.qp[which(data.qp$fc_0dpaVSst54>1),]),rownames(data.qp[which(data.qp$fc_0dpaVSst54<(-1)),]))
genes.fc_Tail0dpaVSst35 <-data.qp[data.qp$fc_Tail0dpaVSst35>0.5 | data.qp$fc_Tail0dpaVSst35<(-0.5),] 
genes.fc_Tail0dpaVSst35 = genes.fc_Tail0dpaVSst35[order(genes.fc_Tail0dpaVSst35$fc_0dpaVSst54),]
genes.fc_Tail0dpaVSst35 = unique(c(tail(rownames(genes.fc_Tail0dpaVSst35),100),rev(head(rownames(genes.fc_Tail0dpaVSst35),100))))


tmp = data.frame(Gene_id = genes.fc_Tail0dpaVSst35)
tmp = merge(tmp,anno,by.x = "Gene_id",by.y = "V1")
tmp = tmp[match(genes.fc_Tail0dpaVSst35,tmp$Gene_id),]
tmp$sample[1:100] = "matureTail" 
tmp$sample[101:200] = "stage35Tail" 

write.table(tmp,"./QP_genes.fc_Tail0dpaVSst35_AxolotlSet_raw.csv",row.names=TRUE,col.names=F,sep=",")

genes.fc_Tail0dpaVSst35 = genes.fc_Tail0dpaVSst35[!genes.fc_Tail0dpaVSst35 %in% c("eGFP","mCherry")]

write.table(genes.fc_Tail0dpaVSst35,"./QP_genes.fc_Tail0dpaVSst35.csv",row.names=TRUE,col.names=F,sep=",")

genes.fc_Tail0dpaVSst35 = read.csv("./QP_genes.fc_0dpaVSst54.csv",header = F)
genes.fc_Tail0dpaVSst35 = as.character(genes.fc_Tail0dpaVSst35$V2)

#keep only relevant genes

data.qp = data.qp[genes.fc_Tail0dpaVSst35,]

X <- as.matrix(data.qp[,c("Tail_0wpa","em_35")])

identity.all<-data.frame()

library("quadprog");
library("reshape")

for (i in 9:ncol(data.qp)){
  identity<-c()
  Y <- as.matrix(data.qp[,i])
  Rinv <- solve(chol(t(X) %*% X));
  C <- cbind(rep(1,2), diag(2))
  b <- c(1,rep(0,2))
  d <- t(Y) %*% X  
  QP<-solve.QP(Dmat = Rinv, factorized = TRUE, dvec = d, Amat = C, bvec = b, meq = 1)
  Error<-sum(abs(Y-X%*%QP$solution))
  identity<-cbind(colnames(data.qp[i]),QP$solution[1],QP$solution[2],QP$Lagrangian[1],Error)
  identity.all<-rbind(identity.all,identity)
  }

colnames(identity.all)<-c("cell_name","frxn_Tail0dpa","frxn_st35Tail","Lagrangian","Error")
rownames(identity.all)<-identity.all[,"cell_name"]
identity.all$frxn_Tail0dpa<-as.numeric(identity.all$frxn_Tail0dpa)
identity.all$frxn_st35Tail<-as.numeric(identity.all$frxn_st35Tail)
identity.all$Lagrangian<-as.numeric(identity.all$Lagrangian)
identity.all$Error<-as.numeric(identity.all$Error)


saveRDS(identity.all,"QP_IdentityResult_Tail0dpaVSst35Tail_onlyTail")
#identity.all = readRDS("QP_IdentityResult_0dpaVSst54")

identity.all.meta<-cbind(data[rownames(identity.all),c("nCount_RNA","nFeature_RNA","origin","time")],identity.all)
identity.all.meta.sort <- identity.all.meta[order(identity.all.meta$frxn_st35Tail),] 
identity.all.meta.sort$Num<-c(1:nrow(identity.all.meta.sort))
identity.all.meta.sort$NumJitter = NA
for (i in 1 : nrow(identity.all.meta.sort)) {
  identity.all.meta.sort[i,"NumJitter"] =  identity.all.meta.sort[i,"Num"] + sample(-1000:1000,1)
}
identity.all.meta.sort$order = NA

identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_0wpa"] = 1
identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_col_1wpa"] = 2
identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_col_2wpa"] = 3
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_25"] = 4
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_28"] = 5
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_30"] = 6
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_35"] = 7



#randomize plotting
identity.all.meta.sort$plot = sample(1:nrow(identity.all.meta.sort),replace = F)
identity.all.meta.sort = identity.all.meta.sort[order(identity.all.meta.sort$plot),]

#-----------------------------------------------
#plot each cell based on fractional identity with all cells
#-----------------------------------------------
library(ggplot2)
library(RColorBrewer)

pdf("QP_Tail0dpaVSst35Tail_onlyTail__Num.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=NumJitter, y=frxn_st35Tail, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + xlab(expression(paste("Cell number"))) +ylab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaVSst35Tail_onlyTail__frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=frxn_st35Tail, y=frxn_Tail0dpa, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + ylab(expression(paste("fraction 0dpa identity"))) +xlab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaVSst35Tail_onlyTail_violin_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_violin(scale="width", trim=T) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("red3","darkslategray","gray20","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()

pdf("QP_Tail0dpaVSst35Tail_onlyTail_box_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_boxplot(outlier.colour="black", outlier.shape=16,outlier.size=1, notch=FALSE) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("red3","darkslategray","gray20","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()

 
pdf("QP_Tail0dpaVSst35Tail_onlyTail_violin_nUMI.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail, color = nCount_RNA)) + geom_jitter() + theme_bw() + ylim(0,1) + scale_colour_gradientn(colors = c(brewer.pal(9,"Greys")[3],brewer.pal(9,"Blues")[3:9]))
dev.off()

#####some mature cells are quite high on embryonic identity --> extract cells and perform DE on Seurat object

saveRDS(identity.all.meta.sort,"QP_Tail0dpaVSst35Tail_Allcells_data.RDS")

```


###QP tail bud and progenitors of uncut and blastema

```{r}

################stage 35 as reference##################


### run with all cells

data.qp = data.means.combined

data.qp = cbind(fc_Tail0dpaVSst35 = data.qp[,"PSMlike_0"] - data.qp[,"em_35"], data.qp, stringsAsFactors = F)

#genes.fc_0dpaVSst54 <- c(rownames(data.qp[which(data.qp$fc_0dpaVSst54>1),]),rownames(data.qp[which(data.qp$fc_0dpaVSst54<(-1)),]))
genes.fc_Tail0dpaVSst35 <-data.qp[data.qp$fc_Tail0dpaVSst35>0.5 | data.qp$fc_Tail0dpaVSst35<(-0.5),] 
genes.fc_Tail0dpaVSst35 = genes.fc_Tail0dpaVSst35[order(genes.fc_Tail0dpaVSst35$fc_0dpaVSst54),]
genes.fc_Tail0dpaVSst35 = unique(c(tail(rownames(genes.fc_Tail0dpaVSst35),100),rev(head(rownames(genes.fc_Tail0dpaVSst35),100))))


tmp = data.frame(Gene_id = genes.fc_Tail0dpaVSst35)
tmp = merge(tmp,anno,by.x = "Gene_id",by.y = "V1")
tmp = tmp[match(genes.fc_Tail0dpaVSst35,tmp$Gene_id),]
tmp$sample[1:100] = "matureTail" 
tmp$sample[101:200] = "stage35Tail" 

write.table(tmp,"./QP_genes.fc_Tail0dpaProgVSst35_AxolotlSet_raw.csv",row.names=TRUE,col.names=F,sep=",")

genes.fc_Tail0dpaVSst35 = genes.fc_Tail0dpaVSst35[!genes.fc_Tail0dpaVSst35 %in% c("eGFP","mCherry")]

write.table(genes.fc_Tail0dpaVSst35,"./QP_genes.fc_Tail0dpaProgVSst35.csv",row.names=TRUE,col.names=F,sep=",")

genes.fc_Tail0dpaVSst35 = read.csv("./QP_genes.fc_0dpaProgVSst54.csv",header = F)
genes.fc_Tail0dpaVSst35 = as.character(genes.fc_Tail0dpaVSst35$V2)

#keep only relevant genes

data.qp = data.qp[genes.fc_Tail0dpaVSst35,]

X <- as.matrix(data.qp[,c("PSMlike_0","em_35")])

identity.all<-data.frame()

library("quadprog");
library("reshape")

for (i in 8:ncol(data.qp)){
  identity<-c()
  Y <- as.matrix(data.qp[,i])
  Rinv <- solve(chol(t(X) %*% X));
  C <- cbind(rep(1,2), diag(2))
  b <- c(1,rep(0,2))
  d <- t(Y) %*% X  
  QP<-solve.QP(Dmat = Rinv, factorized = TRUE, dvec = d, Amat = C, bvec = b, meq = 1)
  Error<-sum(abs(Y-X%*%QP$solution))
  identity<-cbind(colnames(data.qp[i]),QP$solution[1],QP$solution[2],QP$Lagrangian[1],Error)
  identity.all<-rbind(identity.all,identity)
  }

colnames(identity.all)<-c("cell_name","frxn_Tail0dpa","frxn_st35Tail","Lagrangian","Error")
rownames(identity.all)<-identity.all[,"cell_name"]
identity.all$frxn_Tail0dpa<-as.numeric(identity.all$frxn_Tail0dpa)
identity.all$frxn_st35Tail<-as.numeric(identity.all$frxn_st35Tail)
identity.all$Lagrangian<-as.numeric(identity.all$Lagrangian)
identity.all$Error<-as.numeric(identity.all$Error)


saveRDS(identity.all,"QP_IdentityResult_Tail0dpaProgVSst35Tail_onlyTail")
#identity.all = readRDS("QP_IdentityResult_0dpaVSst54")

identity.all.meta<-cbind(data[rownames(identity.all),c("nCount_RNA","nFeature_RNA","origin","time","celltype")],identity.all)
identity.all.meta.sort <- identity.all.meta[order(identity.all.meta$frxn_st35Tail),] 
identity.all.meta.sort$Num<-c(1:nrow(identity.all.meta.sort))
identity.all.meta.sort$NumJitter = NA
for (i in 1 : nrow(identity.all.meta.sort)) {
  identity.all.meta.sort[i,"NumJitter"] =  identity.all.meta.sort[i,"Num"] + sample(-1000:1000,1)
}
identity.all.meta.sort$order = NA

identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_0wpa"] = 1
identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_col_1wpa"] = 2
identity.all.meta.sort$order[identity.all.meta.sort$time == "Tail_col_2wpa"] = 3
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_25"] = 4
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_28"] = 5
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_30"] = 6
identity.all.meta.sort$order[identity.all.meta.sort$time == "em_35"] = 7



#randomize plotting
identity.all.meta.sort$plot = sample(1:nrow(identity.all.meta.sort),replace = F)
identity.all.meta.sort = identity.all.meta.sort[order(identity.all.meta.sort$plot),]

#-----------------------------------------------
#plot each cell based on fractional identity with all cells
#-----------------------------------------------
library(ggplot2)
library(RColorBrewer)

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Bud_0_1_2wpa_Prog__Num.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=NumJitter, y=frxn_st35Tail, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + xlab(expression(paste("Cell number"))) +ylab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Bud_0_1_2wpa_Prog__frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=frxn_st35Tail, y=frxn_Tail0dpa, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + ylab(expression(paste("fraction 0dpa identity"))) +xlab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Bud_0_1_2wpa_Prog_violin_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_violin(scale="width", trim=T) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("red3","darkslategray","gray20","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Bud_0_1_2wpa_Prog_box_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_boxplot(outlier.colour="black", outlier.shape=16,outlier.size=1, notch=FALSE) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("red3","darkslategray","gray20","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()

#remove stage 28 as it is a separate experiment and looks like a outlier which is purely technical

identity.all.meta.sort = identity.all.meta.sort[!identity.all.meta.sort$celltype %in% "em_28",]

#-----------------------------------------------
#plot each cell based on fractional identity with all cells
#-----------------------------------------------
library(ggplot2)
library(RColorBrewer)

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Budno28_0_1_2wpa_Prog__Num.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=NumJitter, y=frxn_st35Tail, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + xlab(expression(paste("Cell number"))) +ylab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "palegreen1","palegreen2","palegreen3","#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Budno28_0_1_2wpa_Prog__frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=frxn_st35Tail, y=frxn_Tail0dpa, color=time))  + geom_point(size = 1) + theme(axis.text.x = element_text(angle = 90, hjust = 1))  + ylab(expression(paste("fraction 0dpa identity"))) +xlab(expression(paste("fraction embryonic identity"))) + theme_bw() + scale_color_manual(values=c( "#A50026","#EFE9C3" , "#ABD9E9" ,"#5C91C2" , "#E54E35","#313695","#FDAE61" , "dodgerblue", "palegreen1","palegreen2","palegreen3","palegreen4","purple","goldenrod2","dodgerblue" ))  
dev.off()

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Budno28_0_1_2wpa_Prog_violin_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_violin(scale="width", trim=T) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("red3","darkslategray","gray20","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()

pdf("QP_Tail0dpaProgVSst35Tail_Tail_Budno28_0_1_2wpa_Prog_box_frxn.pdf",width=10,height=10)
ggplot(data = identity.all.meta.sort, aes(x=as.factor(order), y=frxn_st35Tail,fill=time)) + geom_boxplot(outlier.colour="black", outlier.shape=16,outlier.size=1, notch=FALSE) + theme_bw() + ylim(0,1) + scale_fill_manual(values=c("#fe9929","#d95f0e","#993404","#ffffcc","#a1dab4","#41b6c4","darkslategray4","turquoise2","dodgerblue","palegreen1","palegreen3","palegreen4","purple","goldenrod2","dodgerblue"))
dev.off()





saveRDS(identity.all.meta.sort,"QP_Tail0dpaVSst35Tail_Allcells_data.RDS")

```


Do not run following things

##Pseudotime

```{r}


Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(6)] = "PSMlike"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(1,2,9,4,5,10,12,13,14)] = "Dermis"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(0)] = "Mesenchyme"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(7)] = "Developing Mesenchyme"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(8)] = "Developing Dermis"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(3)] = "Progenitors"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(11)] = "Cartilage"
Tail_0wpa_all@meta.data$celltype[Tail_0wpa_all@meta.data$seurat_cluster %in% c(15)] = "Periskeletal"





######Calc pseudo time 

harmony.embedding = Embeddings(Tail_0wpa_all, "harmony")

meta.data = FetchData(object = Tail_0wpa_all, vars = c("celltype",'ident', 'orig.ident','UMAP_1','UMAP_2'))

cells_PSM = rownames(meta.data)[meta.data$celltype %in% c("PSMlike")]
cells_dermis = rownames(meta.data)[meta.data$celltype %in% c("Dermis")]
cells_mesen = rownames(meta.data)[meta.data$celltype %in% c("Mesenchyme")]
cells_dev_mesen = rownames(meta.data)[meta.data$celltype %in% c("Developing Mesenchyme")]
cells_dev_dermis = rownames(meta.data)[meta.data$celltype %in% c("Developing Dermis")]
cells_peri = rownames(meta.data)[meta.data$celltype %in% c("Periskeletal")]
cells_prog = rownames(meta.data)[meta.data$celltype %in% c("Progenitors")]
cells_cart = rownames(meta.data)[meta.data$celltype %in% c("Cartilage")]

#calculate pseudo time separate on branches

harmony.embedding.PSM = as.data.frame(harmony.embedding[cells_PSM,])
harmony.embedding.prog = as.data.frame(harmony.embedding[cells_prog,])
harmony.embedding.dev.dermis = as.data.frame(harmony.embedding[cells_dev_dermis,])
harmony.embedding.dev.mesen = as.data.frame(harmony.embedding[cells_dev_mesen,])
harmony.embedding.dermis = as.data.frame(harmony.embedding[cells_dermis,])
harmony.embedding.mesen = as.data.frame(harmony.embedding[cells_mesen,])
harmony.embedding.cart = as.data.frame(harmony.embedding[cells_cart,])
harmony.embedding.peri = as.data.frame(harmony.embedding[cells_peri,])

library("destiny")

#PSM
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.PSM),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_PSM = dpt[random_root(dpt), ]

harmony.embedding.PSM$pt = as.numeric(pseudotime_PSM)
harmony.embedding.PSM = harmony.embedding.PSM[order(harmony.embedding.PSM$pt),]
harmony.embedding.PSM$rank = rev(1:nrow(harmony.embedding.PSM))

#prog
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.prog),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_prog = dpt[random_root(dpt), ]

harmony.embedding.prog$pt = as.numeric(pseudotime_prog)
harmony.embedding.prog = harmony.embedding.prog[order(harmony.embedding.prog$pt),]
harmony.embedding.prog$rank = (1:nrow(harmony.embedding.prog))
harmony.embedding.prog$rank = harmony.embedding.prog$rank  + nrow(harmony.embedding.PSM) 

#dev mesen
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.dev.mesen),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_dev_mesen = dpt[random_root(dpt), ]

harmony.embedding.dev.mesen$pt = as.numeric(pseudotime_dev_mesen)
harmony.embedding.dev.mesen = harmony.embedding.dev.mesen[order(harmony.embedding.dev.mesen$pt),]
harmony.embedding.dev.mesen$rank = rev(1:nrow(harmony.embedding.dev.mesen))
harmony.embedding.dev.mesen$rank = harmony.embedding.dev.mesen$rank  + nrow(harmony.embedding.PSM)  + nrow(harmony.embedding.prog) 

#dev dermis
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.dev.dermis),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_dev_dermis = dpt[random_root(dpt), ]

harmony.embedding.dev.dermis$pt = as.numeric(pseudotime_dev_dermis)
harmony.embedding.dev.dermis = harmony.embedding.dev.dermis[order(harmony.embedding.dev.dermis$pt),]
harmony.embedding.dev.dermis$rank = rev(1:nrow(harmony.embedding.dev.dermis))
harmony.embedding.dev.dermis$rank = harmony.embedding.dev.dermis$rank  + nrow(harmony.embedding.PSM)  + nrow(harmony.embedding.prog) 


#mesen
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.mesen),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_mesen = dpt[random_root(dpt), ]

harmony.embedding.mesen$pt = as.numeric(pseudotime_mesen)
harmony.embedding.mesen = harmony.embedding.mesen[order(harmony.embedding.mesen$pt),]
harmony.embedding.mesen$rank = rev(1:nrow(harmony.embedding.mesen))
harmony.embedding.mesen$rank = harmony.embedding.mesen$rank  + nrow(harmony.embedding.PSM)  + nrow(harmony.embedding.prog) + nrow(harmony.embedding.dev.mesen) 

#dermis
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.dermis),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_dermis = dpt[random_root(dpt), ]

harmony.embedding.dermis$pt = as.numeric(pseudotime_dermis)
harmony.embedding.dermis = harmony.embedding.dermis[order(harmony.embedding.dermis$pt),]
harmony.embedding.dermis$rank = rev(1:nrow(harmony.embedding.dermis))
harmony.embedding.dermis$rank = harmony.embedding.dermis$rank  + nrow(harmony.embedding.PSM)  + nrow(harmony.embedding.prog) + nrow(harmony.embedding.dev.dermis) 

#cart
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.cart),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_cart = dpt[random_root(dpt), ]

harmony.embedding.cart$pt = as.numeric(pseudotime_cart)
harmony.embedding.cart = harmony.embedding.cart[order(harmony.embedding.cart$pt),]
harmony.embedding.cart$rank = rev(1:nrow(harmony.embedding.cart))
harmony.embedding.cart$rank = harmony.embedding.cart$rank  + nrow(harmony.embedding.PSM)  + nrow(harmony.embedding.prog)

#periskeletal

DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.peri),stringsAsFactor = F))

dpt = DPT(DatasetDM )

pseudotime_peri = dpt[random_root(dpt), ]

harmony.embedding.peri$pt = as.numeric(pseudotime_peri)
harmony.embedding.peri = harmony.embedding.peri[order(harmony.embedding.peri$pt),]
harmony.embedding.peri$rank = rev(1:nrow(harmony.embedding.peri))
harmony.embedding.peri$rank = harmony.embedding.peri$rank  + nrow(harmony.embedding.PSM)  + nrow(harmony.embedding.prog)

#merge
pseudotime = rbind(harmony.embedding.PSM,harmony.embedding.prog,harmony.embedding.cart,harmony.embedding.peri,harmony.embedding.dermis,harmony.embedding.mesen,harmony.embedding.dev.dermis,harmony.embedding.dev.mesen,stringsAsFactors = F)

plot.pt = cbind(meta.data,rank = pseudotime[match(rownames(meta.data),rownames(pseudotime)),"rank"], stringsAsFactors = F)

saveRDS(plot.pt,"Tail_0wpa_all_Pseudotime_data.RDS")


### set some parameters
# cells
cellcex = 0.6
#color
colramp = colorRampPalette((brewer.pal(9,"Spectral")[3:9]))
colgrad = colramp(101)

pdf("./Tail_0wpa_all_UMAP_PT_Rank.pdf",width=8,height=8)

  # normalize expression [0-1]
  exp = plot.pt [,"rank"]/max(plot.pt [,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt [,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt$UMAP_1, plot.pt$UMAP_2, cex=cellcex, pch=16, col=colgrad[plot.pt[,"bin"]], xlab="", ylab="", main = "pseudotime")

   # points(plot.pt$UMAP_1, plot.pt$UMAP_2, cex=cellcex, pch=16, col=colgrad[plot.pt [,"bin"]], xlab="", ylab="")



dev.off()


```


SPRING visualization

```{r}

Int_col_BL_CT_int = readRDS(file = "Int_col_onlyBL_CT_HarmonyTimeInt_PC50_SeuratObj.RDS")

###Prepare data

harmony.embedding = Embeddings(Int_col_BL_CT_int, "harmony")


#make scaled values positive
for (i in 1:ncol(harmony.embedding)) {
  harmony.embedding[,i] = harmony.embedding[,i] + abs(min(harmony.embedding[,i]))
}


#use only cluster marker to run SPRING

markers <- FindAllMarkers(Int_col_BL_CT_int, only.pos = TRUE,  logfc.threshold = 0.5)
regeneration.marker = as.character(unique(markers$gene[ markers$avg_logFC > 0.7]))

#extract norm data from Seurat object

norm.data = GetAssayData(object = Int_col_BL_CT_int, slot = "data")
norm.data = as.matrix(norm.data)


meta.data = FetchData(object = Int_col_BL_CT_int, vars = c('ident', 'orig.ident','percent.mt',"nFeature_RNA","nCount_RNA"))

##############################
#SPRING
##############################

### export for SPRING (export norma and not scaled data)

#cutoff 0.8
y=as.data.frame(t(meta.data[,c("orig.ident","ident")]),stringsAsFactors=F)

write.table(y,sep = ",",col.names = F,file = "SPRING_Export_Int_col_onlyBL_CT_meta.csv")

x = as.data.frame(t(harmony.embedding[,c(1:50)]))

write.table(x,sep = ",",col.names = F,file ="SPRING_Export_Int_col_onlyBL_CT_HarmonyMatrix.csv")

z=rownames(x)

write.table(z,col.names = F,row.names = F,sep = ",",file = "SPRING_Export_Int_col_onlyBL_CT_GeneList.csv")

###Run with : 50 PCs gene / 3 neighbors / gene variability percentile 20.0


#import SPRING tables


coords = read.csv("SPRING_Import_Int_col_onlyBL_CT_coordinates.txt",header = F)
colnames(coords) = c("Cell","x","y")
edges = read.csv("SPRING_Import_Int_col_onlyBL_CT_edge_list.txt",header = F)
colnames(edges) = c("start","end")

rownames(coords) = rownames(harmony.embedding)



### set some parameters
# cells
colstart = "darkblue"
colmed1 = "green"
colmed2 = "yellow"
colmed3 = "orange"
colend = "firebrick"
cellcex = 0.8
# edges
linecol = adjustcolor("gray10", alpha.f=0.4)
#linecol = adjustcolor("gray")
linewidth = 0.5

coords$ident=Int_col_BL_CT_int@active.ident

coords$color[coords$ident==0]="cyan2"
coords$color[coords$ident==1]="cyan3"
coords$color[coords$ident==2]="turquoise"
coords$color[coords$ident==3]="aquamarine3"
coords$color[coords$ident==4]="aquamarine1"
coords$color[coords$ident==5]="magenta4"
coords$color[coords$ident==6]="darkslategray3"
coords$color[coords$ident==7]="gray50"
coords$color[coords$ident==8]="maroon3"
coords$color[coords$ident==9]="gray80"
coords$color[coords$ident==10]="plum4"
coords$color[coords$ident==11]="turquoise3"
coords$color[coords$ident==12]="gray65"
coords$color[coords$ident==13]="magenta3"
coords$color[coords$ident==14]="hotpink4"
coords$color[coords$ident==15]="gray30"

plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]

pdf("./Int_col_onlyBL_CT_SPRING_ident.pdf",width=8,height=8)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()


coords$time=Int_col_BL_CT_int@meta.data$orig.ident

#coords$color[coords$time=="0dpt"]="#A50026"
#coords$color[coords$time=="3dpa"]="#E54E35"
#coords$color[coords$time=="7dpa"]= "#FDAE61"
#coords$color[coords$time=="10na"]="#EFE9C3"
coords$color[coords$time=="Tail_col_1wpa"]="#ABD9E9"
#coords$color[coords$time=="20na"]="#5C91C2"
coords$color[coords$time=="Tail_col_2wpa"]="#313695"


plot.data = coords
plot.data$order = sample(1:nrow(plot.data),replace = F)
plot.data = plot.data[order(plot.data$order),]

pdf("./Int_col_onlyBL_CT_SPRING_time.pdf",width=8,height=8)
plot(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color , xlab="", ylab="")

edge_start_coords = plot.data[match(edges$start, plot.data$Cell), c("x","y")]
edge_end_coords = plot.data[match(edges$end, plot.data$Cell), c("x","y")]
segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
	x1=edge_end_coords[,"x"], y1=edge_end_coords[,"y"], lwd=linewidth, col=linecol)

points(plot.data$x, plot.data$y, cex=cellcex, pch=16, col=plot.data$color)

dev.off()


########## plot many genes at once

#set genes to plot - cluster markers

tmp=c("twist2.S","runx1.L","mmp8.L","mmp11.L","tnc.L","cav1.L","fn1.S","tgfbi.L","acta2.L","acta2.S","myh11.L","postn.S","egfl6.S","ctgf.L","dio3.L","col2a1.S","col9a1.L","matn4.S","otos.L","runx2.L","sox9.L","sox9.S","prrx1.L","sox5","msx1.L","fgf10.L","bmp4.L","nog.L")
data.plot=as.data.frame(t(norm.data))


# create 200 colors (colstart --> colend)
colramp = colorRampPalette(brewer.pal(11,"Blues")[3:9])
colgrad = colramp(101)


##########

pdf("./Xen_CT_Interstitial_Tendon_Wc13_Blastema_SPRING_feature_Markers.pdf",width=9,height=8)

for (i in colnames(data.plot[,tmp])) {

  # normalize expression [0-1]
  exp = data.plot[,i]/max(data.plot[,i])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  coords[,i] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="", main = i)

    edge_start_coords = coords[match(edges$start, coords$Cell), c("x","y")]
    edge_end_coords = coords[match(edges$end, coords$Cell), c("x","y")]
    segments(x0=edge_start_coords[,"x"], y0=edge_start_coords[,"y"],
    	x=edge_end_coords[,"x"], y=edge_end_coords[,"y"], lwd=linewidth, col=linecol)
		
    points(coords$x, coords$y, cex=cellcex, pch=16, col=colgrad[coords[,i]], xlab="", ylab="")

}

dev.off()


######Calc pseudo time

cells_cart = rownames(coords)[coords$ident %in% c(10,8,0,2)]
cells_krt = rownames(coords)[coords$ident %in% c(5,13)]

#cells_bl = rownames(coords)[coords$ident %in% c(0,2)]
cells_mat = rownames(coords)[coords$ident %in% c(3,4,14)]
#cells_early_bl = rownames(coords)[coords$ident %in% c(1)]

#calculate pseudo time separate on branches

harmony.embedding.cart = as.data.frame(harmony.embedding[cells_cart,])
harmony.embedding.krt =  as.data.frame(harmony.embedding[cells_krt,])
#harmony.embedding.bl = as.data.frame(harmony.embedding[cells_bl,])
harmony.embedding.mat = as.data.frame(harmony.embedding[cells_mat,])
#harmony.embedding.early.bl = as.data.frame(harmony.embedding[cells_early_bl,])
harmony.embedding.rest = as.data.frame(harmony.embedding[!rownames(harmony.embedding) %in% c(rownames(harmony.embedding.cart),rownames(harmony.embedding.krt),rownames(harmony.embedding.mat)),])
#,rownames(harmony.embedding.early.bl),rownames(harmony.embedding.bl)
library("destiny")

#mat
harmony.embedding.mat$pt = 0
harmony.embedding.mat$rank = 0

#cart
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.cart),stringsAsFactor = F))

saveRDS(DatasetDM,"Xen_CT_Interstitial_Tendon_Blastema_DiffusionMap_Cart.RDS")

dpt = DPT(DatasetDM )

saveRDS(dpt,"Xen_CT_Interstitial_Tendon_Blastema_Pseudotime_Cart.RDS")

pseudotime_cart = dpt[random_root(dpt), ]

harmony.embedding.cart$pt = as.numeric(pseudotime_cart)
harmony.embedding.cart = harmony.embedding.cart[order(harmony.embedding.cart$pt),]
harmony.embedding.cart$rank = rev(1:nrow(harmony.embedding.cart))
harmony.embedding.cart$rank = harmony.embedding.cart$rank  + nrow(harmony.embedding.rest)  
#+ nrow(harmony.embedding.early.bl) + nrow(harmony.embedding.bl)


#bl
#DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.bl),stringsAsFactor = F))

#saveRDS(DatasetDM,"Xen_CT_Interstitial_Tendon_Blastema_DiffusionMap_bl.RDS")

#dpt = DPT(DatasetDM )

#saveRDS(dpt,"Xen_CT_Interstitial_Tendon_Blastema_Pseudotime_bl.RDS")

#pseudotime_bl = dpt[random_root(dpt), ]

#harmony.embedding.bl$pt = as.numeric(pseudotime_bl)
#harmony.embedding.bl = harmony.embedding.bl[order(harmony.embedding.bl$pt),]
#harmony.embedding.bl$rank = rev(1:nrow(harmony.embedding.bl))
#harmony.embedding.bl$rank = harmony.embedding.bl$rank  + nrow(harmony.embedding.rest) 
#+ nrow(harmony.embedding.early.bl)

#early bl
#DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.early.bl),stringsAsFactor = F))

#saveRDS(DatasetDM,"Xen_CT_Interstitial_Tendon_Blastema_DiffusionMap_early_bl.RDS")

#dpt = DPT(DatasetDM )

#saveRDS(dpt,"Xen_CT_Interstitial_Tendon_Blastema_Pseudotime_earlybl.RDS")

#pseudotime_early_bl = dpt[random_root(dpt), ]

#harmony.embedding.early.bl$pt = as.numeric(pseudotime_early_bl)
#harmony.embedding.early.bl = harmony.embedding.early.bl[order(harmony.embedding.early.bl$pt),]
#harmony.embedding.early.bl$rank = 1:nrow(harmony.embedding.early.bl)
#harmony.embedding.early.bl$rank = harmony.embedding.early.bl$rank  



#krt
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.krt),stringsAsFactor = F))

saveRDS(DatasetDM,"Xen_CT_Interstitial_Tendon_Blastema_DiffusionMap_Krt.RDS")

dpt = DPT(DatasetDM )

saveRDS(dpt,"Xen_CT_Interstitial_Tendon_Blastema_Pseudotime_Krt.RDS")

pseudotime_krt = dpt[random_root(dpt), ]

harmony.embedding.krt$pt = as.numeric(pseudotime_krt)
harmony.embedding.krt = harmony.embedding.krt[order(harmony.embedding.krt$pt),]
harmony.embedding.krt$rank = 1:nrow(harmony.embedding.krt)
harmony.embedding.krt$rank = harmony.embedding.krt$rank  + nrow(harmony.embedding.rest) + length(rownames(coords)[coords$ident %in% c(0,2)])
#+ nrow(harmony.embedding.early.bl)+  nrow(harmony.embedding.bl)

#rest
DatasetDM <- DiffusionMap(as.data.frame((harmony.embedding.rest),stringsAsFactor = F))

saveRDS(DatasetDM,"Xen_CT_Interstitial_Tendon_Blastema_DiffusionMap_prog.RDS")

dpt = DPT(DatasetDM )

saveRDS(dpt,"Xen_CT_Interstitial_Tendon_Blastema_Pseudotime_prog.RDS")

pseudotime_prog = dpt[random_root(dpt), ]

harmony.embedding.rest$pt = as.numeric(pseudotime_prog)
harmony.embedding.rest = harmony.embedding.rest[order(harmony.embedding.rest$pt),]
#revert order to fit time course (diff map does not know about start or end...)
harmony.embedding.rest$rank =rev(1:nrow(harmony.embedding.rest))  
harmony.embedding.rest$rank = harmony.embedding.rest$rank 
#+ nrow(harmony.embedding.early.bl)

pseudotime = rbind(harmony.embedding.mat,harmony.embedding.rest,harmony.embedding.cart,harmony.embedding.krt,stringsAsFactors = F)
#,harmony.embedding.early.bl,harmony.embedding.bl
plot.pt = cbind(coords,rank = pseudotime[match(rownames(coords),rownames(pseudotime)),"rank"], stringsAsFactors = F)

saveRDS(plot.pt,"Xen_CT_Interstitial_Tendon_Blastema_Final_SPRING_data.RDS")

colramp = colorRampPalette((brewer.pal(9,"Greys")[3:9]))
colgrad = colramp(101)

pdf("./Xen_CT_Interstitial_Tendon_Blastema_SPRING_PT_Rank.pdf",width=9,height=8)

  # normalize expression [0-1]
  exp = plot.pt [,"rank"]/max(plot.pt [,"rank"])

  # bin expression into 100 bins (round to 0.0*), some bins might be empty, add 1 to avoid bin=0
  plot.pt [,"bin"] = round(exp, digits=2) *100 +1
  
  #plot gradient

    plot(plot.pt $x, plot.pt $y, cex=cellcex, pch=16, col=colgrad[plot.pt [,"bin"]], xlab="", ylab="", main = "pseudotime")

    edge_start_plot.pt  = plot.pt [match(edges$start, plot.pt $Cell), c("x","y")]
    edge_end_plot.pt  = plot.pt [match(edges$end, plot.pt $Cell), c("x","y")]
    segments(x0=edge_start_plot.pt [,"x"], y0=edge_start_plot.pt [,"y"],
    	x=edge_end_plot.pt [,"x"], y=edge_end_plot.pt [,"y"], lwd=linewidth, col=linecol)
		
    points(plot.pt $x, plot.pt $y, cex=cellcex, pch=16, col=colgrad[plot.pt [,"bin"]], xlab="", ylab="")



dev.off()


```

##REdo with v2 and v3 0 wpa col1a2


#############Integrate all col1a2 0 to 2 wpa time points##############

```{r}
setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

#load annotation file

anno = read.delim("/mnt/SingleCellGenomics/genome_data/10xAxolotlGenome/AnnotationFile_AmexG_v6_chr_unscaffolded_CherryGFP_v1.1.csv",sep = "\t",header = F)
rownames(anno) = anno[,1]


############# remove weird transposon / unannotated genes
unannotated.transcripts = rownames(anno)[anno$V2 %in% anno$V2[grep("PEG10",anno$V2)] ]
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("L1TD1",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("RTL",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("GIN1",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("L1\\-RT",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("^N\\/A",anno$V2)] ], unannotated.transcripts))
unannotated.transcripts = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("^AMEX",anno$V2)] ], unannotated.transcripts))

anno = anno[!rownames(anno) %in% unannotated.transcripts,]


############# remove rp genes

rp.genes_AmexG = rownames(anno)[anno$V2 %in% anno$V2[grep("RPL",anno$V2)]]
rp.genes_AmexG = unique(c(rownames(anno)[anno$V2 %in% anno$V2[grep("RPS",anno$V2)] ], rp.genes_AmexG))

anno = anno[!rownames(anno) %in% rp.genes_AmexG,]


library(Seurat)

#load data sets

Tail_0wpav2 = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_col_0wpa_CT_SeuratObj.RDS")
Tail_0wpav3 = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_col1a2_All/Tail_col_0wpa_v3_CT_SeuratObj.RDS")
Tail_0wpa_twist = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_0wpa_twist/Tail_twist_0wpa_CT_SeuratObj.RDS")
Tail_1wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_1wpa/Tail_col_1wpa_CT_SeuratObj.RDS")
Tail_2wpa = readRDS(file = "~/Projects/SingleCellGroup/Axolotl/Tail_Project_2wpa/Tail_col_2wpa_CT_SeuratObj.RDS")


#merge ojects
#merge(x = NULL, y = NULL, add.cell.ids = NULL, merge.data = TRUE, project = "SeuratProject", ...)

Int_col_BL_CT = merge(Tail_0wpav2, y = c(Tail_0wpav3,Tail_1wpa, Tail_2wpa , Tail_0wpa_twist), add.cell.ids = c("wpa0_col_v2","wpa0_col_v3", "wpa1_col", "wpa2_col","wpa0_twist"), project = "Tail_col_Integration")


#parallize scaling
library(future)
plan("multiprocess", workers = 8)
plan()

#set maximum to 10GB for each worker
options(future.globals.maxSize = 50000 * 1024^2)


#get cell cycle genes

g2m.genes <- cc.genes$g2m.genes
g2m.contig <- intersect(rownames(anno)[anno$V2 %in% g2m.genes ] , rownames(Int_col_BL_CT) )

s.genes <- cc.genes$s.genes
s.contig <- intersect(rownames(anno)[anno$V2 %in% s.genes ] , rownames(Int_col_BL_CT) )

#cell cycle scoring
Int_col_BL_CT <- CellCycleScoring(Int_col_BL_CT, s.features = s.contig, g2m.features = g2m.contig, set.ident = TRUE)



#scale
#ScaleData(  object,  features = NULL,  assay = NULL,  vars.to.regress = NULL,  split.by = NULL,  model.use = "linear",  use.umi = FALSE,  do.scale = TRUE,  do.center = TRUE,  scale.max = 10,  block.size = 1000,  min.cells.to.block = 3000,  verbose = TRUE)

all.genes <- rownames(Int_col_BL_CT)

Int_col_BL_CT = ScaleData(  Int_col_BL_CT, features = all.genes ,vars.to.regress = c("nCount_RNA","nFeature_RNA","percent.mt","mCherry","eGFP","S.Score", "G2M.Score"))

#run PCA on all genes
#RunPCA(object, assay = NULL, features = NULL,  npcs = 50, rev.pca = FALSE, weight.by.var = TRUE, verbose = TRUE,  ndims.print = 1:5, nfeatures.print = 30, reduction.name = "pca",  reduction.key = "PC_", seed.use = 42)

Int_col_BL_CT = RunPCA(Int_col_BL_CT,features = all.genes,npcs = 100)

#plot PCA heatmaps
#DimHeatmap(pbmc, dims = 1, cells = 500, balanced = TRUE)

library(RColorBrewer)

pdf("Int_CT_v2v3_no3mpa_raw_PCAheatmaps.pdf",width=30,height=20)
DimHeatmap(Int_col_BL_CT, dims = 1:16, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT, dims = 17:32, cells = 1000, balanced = TRUE, ncol = 4,  fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
DimHeatmap(Int_col_BL_CT, dims = 33:48, cells = 1000, balanced = TRUE,  ncol = 4, fast = F) + scale_fill_gradientn( colors = c(brewer.pal(11,"Spectral")[11:1]))
dev.off()

#elbow plot
pdf("Int_CT_v2v3_no3mpa_raw_PCelbow.pdf",width=20,height=10)
ElbowPlot(Int_col_BL_CT, ndims = 100)
dev.off()

#Cluster cells
Int_col_BL_CT<- FindNeighbors(Int_col_BL_CT, dims = 1:50)
Int_col_BL_CT <- FindClusters(Int_col_BL_CT, resolution = 0.6)

#run UMAP
Int_col_BL_CT = RunUMAP(Int_col_BL_CT,dims = 1:50)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_CT_v2v3_no3mpa_raw_UMAP_PC50_res0.6.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT, reduction = 'umap', pt.size = 3)
DimPlot(object = Int_col_BL_CT, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
FeaturePlot(Int_col_BL_CT, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD002658","AMEX60DD052322"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()



plan("multiprocess", workers = 8)
library(tidyverse)

markers <- FindAllMarkers(Int_col_BL_CT, only.pos = TRUE,  logfc.threshold = 0.3)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_log2FC))

write.csv(markers.anno,"Int_CT_v2v3_no3mpa_raw_allMarker.csv")

#some macrophage contamination identified --> remove

Int_col_BL_CT <- subset(Int_col_BL_CT, idents = c(0:19) )

#Cluster cells
Int_col_BL_CT<- FindNeighbors(Int_col_BL_CT, dims = 1:50)
Int_col_BL_CT <- FindClusters(Int_col_BL_CT, resolution = 0.6)

#run UMAP
Int_col_BL_CT = RunUMAP(Int_col_BL_CT,dims = 1:50)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_CT_v2v3_no3mpa_raw_UMAP_PC50_res0.8.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT, reduction = 'umap', pt.size = 3)
DimPlot(object = Int_col_BL_CT, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
FeaturePlot(Int_col_BL_CT, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD002658","AMEX60DD052322"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()

#one cluster is only driven by low nUMI -->remove

Int_col_BL_CT <- FindClusters(Int_col_BL_CT, resolution = 2)

Int_col_BL_CT <- subset(Int_col_BL_CT, idents = c(0:33,35:37) )

#Cluster cells
Int_col_BL_CT<- FindNeighbors(Int_col_BL_CT, dims = 1:50)
Int_col_BL_CT <- FindClusters(Int_col_BL_CT, resolution = 0.6)

#run UMAP
Int_col_BL_CT = RunUMAP(Int_col_BL_CT,dims = 1:50)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_CT_v2v3_no3mpa_raw_UMAP_PC50_res0.6.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT, reduction = 'umap', pt.size = 3)
DimPlot(object = Int_col_BL_CT, reduction = 'umap',group.by = "orig.ident", pt.size = 3)
FeaturePlot(Int_col_BL_CT, pt.size = 1, features = c("nCount_RNA","percent.mt","AMEX60DD002658","AMEX60DD052322"), order = T, cols = c(brewer.pal(9,"Greys")[9:2],brewer.pal(9,"Reds")[2:9]))
dev.off()


#They cluster only based on batch --> integrate

saveRDS(Int_col_BL_CT, file = "Int_CT_v2v3_no3mpa_raw_SeuratObj.RDS")

```

Integrate using Harmony

```{r}

library(Seurat)
library(RColorBrewer)
library(harmony)

setwd("~/Projects/SingleCellGroup/Axolotl/Tail_Project_Integration_col_0_1_2wpa/")

#Harmony

Int_col_BL_CT_int = RunHarmony(object = Int_col_BL_CT,group.by.vars= "orig.ident", assay.use = "RNA", dims.use = 1:50, plot_convergence = TRUE, max.iter.cluster = 30 ,max.iter.harmony=50,theta=30,lambda=5)


#Cluster cells
Int_col_BL_CT_int<- FindNeighbors(Int_col_BL_CT_int, dims = 1:50, reduction = "harmony")
Int_col_BL_CT_int <- FindClusters(Int_col_BL_CT_int, resolution = 0.4)


#run UMAP
Int_col_BL_CT_int = RunUMAP(Int_col_BL_CT_int,dims = 1:50, reduction = "harmony", min.dist = 0.1,  spread = 1,)


#plot result
#DimPlot(object, dims = c(1, 2), cells = NULL, cols = NULL,  pt.size = NULL, reduction = NULL, group.by = NULL,  split.by = NULL, shape.by = NULL, order = NULL, label = FALSE,  label.size = 4, repel = FALSE, cells.highlight = NULL,  cols.highlight = "red", sizes.highlight = 1, na.value = "grey50",  combine = TRUE)

pdf("Int_col_BL_CT_HarmonyTimeInt_t30_l5_PC50_UMAP_PC50_res0.4.pdf",width=20,height=20)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap', pt.size = 3,shuffle = T)
DimPlot(object = Int_col_BL_CT_int, reduction = 'umap',group.by = "orig.ident", pt.size = 3,shuffle = T)
FeaturePlot(Int_col_BL_CT_int, pt.size = 1, raster = T, features = c("percent.mt","nFeature_RNA","nCount_RNA"), order = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))))
dev.off()


breakdown<-as.data.frame(table(Int_col_BL_CT_int@meta.data$seurat_clusters, Int_col_BL_CT_int@meta.data$orig.ident)) 

pdf("Int_col_BL_CT_HarmonyTimeInt_t30_l5_PC50_UMAP_Histogram.pdf", width=10,height=5)
ggplot(breakdown, aes(x = Var1,y = Freq, fill = Var2)) + theme_bw() + geom_col(position="fill")  + xlab("Cluster")    + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) 
dev.off()

gene_ids = c("AMEX60DD021181","AMEX60DD021249","AMEX60DD051586","AMEX60DD018450","AMEX60DD034707","AMEX60DD052891","AMEX60DD033766","AMEX60DD027155","AMEX60DD009987","AMEX60DD022442","AMEX60DD040818","AMEX60DD037674","AMEX60DD007644","AMEX60DD052322","AMEX60DD055512","AMEX60DD029436","AMEX60DD018705","AMEX60DD024616","AMEX60DD038300","AMEX60DD012132","AMEX60DD031236","AMEX60DD002658","AMEX60DD037123","AMEX60DD036008","AMEX60DD004178","AMEX60DD048972","AMEX60DD029426","AMEX60DD005459","AMEX60DD043904","AMEX60DD049488","AMEX60DD033105","AMEX60DD038318","AMEX60DD031842","AMEX60DD047873","AMEX60DD034687","AMEX60DD008143","AMEX60DD048840")
gene_names =c("LFNG","UNCX","HES5","PRRX1","RSPO3","FGF8","TBX18","TCF15","MEOX1","MEOX2","SCX","TNMD","MYF5","PAX7","TWIST2","TWIST3","F13B","LTBP4","F13A1","GREM1","SOX9","OTOS","CHRDL1","PKDCC","ACAN","CNMD","COL2A1","MATN1","SPP1","MMP13","MATN3","DSP","COL5A3","COL4A1","LAMA2","ITIH5","KLF5")




gg_Fig <- FeaturePlot(Int_col_BL_CT_int, pt.size = 1, features = gene_ids,order = T,raster = T, cols = c("grey",rev(viridis_pal(option = "mako")(12))), repel=T)
gg_Fig <- lapply( 1:length(gene_ids), function(x) { gg_Fig[[x]] + labs(title=gene_names[x]) })

pdf("Int_col_BL_CT_HarmonyTimeInt_t30_l5_PC50_UMAP_feature_GeneralMarker.pdf",width=22,height=15)
CombinePlots( gg_Fig )
dev.off()









library(future)
plan("multiprocess", workers = 8)
library(tidyverse)

markers <- FindAllMarkers(Int_col_BL_CT_int, only.pos = TRUE,  logfc.threshold = 0.3)
markers.anno = markers
markers.anno = merge(markers.anno,anno, by.x="gene" , by.y="V1")
markers.anno$ID = markers.anno$gene

#markers.anno = markers.anno[order(as.numeric(markers.anno$cluster)),]
markers.anno = markers.anno  %>% arrange(cluster , desc(avg_logFC))

write.csv(markers.anno,"Int_col_BL_CT_HarmonyTimeInt_ClustMarker.csv")



saveRDS(Int_col_BL_CT_int, file = "Int_col_BL_CT_HarmonyTimeInt_final_SeuratObj.RDS")



